
---
title: "Paper-Paper Network Interaction (same number)"

output:
  html_document:
    toc: true
    toc_depth: 6
    toc_float: true
    collapsed: true
    smooth_scroll: true
    number_sections: false
---

```{r, echo=FALSE, warning=FALSE, comment=NA, message=FALSE}
library(knitr)
opts_chunk$set(echo=FALSE, warning=FALSE, comment=NA, message=FALSE, width=1000, cache=TRUE)
options(width=1200)
```

```{r}
list_to_df=function(lista_prova){
df=as.data.frame(matrix(nrow=length(unlist(lista_prova)), ncol=2))

df$V1=unlist(lista_prova)

V2=c()

for(i in 1:length(lista_prova)){
  V2=c(V2, rep(names(lista_prova)[i], length(lista_prova[[i]])))
}

df$V2=V2

return(df)

}


list_ensembl=list()

list_adjpval=list()

list_log2FC=list()

pthreshold=0.1

```

# Overview

Intellectual Disabilities share similarities such as structural defects at the neural level, and transcriptomic deregulation of cognitive pathway. For this reason, they could also be targeted similarly from a therapeutic point of view.
I propose a metanalysis collecting all available transcriptomic data on IDs.
I will start collecting all RNA-seq experiment and create a network interaction were each node is a dataset and the strength of the connection will be given by the correlation of the transcriptomic deregulation when comparing the disease group to the control group.
I processed all dataset similary mapping and counting reads with `STAR` on and performing DE expression with `DESeq2`.

## Down Syndrome Datasets

### iPSCs from monozygotic twins discordant for trisomy 21

The dataset was taken from this article:

https://www.ncbi.nlm.nih.gov/pubmed/24375627

4 sample euploid vs 3 trisomic

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP032928.RData")
require(DESeq2)
summary(res)
resSRP032928=res
universe_human=rownames(resSRP032928)
resSRP032928=as.data.frame(resSRP032928)
resSRP032928=resSRP032928[order(resSRP032928$padj),]
resSRP032928=resSRP032928[complete.cases(resSRP032928),]
resSRP032928_sig=resSRP032928[which(resSRP032928$padj<pthreshold),]
summary(resLFC)
resSRP032928_sigs=rownames(resSRP032928_sig)
resLFCSRP032928=resLFC

list_ensembl$GSE52249=resSRP032928_sigs
list_adjpval$GSE52249=resSRP032928_sig$padj
list_log2FC$GSE52249=resSRP032928_sig$log2FoldChange
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```



### Letourneau GEDD's paper


#### Primary fibroblast from discordant twins

Forearm primary fetal skin fibroblasts were collected post mortem from the T1DS and T2N discordant twins at 16 fetal weeks.
Three replicate experiments (Rep0, Rep1 and Rep2) were performed from independent cultures of the same fibroblasts. The fourth replicate (Rep3) was performed using the RNA material of Rep0. Primary fibroblasts from MZ1 and MZ2 healthy monozygotic twins were derived from umbilical cord tissue collected as part of the GenCord project in the University Hospitals of Geneva. RNA isolation was performed after 10 or 13 passages of the twins’ primary fibroblasts. 
.

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP039348_twins.RData")

summary(res)
resSRP039348_twins=res


resSRP039348_twins=as.data.frame(resSRP039348_twins)
resSRP039348_twins=resSRP039348_twins[order(resSRP039348_twins$padj),]
resSRP039348_twins=resSRP039348_twins[complete.cases(resSRP039348_twins),]
resSRP039348_twins_sig=resSRP039348_twins[which(resSRP039348_twins$padj < pthreshold),]

resSRP039348_twins_sigs=rownames(resSRP039348_twins_sig)

summary(resLFC)

resLFCSSRP039348_twins=resLFC


list_ensembl$GSE55504=resSRP039348_twins_sigs
list_adjpval$GSE55504=resSRP039348_twins_sig$padj
list_log2FC$GSE55504=resSRP039348_twins_sig$log2FoldChange
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


#### Primary fibroblast from unrelated individuals

Primary fibroblasts from unrelated individual (discordant for chr21) one euploid and one with trisomy 21).

Primary skin fibroblasts from trisomy 21 and normal unrelated individuals were taken from ref. 15 (GM08447, GM05756, GM00969, GM00408, GM02036, GM03377, GM03440, PM9726F, GM04616, AG07409, AG06872, AG06922, GM02767, AG08941, AG08942, D-S99124M).

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP039348_fibroblasts.RData")

summary(res)
resSRP039348_fibroblasts=res

summary(resLFC)

resLFCSSRP039348_fibroblasts=resLFC




resSRP039348_fibroblasts=as.data.frame(resSRP039348_fibroblasts)
resSRP039348_fibroblasts=resSRP039348_fibroblasts[order(resSRP039348_fibroblasts$padj),]
resSRP039348_fibroblasts=resSRP039348_fibroblasts[complete.cases(resSRP039348_fibroblasts),]
resSRP039348_fibroblasts_sig=resSRP039348_fibroblasts[which(resSRP039348_fibroblasts$padj< pthreshold),]
resSRP039348_fibroblasts_sigs=rownames(resSRP039348_fibroblasts_sig)


list_ensembl$GSE55504_fibroblasts=resSRP039348_fibroblasts_sigs
list_adjpval$GSE55504_fibroblasts=resSRP039348_fibroblasts_sig$padj
list_log2FC$GSE55504_fibroblasts=resSRP039348_fibroblasts_sig$log2FoldChange
```

```{r}

vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```
https://www.ncbi.nlm.nih.gov/pubmed/24740065


#### iPSCs from twins

Primary fetal skin fibroblasts isolated from T1DS and T2N (10 passages) were used to establish normal (Twin-N-iPSCs) and Down’s syndrome (Twin-DS-iPSCs) induced pluripotent stem cells (iPS cells). These iPS cells were generated by transducing the parental fibroblasts with polycistronic lentiviral vectors expressing OCT4, SOX2, KLF4 and MYC reprogramming factors. iPS cells were grown for 17 passages before the RNA isolation step. iPSCs from twin 1 (T21) and twin 2 (euploid).

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP039348_ipsc.RData")

summary(res)
resSRP039348_ipsc=res

summary(resLFC)

resLFCSSRP039348_ipsc=resLFC



resSRP039348_ipsc=as.data.frame(resSRP039348_ipsc)
resSRP039348_ipsc=resSRP039348_ipsc[order(resSRP039348_ipsc$log2FoldChange, decreasing = T),]
resSRP039348_ipsc=resSRP039348_ipsc[complete.cases(resSRP039348_ipsc),]
resSRP039348_ipsc_sig=resSRP039348_ipsc[which(resSRP039348_ipsc$padj < pthreshold),]
resSRP039348_ipsc_sigs=rownames(resSRP039348_ipsc_sig)


list_ensembl$GSE55504_ipsc=resSRP039348_ipsc_sigs
list_adjpval$GSE55504_ipsc=resSRP039348_ipsc_sig$padj
list_log2FC$GSE55504_ipsc=resSRP039348_ipsc_sig$log2FoldChange
```



#### MEFs

Mouse embryonic fibroblasts (MEFs) were obtained from 14.5-day-old mouse embryos. MEFs from Ts65Dn and WT mice:

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP039348_mef.RData")

summary(res)
resSRP039348_mef=res
universe_mouse=rownames(resSRP039348_mef)
summary(resLFC)

resLFCSSRP039348_mef=resLFC

resSRP039348_mef=as.data.frame(resSRP039348_mef)
require(biomaRt)
human = useMart("ensembl", dataset = "hsapiens_gene_ensembl",host="oct2018.archive.ensembl.org")
mouse = useMart("ensembl", dataset = "mmusculus_gene_ensembl",host="oct2018.archive.ensembl.org") 

resSRP039348_mef=resSRP039348_mef[order(resSRP039348_mef$log2FoldChange, decreasing = T),]
resSRP039348_mef=resSRP039348_mef[complete.cases(resSRP039348_mef),]
resSRP039348_mef_sig=resSRP039348_mef[which(resSRP039348_mef$padj<pthreshold),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = universe_mouse, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=mouse_human[which(mouse_human$human %in% rownames(resLFCSSRP039348_twins)),]

###eliminate mouse identifiers that maps for more than one human gene

mouse_human=mouse_human[which(!(mouse_human$human %in% mouse_human$human[duplicated(mouse_human$human)])),]

mouse_human=mouse_human[which(!(mouse_human$mouse %in% mouse_human$mouse[duplicated(mouse_human$mouse)])),]

universe_common=mouse_human$human



list_ensembl$GSE55504_mef=list_ensembl$GSE55504_ipsc
list_adjpval$GSE55504_mef=list_ensembl$GSE55504_ipsc
list_log2FC$GSE55504_mef=list_ensembl$GSE55504_ipsc
```



### Down Syndrom Placenta

Four disomic samples from first trimester CVS, one disomic from CVS; 
versus
3 trisomic second trimester CVS, one trisomic First trimeste

Reference paper: https://www.ncbi.nlm.nih.gov/pubmed/23754950

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP017123.RData")

summary(res)
resSRP017123=res

summary(resLFC)

resLFCSRP017123=resLFC


resSRP017123=as.data.frame(resSRP017123)
resSRP017123=resSRP017123[order(resSRP017123$padj),]
resSRP017123=resSRP017123[complete.cases(resSRP017123),]
resSRP017123_sig=resSRP017123[which(resSRP017123$padj < pthreshold),]

resSRP017123_sigs=rownames(resSRP017123_sig)


list_ensembl$GSE42142=resSRP017123_sigs
list_adjpval$GSE42142=resSRP017123_sig$padj
list_log2FC$GSE42142=resSRP017123_sig$log2FoldChange
```

```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])


data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment, shape=colData$sex)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=colData$trimester, shape=colData$sex)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


### Mouse embryonic fibroblasts

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE64840

Fibroblast from 18.5-day-old Embryos from Ts65Dn and WT (3 reps each).

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP051977.RData")

summary(res)
resSRP051977=as.data.frame(res)

summary(resLFC)


resSRP051977=resSRP051977[order(resSRP051977$padj),]
resSRP051977_sig=resSRP051977[complete.cases(resSRP051977),]
resSRP051977_sig=resSRP051977[which(resSRP051977$padj < pthreshold),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = rownames(resSRP051977_sig), mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")
resSRP051977_sig$ens=rownames(resSRP051977_sig)

mouse_human=merge(resSRP051977_sig,mouse_human, by.x="ens", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$padj, abs(mouse_human$log2FoldChange), decreasing=c(FALSE, TRUE)),]
resSRP051977_sig_hum=mouse_human[!duplicated(mouse_human$human),]

resSRP051977_sigs_hum=unique(resSRP051977_sig_hum$human)
  
length(resSRP051977_sigs_hum)


list_ensembl$GSE64840=resSRP051977_sigs_hum
list_adjpval$GSE64840=resSRP051977_sig_hum$padj
list_log2FC$GSE64840=resSRP051977_sig_hum$log2FoldChange
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


### Genetically matched human induced pluripotent stem cells and derived neurons from disomic or trisomic for chromosome 21

https://trace.ddbj.nig.ac.jp/DRASearch/study?acc=SRP113668


induced pluripotent stem cells (iPSCs) derived from a single individual with Down syndrome. These cells were originally derived by Li et al, who genetically targeted chromosome 21 in trisomic iPSCs, allowing selection of disomic sibling iPSC clones. Analyses were conducted on trisomic/disomic cell pairs maintained as iPSCs or differentiated into cortical neuronal cultures.

#### 6 IPSC samples with 3 biological repeats for trisomic samples and 3 biological repeats for disomic samples

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP113668_ipsc.RData")

summary(res)
resSRP113668_ipsc=as.data.frame(res)

summary(resLFC)

resLFCSRP113668_ipsc=resLFC





resSRP113668_ipsc=as.data.frame(resSRP113668_ipsc)
resSRP113668_ipsc=resSRP113668_ipsc[order(resSRP113668_ipsc$padj),]
resSRP113668_ipsc=resSRP113668_ipsc[complete.cases(resSRP113668_ipsc),]
resSRP113668_ipsc_sig=resSRP113668_ipsc[which(resSRP113668_ipsc$padj < pthreshold),]
resSRP113668_ipsc_sigs=rownames(resSRP113668_ipsc_sig)


list_ensembl$GSE101942_ipsc=resSRP113668_ipsc_sigs
list_adjpval$GSE101942_ipsc=resSRP113668_ipsc_sig$padj
list_log2FC$GSE101942_ipsc=resSRP113668_ipsc_sig$log2FoldChange

```

```{r}

vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


#### 6 IPSC derived neuronal samples with 3 biological repeats for trisomic samples and 3 biological repeats for disomic samples

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP113668_neurons.RData")

summary(res)
resSRP113668_neurons=res

summary(resLFC)

resLFCSRP113668_neurons=resLFC



resSRP113668_neurons=as.data.frame(resSRP113668_neurons)
resSRP113668_neurons=resSRP113668_neurons[order(resSRP113668_neurons$padj),]
resSRP113668_neurons=resSRP113668_neurons[complete.cases(resSRP113668_neurons),]
resSRP113668_neurons_sig=resSRP113668_neurons[which(resSRP113668_neurons$padj < pthreshold),]
resSRP113668_neurons_sigs=rownames(resSRP113668_neurons_sig)


list_ensembl$GSE101942_neurons=resSRP113668_neurons_sigs
list_adjpval$GSE101942_neurons=resSRP113668_neurons_sig$padj
list_log2FC$GSE101942_neurons=resSRP113668_neurons_sig$log2FoldChange


```


```{r}

vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

### GSE6283

Specific transcriptional changes in human fetuses with autosomal trisomies.
We analyzed whole genome expression profiles in cultured amniocytes (AC) and chorionic villus cells (CV) from pregnancies with a normal karyotype and with trisomies of human chromosomes 21

```{r}
require(affy)
require(limma)

phenodata=data.frame(celltype=c("amniocyte","amniocyte","amniocyte", "CVS", "CVS", "amniocyte","amniocyte","amniocyte", "CVS", "CVS", "CVS","CVS","CVS","CVS","CVS"),
                     ploidy=c(rep("disomic", 5), rep("trisomic", 6), rep("disomic", 4)), sex=c("male", "male", "male", "female", "female", "male", "male", "male", "male", "female", "female", "female", "male", "female", "male"))
rownames(phenodata)=c("amniocyte_normalkaryotype_rep1","amniocyte_normalkaryotype_rep2", "amniocyte_normalkaryotype_rep3", "chorion villus cells_normalkaryotype_rep1", "chorion villus cells_normalkaryotype_rep2","amniocytes_trisomy21_rep1", "amniocytes_trisomy21_rep2", "amniocytes_trisomy21_rep3", "chorion villus cells_trisomy21_rep1","chorion villus cells_trisomy21_rep2", "chorion villus cells_trisomy21_rep3", "chorion villus cells_normalkaryotype_rep3", "chorion villus cells_normalkaryotype_rep5", "chorion villus cells_normalkaryotype_rep4", "chorion villus cells_normalkaryotype_rep6")


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE6283_RAW/", sampleNames = c("amniocyte_normalkaryotype_rep1","amniocyte_normalkaryotype_rep2", "amniocyte_normalkaryotype_rep3", "chorion villus cells_normalkaryotype_rep1", "chorion villus cells_normalkaryotype_rep2","amniocytes_trisomy21_rep1", "amniocytes_trisomy21_rep2", "amniocytes_trisomy21_rep3", "chorion villus cells_trisomy21_rep1","chorion villus cells_trisomy21_rep2", "chorion villus cells_trisomy21_rep3", "chorion villus cells_normalkaryotype_rep3", "chorion villus cells_normalkaryotype_rep5", "chorion villus cells_normalkaryotype_rep4", "chorion villus cells_normalkaryotype_rep6"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("hgu133plus2.db")

ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(paste(pData(eset)$celltype, pData(eset)$ploidy,sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(amniocyte.trisomic-amniocyte.disomic,CVS.trisomic - CVS.disomic,levels=design)


corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)
aw=arrayWeights((eset)[nomi,], design)

fit <- lmFit(exprs(eset)[nomi,], design, block=phenodata$sex, cor=corfit$consensus, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


amnio=topTable(contr.fit, coef=1, number=dim(eset)[1])

cvs=topTable(contr.fit, coef=2, number=dim(eset)[1])

# pc=prcomp(t(exprs(eset)), scale.=T, center=T)
amnio_sig=amnio[which(amnio$adj.P.Val< pthreshold),]
cvs_sig=cvs[which(cvs$adj.P.Val< pthreshold),]

GSE6283amnio_sigs=unique(unlist(ids[rownames(amnio_sig)]))[!is.na(unique(unlist(ids[rownames(amnio_sig)])))]
GSE6283cvs_sigs=unique(unlist(ids[rownames(cvs_sig)]))[!is.na(unique(unlist(ids[rownames(cvs_sig)])))]


amnio_sig$id=rownames(amnio_sig)

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

amnio_sig=merge(amnio_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

amnio_sig=amnio_sig[order(amnio_sig$adj.P.Val, abs(amnio_sig$logFC), decreasing=c(FALSE, TRUE)),]
amnio_sig=amnio_sig[!duplicated(amnio_sig$V1),]



cvs_sig$id=rownames(cvs_sig)
cvs_sig=merge(cvs_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

cvs_sig=cvs_sig[order(cvs_sig$adj.P.Val, abs(cvs_sig$logFC), decreasing=c(FALSE, TRUE)),]
cvs_sig=cvs_sig[!duplicated(cvs_sig$V1),]  


list_ensembl$GSE6283_amnio=GSE6283amnio_sigs
list_adjpval$GSE6283_amnio=amnio_sig$adj.P.Val
list_log2FC$GSE6283_amnio=amnio_sig$logFC

list_ensembl$GSE6283_cvs=GSE6283cvs_sigs
list_adjpval$GSE6283_cvs=cvs_sig$adj.P.Val
list_log2FC$GSE6283_cvs=cvs_sig$logFC
```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)), pch=as.numeric(as.factor(pData(eset)$celltype)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))
legend("bottomleft", legend=levels(pData(eset)$celltype), pch=as.numeric(as.factor(levels(pData(eset)$celltype))))

```

### GSE1294

Six Ts1Cje and six normal littermate (2N) mouse brains at postnatal day 0 with probe sets representing approximately 11,300 genes revealed that the number of expressed genes and their identities in Ts1Cje mice were almost same in 2N mice. Notably, the expression levels of most genes in the trisomic region were increased approximately 1.5-fold, and the top 24 most consistently over-expressed genes in the Ts1Cje mice were all located in the trisomic region. 

```{r}

phenodata=data.frame(
                     ploidy=c(rep("disomic", 6), rep("trisomic", 6)))
rownames(phenodata)=c("2N1", "2N2", "2N3","2N4","2N5","2N6", "Ts1Cje1","Ts1Cje2","Ts1Cje3", "Ts1Cje4" , "Ts1Cje5", "Ts1Cje6")


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE1294_RAW/Av2/", compress=T, sampleNames = c("2N1", "2N2", "2N3","2N4","2N5","2N6", "Ts1Cje1","Ts1Cje2","Ts1Cje3", "Ts1Cje4" , "Ts1Cje5", "Ts1Cje6"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("mgu74av2.db")

ids <-mget(featureNames(eset), mgu74av2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights((eset)[nomi,], design)


fit <-lmFit(exprs(eset)[nomi,], design, weights = aw)
contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE1294=topTable(contr.fit, coef=1, number=dim(eset)[1])


GSE1294_sig=GSE1294[which(GSE1294$adj.P.Val< pthreshold),]

GSE1294_sig$id=rownames(GSE1294_sig)
GSE1294_sigs=unique(unlist(ids[rownames(GSE1294_sig)]))[!is.na(unique(unlist(ids[rownames(GSE1294_sig)])))]

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE1294_sig=merge(GSE1294_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

GSE1294_sig=GSE1294_sig[order(GSE1294_sig$adj.P.Val, abs(GSE1294_sig$logFC), decreasing=c(FALSE, TRUE)),]

GSE1294_sig=GSE1294_sig[!duplicated(GSE1294_sig$V1),]



mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE1294_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE1294_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE1294_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE1294_sigs_hum=unique(GSE1294_sig_hum$human)
  
length(GSE1294_sigs_hum)


list_ensembl$GSE1294=GSE1294_sigs_hum
list_adjpval$GSE1294=GSE1294_sig_hum$adj.P.Val
list_log2FC$GSE1294=GSE1294_sig_hum$logFC



```
```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)), pch=as.numeric(as.factor(pData(eset)$celltype)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))
```


```{r}

data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE1294_RAW/Bv2/", compress=T, sampleNames = c("2N1", "2N2", "2N3","2N4","2N5","2N6", "Ts1Cje1","Ts1Cje2","Ts1Cje3", "Ts1Cje4" , "Ts1Cje5", "Ts1Cje6"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("mgu74bv2.db")

ids <-mget(featureNames(eset), mgu74bv2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)


aw=arrayWeights((eset)[nomi,], design)

fit <-lmFit(exprs(eset)[nomi,], design, weights=aw)
contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE1294b=topTable(contr.fit, coef=1, number=dim(eset)[1])


GSE1294b_sig=GSE1294b[which(GSE1294b$adj.P.Val< pthreshold),]

GSE1294b_sig$id=rownames(GSE1294b_sig)

GSE1294b_sigs=unique(unlist(ids[rownames(GSE1294b_sig)]))[!is.na(unique(unlist(ids[rownames(GSE1294b_sig)])))]



ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE1294b_sig=merge(GSE1294b_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

GSE1294b_sig=GSE1294b_sig[order(GSE1294b_sig$adj.P.Val, abs(GSE1294b_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE1294b_sig=GSE1294b_sig[!duplicated(GSE1294b_sig$V1),]




mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE1294b_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE1294b_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE1294b_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE1294b_sigs_hum=unique(GSE1294b_sig_hum$human)
  
length(GSE1294b_sigs_hum)

list_ensembl$GSE1294b=GSE1294b_sigs_hum
list_adjpval$GSE1294b=GSE1294b_sig_hum$adj.P.Val
list_log2FC$GSE1294b=GSE1294b_sig_hum$logFC

```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)), pch=as.numeric(as.factor(pData(eset)$celltype)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))
```


### GSE1611

transcriptome of the cerebellum of the Ts1Cje mouse model of Down syndrome during postnatal development using microarrays


```{r}
phenodata=data.frame(ploidy=c(rep(c("disomic"),6),rep(c("trisomic"),6)),time=c(rep(c("P0", "P0", "P15" ,"P15", "P30" ,"P30"),2)))
data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE1611_RAW/", compress=T, sampleNames = c("2N1", "2N2", "2N3","2N4","2N5","2N6", "Ts1Cje1","Ts1Cje2","Ts1Cje3", "Ts1Cje4" , "Ts1Cje5", "Ts1Cje6"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), mgu74av2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

aw=arrayWeights((eset)[nomi,], design)

contrasts <-makeContrasts(trisomic-disomic,levels=design)


corfit<-duplicateCorrelation(eset,design, ndups=1, block=phenodata$time)
fit <- lmFit(eset, design, block=phenodata$time, cor=corfit$consensus, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE1611=topTable(contr.fit, coef=1, number=dim(eset)[1])


GSE1611_sig=GSE1611[which(GSE1611$adj.P.Val< pthreshold),]

GSE1611_sigs=unique(unlist(ids[rownames(GSE1611_sig)]))[!is.na(unique(unlist(ids[rownames(GSE1611_sig)])))]

GSE1611_sig$id=rownames(GSE1611_sig)


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE1611_sig=merge(GSE1611_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

GSE1611_sig=GSE1611_sig[order(GSE1611_sig$adj.P.Val, abs(GSE1611_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE1611_sig=GSE1611_sig[!duplicated(GSE1611_sig$V1),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE1611_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE1611_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE1611_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE1611_sigs_hum=unique(GSE1611_sig_hum$human)
  

list_ensembl$GSE1611=GSE1611_sigs_hum
list_adjpval$GSE1611=GSE1611_sig_hum$adj.P.Val
list_log2FC$GSE1611=GSE1611_sig_hum$logFC
```


```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)), pch=as.numeric(as.factor(pData(eset)$time)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))
legend("bottomleft", legend=levels(as.factor(pData(eset)$time)), pch=as.numeric(as.factor(levels(as.factor(pData(eset)$time)))))

```


### GSE5390

Expression profiling of human adult postmortem brain tissue from Down syndrome and healthy control subjects (dorsolateral prefrontal cortex)

```{r}
phenodata=data.frame(sex=c("female", "female", "female", "male", "female", "male","male","male", "female","male","male","male","male", "male", "male"),
                         ploidy=c(rep("trisomic", 7), rep("disomic",8)))


rownames(phenodata)=c("DS1", "DS2", "D3", "DS4", "DS5", "DS6", "DS7", "C1", "C2", "C3", "C4", "C5", "C6", "C7", "C8")


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE5390_RAW/", sampleNames =c("DS1", "DS2", "D3", "DS4", "DS5", "DS6", "DS7", "C1", "C2", "C3", "C4", "C5", "C6", "C7", "C8"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("hgu133a.db")

ids <-mget(featureNames(eset), hgu133aENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)
aw=arrayWeights((eset)[nomi,], design)



contrasts <-makeContrasts(trisomic-disomic,levels=design)


corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)
fit <- lmFit(exprs(eset)[nomi,], design, block=phenodata$sex, cor=corfit$consensus, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE5390=topTable(contr.fit, coef=1, number=dim(eset)[1])

GSE5390_sig=GSE5390[which(GSE5390$adj.P.Val< pthreshold),]

GSE5390_sig$id=rownames(GSE5390_sig)

GSE5390_sigs=unique(unlist(ids[rownames(GSE5390_sig)]))[!is.na(unique(unlist(ids[rownames(GSE5390_sig)])))]


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE5390_sig=merge(GSE5390_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

GSE5390_sig=GSE5390_sig[order(GSE5390_sig$adj.P.Val, abs(GSE5390_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE5390_sig=GSE5390_sig[!duplicated(GSE5390_sig$V1),]

list_ensembl$GSE5390=GSE5390_sigs
list_adjpval$GSE5390=GSE5390_sig$adj.P.Val
list_log2FC$GSE5390=GSE5390_sig$logFC

```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)), pch=as.numeric(as.factor(pData(eset)$sex)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))
legend("bottomleft", legend=levels(as.factor(pData(eset)$sex)), pch=as.numeric(as.factor(levels(as.factor(pData(eset)$sex)))))

```

### Tc1 mice

Embrio 14.5. Transchromosomic, carries copy of Hsa21 on mouse background.

```{r}
phenodata=data.frame(ploidy=c("trisomic", "disomic", "trisomic"))


rownames(phenodata)=c("Tc1_1", "Wt", "Tc1_2")


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-MEXP-409.raw.1/", sampleNames =c("Tc1_1", "Wt", "Tc1_2"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("hgu133a.db")

ids <-mget(featureNames(eset), hgu133aENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


EMEXP409=topTable(contr.fit, coef=1, number=dim(eset)[1])

EMEXP409_sig=EMEXP409[which(EMEXP409$adj.P.Val< pthreshold),]

EMEXP409_sig$id=rownames(EMEXP409_sig)

EMEXP409_sigs=unique(unlist(ids[rownames(EMEXP409_sig)]))[!is.na(unique(unlist(ids[rownames(EMEXP409_sig)])))]


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

EMEXP409_sig=merge(EMEXP409_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

EMEXP409_sig=EMEXP409_sig[order(EMEXP409_sig$adj.P.Val, abs(EMEXP409_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMEXP409_sig=EMEXP409_sig[!duplicated(EMEXP409_sig$V1),]

list_ensembl$EMEXP409=EMEXP409_sigs
list_adjpval$EMEXP409=EMEXP409_sig$adj.P.Val
list_log2FC$EMEXP409=EMEXP409_sig$logFC
```



```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))

```


### GSE1397

human fetal samples. For TS21, regions assayed were cerebrum, cerebellum, heart, and cerebrum-derived astrocyte cell lines.

```{r}
require("GEOquery")
GSE1397 <- getGEO('GSE1397',GSEMatrix=TRUE, GSElimits = c(1,23))
eset=GSE1397[[1]]
eset=log2(exprs(eset[,1:22]))

colnames(eset)=pData(GSE1397$GSE1397_series_matrix.txt.gz)$title[1:22]

tissue=c(rep("cerebrum", 8), rep("cerebellum", 6),rep("astrocyte",4),  rep("heart",4))

ploidy=c(rep("trisomic", 4),rep("disomic", 7),rep("trisomic", 3),rep("disomic", 2),rep("trisomic", 2),rep("disomic", 2),rep("trisomic", 2))


ids <-mget(featureNames(GSE1397[[1]]), hgu133aENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]

TS <- as.factor( paste(tissue, ploidy, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(cerebrum.trisomic-cerebrum.disomic,cerebellum.trisomic-cerebellum.disomic,astrocyte.trisomic-astrocyte.disomic,heart.trisomic-heart.disomic,levels=design)

aw=arrayWeights((eset)[nomi,], design)

fit <- lmFit(eset[nomi,], design, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE1397cerebrum=topTable(contr.fit, coef=1, number=dim(eset)[1])
GSE1397cerebelum=topTable(contr.fit, coef=2, number=dim(eset)[1])
GSE1397astrocyte=topTable(contr.fit, coef=3, number=dim(eset)[1])
GSE1397heart=topTable(contr.fit, coef=4, number=dim(eset)[1])

GSE1397cerebrum_sig=GSE1397cerebrum[which(GSE1397cerebrum$adj.P.Val< pthreshold),]
GSE1397cerebelum_sig=GSE1397cerebelum[which(GSE1397cerebelum$adj.P.Val< pthreshold),]
GSE1397astrocyte_sig=GSE1397astrocyte[which(GSE1397astrocyte$adj.P.Val< pthreshold),]
GSE1397heart_sig=GSE1397heart[which(GSE1397heart$adj.P.Val< pthreshold),]

GSE1397cerebrum_sigs=unique(unlist(ids[rownames(GSE1397cerebrum_sig)]))[!is.na(unique(unlist(ids[rownames(GSE1397cerebrum_sig)])))]
GSE1397cerebelum_sigs=unique(unlist(ids[rownames(GSE1397cerebelum_sig)]))[!is.na(unique(unlist(ids[rownames(GSE1397cerebelum_sig)])))]
GSE1397astrocyte_sigs=unique(unlist(ids[rownames(GSE1397astrocyte_sig)]))[!is.na(unique(unlist(ids[rownames(GSE1397astrocyte_sig)])))]
GSE1397heart_sigs=unique(unlist(ids[rownames(GSE1397heart_sig)]))[!is.na(unique(unlist(ids[rownames(GSE1397heart_sig)])))]

GSE1397cerebrum_sig$id=rownames(GSE1397cerebrum_sig)
GSE1397cerebelum_sig$id=rownames(GSE1397cerebelum_sig)
GSE1397astrocyte_sig$id=rownames(GSE1397astrocyte_sig)
GSE1397heart_sig$id=rownames(GSE1397heart_sig)

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE1397cerebrum_sig=merge(GSE1397cerebrum_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE1397cerebrum_sig=GSE1397cerebrum_sig[order(GSE1397cerebrum_sig$adj.P.Val, abs(GSE1397cerebrum_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE1397cerebrum_sig=GSE1397cerebrum_sig[!duplicated(GSE1397cerebrum_sig$V1),]

GSE1397cerebrum_sigs=unique(GSE1397cerebrum_sig$id)[!is.na(unique(GSE1397cerebrum_sig$id))]
list_ensembl$GSE1397.crb=GSE1397cerebrum_sigs
list_adjpval$GSE1397.crb=GSE1397cerebrum_sig$adj.P.Val
list_log2FC$GSE1397.crb=GSE1397cerebrum_sig$logFC


GSE1397cerebelum_sig=merge(GSE1397cerebelum_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE1397cerebelum_sig=GSE1397cerebelum_sig[order(GSE1397cerebelum_sig$adj.P.Val, abs(GSE1397cerebelum_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE1397cerebelum_sig=GSE1397cerebelum_sig[!duplicated(GSE1397cerebelum_sig$V1),]


list_ensembl$GSE1397.crbl=GSE1397cerebelum_sigs
list_adjpval$GSE1397.crbl=GSE1397cerebelum_sig$adj.P.Val
list_log2FC$GSE1397.crbl=GSE1397cerebelum_sig$logFC

GSE1397astrocyte_sig=merge(GSE1397astrocyte_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE1397astrocyte_sig=GSE1397astrocyte_sig[order(GSE1397astrocyte_sig$adj.P.Val, abs(GSE1397astrocyte_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE1397astrocyte_sig=GSE1397astrocyte_sig[!duplicated(GSE1397astrocyte_sig$V1),]


list_ensembl$GSE1397.astro=GSE1397astrocyte_sigs
list_adjpval$GSE1397.astro=GSE1397astrocyte_sig$adj.P.Val
list_log2FC$GSE1397.astro=GSE1397astrocyte_sig$logFC

GSE1397heart_sig=merge(GSE1397heart_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE1397heart_sig=GSE1397heart_sig[order(GSE1397heart_sig$adj.P.Val, abs(GSE1397heart_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE1397heart_sig=GSE1397heart_sig[!duplicated(GSE1397heart_sig$V1),]

list_ensembl$GSE1397.heart=GSE1397heart_sigs
list_adjpval$GSE1397.heart=GSE1397heart_sig$adj.P.Val
list_log2FC$GSE1397.heart=GSE1397heart_sig$logFC


```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(ploidy)), pch=as.numeric(as.factor(tissue)))
legend("topleft", legend=levels(as.factor(ploidy)), fill=as.numeric(as.factor(levels(as.factor(ploidy)))))
legend("bottomleft", legend=levels(as.factor(tissue)), pch=as.numeric(as.factor(levels(as.factor(tissue)))))

```



## GSE9762

Euploid and trisomic human fibroblasts.

```{r}

phenodata=data.frame(ploidy=c("disomic", "disomic", "disomic", "trisomic", "trisomic",  "trisomic", "disomic", "disomic", "trisomic", "trisomic"))


rownames(phenodata)=c("D1", "D2", "D3", "T1", "T2", "T3", "D4", "D5", "T4", "T5")


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE9762_RAW/", sampleNames =c("D1", "D2", "D3", "T1", "T2", "T3", "D4", "D5", "T4", "T5"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

aw=arrayWeights((eset)[nomi,], design)


contrasts <-makeContrasts(trisomic-disomic,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE9762=topTable(contr.fit, coef=1, number=dim(eset)[1])

GSE9762_sig=GSE9762[which(GSE9762$adj.P.Val< pthreshold),]

GSE9762_sig$id=rownames(GSE9762_sig)

GSE9762_sigs=unique(unlist(ids[rownames(GSE9762_sig)]))[!is.na(unique(unlist(ids[rownames(GSE9762_sig)])))]

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE9762_sig=merge(GSE9762_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

GSE9762_sig=GSE9762_sig[order(GSE9762_sig$adj.P.Val, abs(GSE9762_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE9762_sig=GSE9762_sig[!duplicated(GSE9762_sig$V1),]

list_ensembl$GSE9762=GSE9762_sigs
list_adjpval$GSE9762=GSE9762_sig$adj.P.Val
list_log2FC$GSE9762=GSE9762_sig$logFC
```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))

```



### GSE16176

differences between second trimester Down syndrome (DS) and euploid fetuses, we compared gene expression in uncultured amniotic fluid supernatant samples.



```{r}
phenodata=data.frame(sex=c(rep("female",4), rep("male", 10)),
                         ploidy=rep(c("disomic","trisomic"), 7))


rownames(phenodata)=c("CF1", "DF1", "CF2", "DF2", "CM1", "DM1",  "CM2", "DM2", "CM3", "DM3", "CM4", "DM4", "CM5", "DM5")


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE16176_RAW/", sampleNames =c("CF1", "DF1", "CF2", "DF2", "CM1", "DM1",  "CM2", "DM2", "CM3", "DM3", "CM4", "DM4", "CM5", "DM5"),phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]
# ids=as.character(unlist(ids))



TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights((eset)[nomi,], design)
corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)
fit <- lmFit(exprs(eset)[nomi,], design, block=phenodata$sex, cor=corfit$consensus, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE16176=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])

GSE16176_sig=GSE16176[which(GSE16176$adj.P.Val< pthreshold),]


GSE16176_sigs=unique(unlist(ids[rownames(GSE16176_sig)]))[!is.na(unique(unlist(ids[rownames(GSE16176_sig)])))]


GSE16176_sig$id=rownames(GSE16176_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE16176_sig=merge(GSE16176_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

GSE16176_sig=GSE16176_sig[order(GSE16176_sig$adj.P.Val, abs(GSE16176_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE16176_sig=GSE16176_sig[!duplicated(GSE16176_sig$V1),]

if(is.null(GSE16176_sigs)){
list_ensembl$GSE16176=character(0)}else{list_ensembl$GSE16176=GSE16176_sigs}
list_adjpval$GSE16176=GSE16176_sig$adj.P.Val
list_log2FC$GSE16176=GSE16176_sig$logFC

```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)), pch=as.numeric(as.factor(pData(eset)$sex)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))
legend("bottomleft", legend=levels(as.factor(pData(eset)$sex)), pch=as.numeric(as.factor(levels(as.factor(pData(eset)$sex)))))
```



### E-MEXP-654

Transcription profiling of mouse embryonic stem cells carrying an extra copy of human chromosome 21. The transchromosomic cells are called 47-1, and the parental WT cells are called D3.




```{r}
phenodata=data.frame(ploidy=c(rep("trisomic", 4), rep("disomic", 4)))



data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-MEXP-654.raw.1/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), mgu74av2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]
# ids=as.character(unlist(ids))



TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights((eset)[nomi,], design)

fit <- lmFit(exprs(eset)[nomi,], design, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


EMEXP654=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])

EMEXP654_sig=EMEXP654[which(EMEXP654$adj.P.Val< pthreshold),]

EMEXP654_sig$id=rownames(EMEXP654_sig)

EMEXP654_sigs=unique(unlist(ids[rownames(EMEXP654_sig)]))[!is.na(unique(unlist(ids[rownames(EMEXP654_sig)])))]

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

EMEXP654_sig=merge(EMEXP654_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

EMEXP654_sig=EMEXP654_sig[order(EMEXP654_sig$adj.P.Val, abs(EMEXP654_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMEXP654_sig=EMEXP654_sig[!duplicated(EMEXP654_sig$V1),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = EMEXP654_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(EMEXP654_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
EMEXP654_sig_hum=mouse_human[!duplicated(mouse_human$human),]

EMEXP654_sigs_hum=unique(EMEXP654_sig_hum$human)
  
length(EMEXP654_sigs_hum)


list_ensembl$EMEXP654=EMEXP654_sigs_hum
list_adjpval$EMEXP654=EMEXP654_sig_hum$adj.P.Val
list_log2FC$EMEXP654=EMEXP654_sig_hum$logFC
```


```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)))
legend("topleft", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))
```

### GSE35561

We modeled human Trisomy 21 primitive hematopoiesis using induced pluripotent stem cells (iPSCs). Primitive multipotent progenitor populations generated from Trisomy 21 iPSCs showed normal proliferative capacity and megakaryocyte production, enhanced erythropoiesis and reduced myeloid development compared to euploid iPSCs.D41+/235+ iPSC-derived progenitor cells were flow cytometrically sorted for RNA extraction and hybridization onto Affymetrix microarrays.


```{r}
phenodata=data.frame(ploidy=c(rep("disomic", 3), rep("trisomic", 3)))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE35561_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

anno=getBM(attributes=c("ensembl_gene_id", "affy_hugene_1_0_st_v1"),filters="affy_hugene_1_0_st_v1", values =featureNames(eset), mart = human)
nomi=unique(anno$affy_hugene_1_0_st_v1)


TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


aw=arrayWeights((eset)[as.character(nomi),], design)

contrasts <-makeContrasts(trisomic-disomic,levels=design)


fit <- lmFit(exprs(eset)[as.character(nomi),], design, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE35561=topTable(contr.fit, coef=1, number=dim(exprs(eset)[as.character(nomi),])[1])

GSE35561_sig=GSE35561[which(GSE35561$adj.P.Val< pthreshold),]

GSE35561_sigs=unique(unlist(ids[rownames(GSE35561_sig)]))[!is.na(unique(unlist(ids[rownames(GSE35561_sig)])))]
#no enirchment

list_ensembl$GSE35561=character(0)
list_adjpval$GSE35561=numeric(0)
list_log2FC$GSE35561=numeric(0)

```

```{r}
 plotMDS((eset)[as.character(nomi),], col=as.numeric(as.factor(pData(eset)$ploidy)))
legend("topleft", legend=levels(as.factor(pData(eset)$ploidy)), fill=as.numeric(as.factor(levels(as.factor(pData(eset)$ploidy)))))
```

### GSE39159

Soleus muscles were obtained from four male Ts65Dn (DS) mice and four male colony control mice. Animals were apparently healthy and approximately five months old.

```{r}
phenodata=data.frame(ploidy=c(rep("trisomic", 4), rep("disomic", 4)))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE39159_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

# require("mogene10sttranscriptcluster.db")
# ids <-mget(featureNames(eset), mogene10sttranscriptclusterENSEMBL, ifnotfound=NA)
# nomi=names(ids)[!is.na(ids)]
# # ids=as.character(unlist(ids))

anno=getBM(attributes=c("ensembl_gene_id", "affy_mogene_1_0_st_v1"),filters="affy_mogene_1_0_st_v1", values =featureNames(eset), mart = mouse)

TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)
nomi=as.character(unique(anno$affy_mogene_1_0_st_v1))

aw=arrayWeights((eset)[nomi,], design)

contrasts <-makeContrasts(trisomic-disomic,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE39159=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])

GSE39159_sig=GSE39159[which(GSE39159$adj.P.Val< pthreshold),]

GSE39159_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE39159_sig))])

GSE39159_sig$id=rownames(GSE39159_sig)


GSE39159_sig=merge(GSE39159_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)

GSE39159_sig=GSE39159_sig[order(GSE39159_sig$adj.P.Val, abs(GSE39159_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE39159_sig=GSE39159_sig[!duplicated(GSE39159_sig$ensembl_gene_id),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE39159_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE39159_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE39159_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE39159_sigs_hum=unique(GSE39159_sig_hum$human)
  


list_ensembl$GSE39159=GSE39159_sigs_hum
list_adjpval$GSE39159=GSE39159_sig_hum$adj.P.Val
list_log2FC$GSE39159=GSE39159_sig_hum$logFC
```


```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)))
legend("topright", legend=levels(pData(eset)$ploidy), fill=as.numeric(as.factor(levels(pData(eset)$ploidy))))

```

### GSE110064

Disomic and trisomic isogenic iPS cells were differentiated in parallel, and after co-culture with MS5 cells, CD45+ CD34- CD19+ cells were isolated by FACS. RNA was extracted using Trizol. Global gene expression analysis by microarray was performed on the Human HG U133 Plus 2.0 platform. Each group contains three biological replicates.

```{r}
phenodata=data.frame(ploidy=c(rep("disomic", 3), rep("trisomic", 3)))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE110064_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]
# ids=as.character(unlist(ids))

TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

aw=arrayWeights((eset)[nomi,], design)


contrasts <-makeContrasts(trisomic-disomic,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE110064=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])

GSE110064_sig=GSE110064[which(GSE110064$adj.P.Val< pthreshold),]

GSE110064_sigs=unique(unlist(ids[rownames(GSE110064_sig)]))[!is.na(unique(unlist(ids[rownames(GSE110064_sig)])))]

GSE110064_sig$id=rownames(GSE110064_sig)


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE110064_sig=merge(GSE110064_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)

GSE110064_sig=GSE110064_sig[order(GSE110064_sig$adj.P.Val, abs(GSE110064_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE110064_sig=GSE110064_sig[!duplicated(GSE110064_sig$V1),]


if(is.null(GSE110064_sigs)){
list_ensembl$GSE110064=character(0)}else{list_ensembl$GSE110064=GSE110064_sigs}

list_adjpval$GSE110064=GSE110064_sig$adj.P.Val
list_log2FC$GSE110064=GSE110064_sig$logFC

```

```{r}
  plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))


```


### GSE99135

At embryonic day 15.5 (E15.5), we analyzed forebrain gene expression in Dp16 (n=6) and euploid littermates (n=6), Ts65Dn (n=5) and euploid littermates (n=6), and Ts1Cje (n=5) and euploid littermates (n=5) using Affymetrix mouse gene 1.0 ST array. At adulthood, we investigated gene expression and pathway abnormalities in the cerebral cortex, hippocampus and cerebellum obtained from 6-7 month old Dp16 (n=5) and euploid littermates (n=5), Ts65Dn (n=5) and euploid littermates (n=5), and Ts1Cje (n=5) and euploid littermates (n=5) using the same Affymetrix arrays.

```{r}
phenodata=data.frame(ploidy=c(rep("disomic", 6), rep("trisomic", 6),rep("disomic", 7), rep("trisomic", 6),rep("disomic", 5), rep("trisomic", 5),rep(c(rep("trisomic", 5), rep("disomic", 5)),8),rep("trisomic", 6), rep("disomic", 7)), area=c(rep("E15", 35),rep(c(rep("Crblm", 10),rep("Crtx", 10),rep("Hpcmp", 10)),2),rep("Crblm", 10),rep("Crtx", 11),rep("Hpcmp", 12)), type=c(rep("Dp16",12),rep("Ts65Dn",13),rep("Ts1Cje",10),rep("Dp16",30),rep("Ts65Dn",30),rep("Ts1Cje",33)),
                     
                     sex=c("Female","Female","Male","Male","Male","Male","Female","Female","Male","Male","Male","Male","Male","Male","Male","Female","Female","Female","Female","Male","Male","Male","Male","Female","Female","Male","Female","Female","Female","Male","Male","Female","Male","Male","Female","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male","Male"))

rm(data)
data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE99135_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

rm(data)
anno=getBM(attributes=c("ensembl_gene_id", "affy_mogene_1_0_st_v1"),filters="affy_mogene_1_0_st_v1", values =featureNames(eset), mart = mouse)

# ids=as.character(unlist(ids))
nomi=as.character(unique(anno$affy_mogene_1_0_st_v1))


TS <- as.factor( paste(paste(pData(eset)$type, pData(eset)$ploidy, sep="."), pData(eset)$area, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(Dp16.trisomic.Crblm-Dp16.disomic.Crblm,
                          Dp16.trisomic.E15-Dp16.disomic.E15,
                          Dp16.trisomic.Hpcmp-Dp16.disomic.Hpcmp,
                          Dp16.trisomic.Crtx-Dp16.disomic.Crtx,
                          Ts1Cje.trisomic.Crblm-Ts1Cje.disomic.Crblm,
                          Ts1Cje.trisomic.E15-Ts1Cje.disomic.E15,
                          Ts1Cje.trisomic.Hpcmp-Ts1Cje.disomic.Hpcmp,
                          Ts1Cje.trisomic.Crtx-Ts1Cje.disomic.Crtx,
                          Ts65Dn.trisomic.Crblm-Ts65Dn.disomic.Crblm,
                          Ts65Dn.trisomic.E15-Ts65Dn.disomic.E15,
                          Ts65Dn.trisomic.Hpcmp-Ts65Dn.disomic.Hpcmp,
                          Ts65Dn.trisomic.Crtx-Ts65Dn.disomic.Crtx,
                          levels=design)


aw=arrayWeights((eset)[nomi,], design)
# corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)
fit <- lmFit(exprs(eset)[nomi,], design,  weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))

GSE99135dp16.Crblm=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])
GSE99135dp16.E15=topTable(contr.fit, coef=2, number=dim(exprs(eset)[nomi,])[1])
GSE99135dp16.Hpcmp=topTable(contr.fit, coef=3, number=dim(exprs(eset)[nomi,])[1])
GSE99135dp16.Crtx=topTable(contr.fit, coef=4, number=dim(exprs(eset)[nomi,])[1])

GSE99135Ts1Cje.Crblm=topTable(contr.fit, coef=5, number=dim(exprs(eset)[nomi,])[1])
GSE99135Ts1Cje.E15=topTable(contr.fit, coef=6, number=dim(exprs(eset)[nomi,])[1])
GSE99135Ts1Cje.Hpcmp=topTable(contr.fit, coef=7, number=dim(exprs(eset)[nomi,])[1])
GSE99135Ts1Cje.Crtx=topTable(contr.fit, coef=8, number=dim(exprs(eset)[nomi,])[1])


GSE99135Ts65Dn.Crblm=topTable(contr.fit, coef=9, number=dim(exprs(eset)[nomi,])[1])
GSE99135Ts65Dn.E15=topTable(contr.fit, coef=10, number=dim(exprs(eset)[nomi,])[1])
GSE99135Ts65Dn.Hpcmp=topTable(contr.fit, coef=11, number=dim(exprs(eset)[nomi,])[1])
GSE99135Ts65Dn.Crtx=topTable(contr.fit, coef=12, number=dim(exprs(eset)[nomi,])[1])

GSE99135dp16.Crblm_sig=GSE99135dp16.Crblm[which(GSE99135dp16.Crblm$adj.P.Val< pthreshold),]
GSE99135dp16.E15_sig=GSE99135dp16.Crblm[which(GSE99135dp16.E15$adj.P.Val< pthreshold),]
GSE99135dp16.Hpcmp_sig=GSE99135dp16.Hpcmp[which(GSE99135dp16.Hpcmp$adj.P.Val< pthreshold),]
GSE99135dp16.Crtx_sig=GSE99135dp16.Crtx[which(GSE99135dp16.Crtx$adj.P.Val< pthreshold),]

GSE99135Ts1Cje.Crblm_sig=GSE99135Ts1Cje.Crblm[which(GSE99135Ts1Cje.Crblm$adj.P.Val< pthreshold),]
GSE99135Ts1Cje.E15_sig=GSE99135Ts1Cje.Crblm[which(GSE99135Ts1Cje.E15$adj.P.Val< pthreshold),]
GSE99135Ts1Cje.Hpcmp_sig=GSE99135Ts1Cje.Hpcmp[which(GSE99135Ts1Cje.Hpcmp$adj.P.Val< pthreshold),]
GSE99135Ts1Cje.Crtx_sig=GSE99135Ts1Cje.Crtx[which(GSE99135Ts1Cje.Crtx$adj.P.Val< pthreshold),]


GSE99135Ts65Dn.Crblm_sig=GSE99135Ts65Dn.Crblm[which(GSE99135Ts65Dn.Crblm$adj.P.Val< pthreshold),]
GSE99135Ts65Dn.E15_sig=GSE99135Ts65Dn.Crblm[which(GSE99135Ts65Dn.E15$adj.P.Val< pthreshold),]
GSE99135Ts65Dn.Hpcmp_sig=GSE99135Ts65Dn.Hpcmp[which(GSE99135Ts65Dn.Hpcmp$adj.P.Val< pthreshold),]
GSE99135Ts65Dn.Crtx_sig=GSE99135Ts65Dn.Crtx[which(GSE99135Ts65Dn.Crtx$adj.P.Val< pthreshold),]

GSE99135dp16.Crblm_sig$id=rownames(GSE99135dp16.Crblm_sig)
GSE99135dp16.E15_sig$id=rownames(GSE99135dp16.E15_sig)
GSE99135dp16.Hpcmp_sig$id=rownames(GSE99135dp16.Hpcmp_sig)
GSE99135dp16.Crtx_sig$id=rownames(GSE99135dp16.Crtx_sig)
GSE99135Ts1Cje.Crblm_sig$id=rownames(GSE99135Ts1Cje.Crblm_sig)
GSE99135Ts1Cje.E15_sig$id=rownames(GSE99135Ts1Cje.E15_sig)
GSE99135Ts1Cje.Hpcmp_sig$id=rownames(GSE99135Ts1Cje.Hpcmp_sig)
GSE99135Ts1Cje.Crtx_sig$id=rownames(GSE99135Ts1Cje.Crtx_sig)
GSE99135Ts65Dn.Crblm_sig$id=rownames(GSE99135Ts65Dn.Crblm_sig)
GSE99135Ts65Dn.E15_sig$id=rownames(GSE99135Ts65Dn.E15_sig)
GSE99135Ts65Dn.Hpcmp_sig$id=rownames(GSE99135Ts65Dn.Hpcmp_sig)
GSE99135Ts65Dn.Crtx_sig$id=rownames(GSE99135Ts65Dn.Crtx_sig)


GSE99135dp16.Crblm_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135dp16.Crblm_sig))])
GSE99135dp16.E15_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135dp16.E15_sig))])
GSE99135dp16.Hpcmp_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135dp16.Hpcmp_sig))])
GSE99135dp16.Crtx_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135dp16.Crtx_sig))])

GSE99135Ts1Cje.Crblm_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts1Cje.Crblm_sig))])
GSE99135Ts1Cje.E15_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts1Cje.E15_sig))])
GSE99135Ts1Cje.Hpcmp_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts1Cje.Hpcmp_sig))])
GSE99135Ts1Cje.Crtx_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts1Cje.Crtx_sig))])


GSE99135Ts65Dn.Crblm_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts65Dn.Crblm_sig))])
GSE99135Ts65Dn.E15_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts65Dn.E15_sig))])
GSE99135Ts65Dn.Hpcmp_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts65Dn.Hpcmp_sig))])
GSE99135Ts65Dn.Crtx_sigs=unique(anno$ensembl_gene_id[which(anno$affy_mogene_1_0_st_v1 %in% rownames(GSE99135Ts65Dn.Crtx_sig))])


GSE99135dp16.E15_sig=merge(GSE99135dp16.E15_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135dp16.E15_sig=GSE99135dp16.E15_sig[order(GSE99135dp16.E15_sig$adj.P.Val, abs(GSE99135dp16.E15_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.E15_sig=GSE99135dp16.E15_sig[!duplicated(GSE99135dp16.E15_sig$ensembl_gene_id),]

GSE99135Ts65Dn.E15_sig=merge(GSE99135Ts65Dn.E15_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts65Dn.E15_sig=GSE99135Ts65Dn.E15_sig[order(GSE99135Ts65Dn.E15_sig$adj.P.Val, abs(GSE99135Ts65Dn.E15_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.E15_sig=GSE99135Ts65Dn.E15_sig[!duplicated(GSE99135Ts65Dn.E15_sig$ensembl_gene_id),]

GSE99135Ts1Cje.E15_sig=merge(GSE99135Ts1Cje.E15_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts1Cje.E15_sig=GSE99135Ts1Cje.E15_sig[order(GSE99135Ts1Cje.E15_sig$adj.P.Val, abs(GSE99135Ts1Cje.E15_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.E15_sig=GSE99135Ts1Cje.E15_sig[!duplicated(GSE99135Ts1Cje.E15_sig$ensembl_gene_id),]

GSE99135dp16.Crblm_sig=merge(GSE99135dp16.Crblm_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135dp16.Crblm_sig=GSE99135dp16.Crblm_sig[order(GSE99135dp16.Crblm_sig$adj.P.Val, abs(GSE99135dp16.Crblm_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.Crblm_sig=GSE99135dp16.Crblm_sig[!duplicated(GSE99135dp16.Crblm_sig$ensembl_gene_id),]

GSE99135Ts65Dn.Crblm_sig=merge(GSE99135Ts65Dn.Crblm_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts65Dn.Crblm_sig=GSE99135Ts65Dn.Crblm_sig[order(GSE99135Ts65Dn.Crblm_sig$adj.P.Val, abs(GSE99135Ts65Dn.Crblm_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.Crblm_sig=GSE99135Ts65Dn.Crblm_sig[!duplicated(GSE99135Ts65Dn.Crblm_sig$ensembl_gene_id),]

GSE99135Ts1Cje.Crblm_sig=merge(GSE99135Ts1Cje.Crblm_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts1Cje.Crblm_sig=GSE99135Ts1Cje.Crblm_sig[order(GSE99135Ts1Cje.Crblm_sig$adj.P.Val, abs(GSE99135Ts1Cje.Crblm_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.Crblm_sig=GSE99135Ts1Cje.Crblm_sig[!duplicated(GSE99135Ts1Cje.Crblm_sig$ensembl_gene_id),]

GSE99135dp16.Crtx_sig=merge(GSE99135dp16.Crtx_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135dp16.Crtx_sig=GSE99135dp16.Crtx_sig[order(GSE99135dp16.Crtx_sig$adj.P.Val, abs(GSE99135dp16.Crtx_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.Crtx_sig=GSE99135dp16.Crtx_sig[!duplicated(GSE99135dp16.Crtx_sig$ensembl_gene_id),]

GSE99135Ts65Dn.Crtx_sig=merge(GSE99135Ts65Dn.Crtx_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts65Dn.Crtx_sig=GSE99135Ts65Dn.Crtx_sig[order(GSE99135Ts65Dn.Crtx_sig$adj.P.Val, abs(GSE99135Ts65Dn.Crtx_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.Crtx_sig=GSE99135Ts65Dn.Crtx_sig[!duplicated(GSE99135Ts65Dn.Crtx_sig$ensembl_gene_id),]

GSE99135Ts1Cje.Crtx_sig=merge(GSE99135Ts1Cje.Crtx_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts1Cje.Crtx_sig=GSE99135Ts1Cje.Crtx_sig[order(GSE99135Ts1Cje.Crtx_sig$adj.P.Val, abs(GSE99135Ts1Cje.Crtx_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.Crtx_sig=GSE99135Ts1Cje.Crtx_sig[!duplicated(GSE99135Ts1Cje.Crtx_sig$ensembl_gene_id),]

GSE99135dp16.Hpcmp_sig=merge(GSE99135dp16.Hpcmp_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135dp16.Hpcmp_sig=GSE99135dp16.Hpcmp_sig[order(GSE99135dp16.Hpcmp_sig$adj.P.Val, abs(GSE99135dp16.Hpcmp_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.Hpcmp_sig=GSE99135dp16.Hpcmp_sig[!duplicated(GSE99135dp16.Hpcmp_sig$ensembl_gene_id),]

GSE99135Ts65Dn.Hpcmp_sig=merge(GSE99135Ts65Dn.Hpcmp_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts65Dn.Hpcmp_sig=GSE99135Ts65Dn.Hpcmp_sig[order(GSE99135Ts65Dn.Hpcmp_sig$adj.P.Val, abs(GSE99135Ts65Dn.Hpcmp_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.Hpcmp_sig=GSE99135Ts65Dn.Hpcmp_sig[!duplicated(GSE99135Ts65Dn.Hpcmp_sig$ensembl_gene_id),]

GSE99135Ts1Cje.Hpcmp_sig=merge(GSE99135Ts1Cje.Hpcmp_sig, anno, by.x="id", by.y="affy_mogene_1_0_st_v1", all.x=TRUE, all.y=FALSE)
GSE99135Ts1Cje.Hpcmp_sig=GSE99135Ts1Cje.Hpcmp_sig[order(GSE99135Ts1Cje.Hpcmp_sig$adj.P.Val, abs(GSE99135Ts1Cje.Hpcmp_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.Hpcmp_sig=GSE99135Ts1Cje.Hpcmp_sig[!duplicated(GSE99135Ts1Cje.Hpcmp_sig$ensembl_gene_id),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135dp16.E15_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135dp16.E15_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135dp16.E15_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.E15_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135dp16.E15_sigs_hum=unique(GSE99135dp16.E15_sig_hum$human)
list_ensembl$GSE99135dp16.E15=GSE99135dp16.E15_sigs_hum
list_adjpval$GSE99135dp16.E15=GSE99135dp16.E15_sig_hum$adj.P.Val
list_log2FC$GSE99135dp16.E15=GSE99135dp16.E15_sig_hum$logFC


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts65Dn.E15_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts65Dn.E15_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135Ts65Dn.E15_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.E15_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts65Dn.E15_sigs_hum=unique(GSE99135Ts65Dn.E15_sig_hum$human)
list_ensembl$GSE99135Ts65Dn.E15=GSE99135Ts65Dn.E15_sigs_hum
list_adjpval$GSE99135Ts65Dn.E15=GSE99135Ts65Dn.E15_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts65Dn.E15=GSE99135Ts65Dn.E15_sig_hum$logFC



mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts1Cje.E15_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts1Cje.E15_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135Ts1Cje.E15_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.E15_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts1Cje.E15_sigs_hum=unique(GSE99135Ts1Cje.E15_sig_hum$human)
list_ensembl$GSE99135Ts1Cje.E15=GSE99135Ts1Cje.E15_sigs_hum
list_adjpval$GSE99135Ts1Cje.E15=GSE99135Ts1Cje.E15_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts1Cje.E15=GSE99135Ts1Cje.E15_sig_hum$logFC




mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135dp16.Crblm_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135dp16.Crblm_sigs_hum=unique(mouse_human$human)

mouse_human=merge(GSE99135dp16.Crblm_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.Crblm_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135dp16.Crblm_sigs_hum=unique(GSE99135dp16.Crblm_sig_hum$human)
list_ensembl$GSE99135dp16.Crblm=GSE99135dp16.Crblm_sigs_hum
list_adjpval$GSE99135dp16.Crblm=GSE99135dp16.Crblm_sig_hum$adj.P.Val
list_log2FC$GSE99135dp16.Crblm=GSE99135dp16.Crblm_sig_hum$logFC


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts65Dn.Crblm_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts65Dn.Crblm_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135Ts65Dn.Crblm_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.Crblm_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts65Dn.Crblm_sigs_hum=unique(GSE99135Ts65Dn.Crblm_sig_hum$human)
list_ensembl$GSE99135Ts65Dn.Crblm=GSE99135Ts65Dn.Crblm_sigs_hum
list_adjpval$GSE99135Ts65Dn.Crblm=GSE99135Ts65Dn.Crblm_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts65Dn.Crblm=GSE99135Ts65Dn.Crblm_sig_hum$logFC



mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts1Cje.Crblm_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts1Cje.Crblm_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135Ts1Cje.Crblm_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.Crblm_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts1Cje.Crblm_sigs_hum=unique(GSE99135Ts1Cje.Crblm_sig_hum$human)
list_ensembl$GSE99135Ts1Cje.Crblm=GSE99135Ts1Cje.Crblm_sigs_hum
list_adjpval$GSE99135Ts1Cje.Crblm=GSE99135Ts1Cje.Crblm_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts1Cje.Crblm=GSE99135Ts1Cje.Crblm_sig_hum$logFC




mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135dp16.Crtx_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135dp16.Crtx_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135dp16.Crtx_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.Crtx_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135dp16.Crtx_sigs_hum=unique(GSE99135dp16.Crtx_sig_hum$human)
list_ensembl$GSE99135dp16.Crtx=GSE99135dp16.Crtx_sigs_hum
list_adjpval$GSE99135dp16.Crtx=GSE99135dp16.Crtx_sig_hum$adj.P.Val
list_log2FC$GSE99135dp16.Crtx=GSE99135dp16.Crtx_sig_hum$logFC


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts65Dn.Crtx_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts65Dn.Crtx_sigs_hum=unique(mouse_human$human)

mouse_human=merge(GSE99135Ts65Dn.Crtx_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.Crtx_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts65Dn.Crtx_sigs_hum=unique(GSE99135Ts65Dn.Crtx_sig_hum$human)
list_ensembl$GSE99135Ts65Dn.Crtx=GSE99135Ts65Dn.Crtx_sigs_hum
list_adjpval$GSE99135Ts65Dn.Crtx=GSE99135Ts65Dn.Crtx_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts65Dn.Crtx=GSE99135Ts65Dn.Crtx_sig_hum$logFC




mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts1Cje.Crtx_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts1Cje.Crtx_sigs_hum=unique(mouse_human$human)

mouse_human=merge(GSE99135Ts1Cje.Crtx_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.Crtx_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts1Cje.Crtx_sigs_hum=unique(GSE99135Ts1Cje.Crtx_sig_hum$human)
list_ensembl$GSE99135Ts1Cje.Crtx=GSE99135Ts1Cje.Crtx_sigs_hum
list_adjpval$GSE99135Ts1Cje.Crtx=GSE99135Ts1Cje.Crtx_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts1Cje.Crtx=GSE99135Ts1Cje.Crtx_sig_hum$logFC






mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135dp16.Hpcmp_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135dp16.Hpcmp_sigs_hum=unique(mouse_human$human)



mouse_human=merge(GSE99135dp16.Hpcmp_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135dp16.Hpcmp_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135dp16.Hpcmp_sigs_hum=unique(GSE99135dp16.Hpcmp_sig_hum$human)
list_ensembl$GSE99135dp16.Hpcmp=GSE99135dp16.Hpcmp_sigs_hum
list_adjpval$GSE99135dp16.Hpcmp=GSE99135dp16.Hpcmp_sig_hum$adj.P.Val
list_log2FC$GSE99135dp16.Hpcmp=GSE99135dp16.Hpcmp_sig_hum$logFC


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts65Dn.Hpcmp_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts65Dn.Hpcmp_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135Ts65Dn.Hpcmp_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts65Dn.Hpcmp_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts65Dn.Hpcmp_sigs_hum=unique(GSE99135Ts65Dn.Hpcmp_sig_hum$human)
list_ensembl$GSE99135Ts65Dn.Hpcmp=GSE99135Ts65Dn.Hpcmp_sigs_hum
list_adjpval$GSE99135Ts65Dn.Hpcmp=GSE99135Ts65Dn.Hpcmp_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts65Dn.Hpcmp=GSE99135Ts65Dn.Hpcmp_sig_hum$logFC




mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE99135Ts1Cje.Hpcmp_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE99135Ts1Cje.Hpcmp_sigs_hum=unique(mouse_human$human)


mouse_human=merge(GSE99135Ts1Cje.Hpcmp_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE99135Ts1Cje.Hpcmp_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE99135Ts1Cje.Hpcmp_sigs_hum=unique(GSE99135Ts1Cje.Hpcmp_sig_hum$human)
list_ensembl$GSE99135Ts1Cje.Hpcmp=GSE99135Ts1Cje.Hpcmp_sigs_hum
list_adjpval$GSE99135Ts1Cje.Hpcmp=GSE99135Ts1Cje.Hpcmp_sig_hum$adj.P.Val
list_log2FC$GSE99135Ts1Cje.Hpcmp=GSE99135Ts1Cje.Hpcmp_sig_hum$logFC
```

```{r}
  plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$area)), pch=as.numeric(as.factor(phenodata$type)))
legend("topleft", legend=levels(phenodata$area), fill=as.numeric(as.factor(levels(phenodata$area))))
legend("bottomleft", legend=levels(as.factor(phenodata$type)), pch=as.numeric(as.factor(levels(as.factor(phenodata$type)))))


  plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$type)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$type)), pch=as.numeric(as.factor(levels(as.factor(phenodata$type)))))

rm(eset)


rm(GSE99135dp16.Crblm)
rm(GSE99135dp16.E15)
rm(GSE99135dp16.Hpcmp)
rm(GSE99135dp16.Crtx)

rm(GSE99135Ts1Cje.Crblm)
rm(GSE99135Ts1Cje.E15)
rm(GSE99135Ts1Cje.Hpcmp)
rm(GSE99135Ts1Cje.Crtx)

rm(GSE99135Ts65Dn.Crblm)
rm(GSE99135Ts65Dn.E15)
rm(GSE99135Ts65Dn.Hpcmp)
rm(GSE99135Ts65Dn.Crtx)

```

### GSE69210


Thymus transcriptomic profiles of patients with and without Down Syndrome were compared in order to identify differentially expressed transcripts.

```{r}
phenodata=data.frame(sex=c("female", "male", "female", "female", "female", "male", "female", "male", "male", "male",
"female", "female", "male", "male", "female", "female", "male", "male", "female", "male"),
ploidy=c(rep("trisomic", 10), rep("disomic", 10)))


require(limma)
#reading into the files
x <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE69210_RAW/"),source="agilent",green.only=TRUE, path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE69210_RAW/")
#background correction and normalization
y <- limma::backgroundCorrect(x,method="normexp",normexp.method="rma")
y <- normalizeBetweenArrays(y,method="quantile")



##filtering out low expressed probes
#compute the 95% percentile of the negative control probes on each array
neg95 <- apply(y$E[y$genes$ControlType==-1,],2,function(x) quantile(x,p=0.95))
cutoff <- matrix(1.1*neg95,nrow(y),ncol(y),byrow=TRUE)
#keep probes that are at least 10% brighter than the negative controls on at least four arrays (because there are four replicates)
isexpr <- rowSums(y$E > cutoff) >= 10

#Regular probes are code as ??0?? in the ControlType column
y0 <- y[y$genes$ControlType==0 & isexpr,]
yave <- avereps(y0,ID=y0$genes[,"SystematicName"])

TS <- as.factor( phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

library(statmod)
corfit <- duplicateCorrelation(yave,design,block=phenodata$sex)#it represents the correlation between measurements made on the same mouse

#array weights
aw=arrayWeights(yave, design)
#fit the linear model blocking for the mouse
fit <- lmFit(yave,design,block=phenodata$sex,correlation=corfit$consensus, weights=aw)

contrasts <-makeContrasts(trisomic-disomic,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE69210=topTable(contr.fit, coef=1, number=Inf)

GSE69210_sig=GSE69210[which(GSE69210$adj.P.Val< pthreshold),]

GSE69210_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="agilent_wholegenome_4x44k_v2", values =GSE69210_sig$ProbeName, mart = human)$ensembl_gene_id)


ids=getBM(attributes=c("ensembl_gene_id","agilent_wholegenome_4x44k_v2"),filters="agilent_wholegenome_4x44k_v2", values =GSE69210_sig$ProbeName, mart = human)
colnames(ids)=c("V1", "V2")



GSE69210_sig=merge(GSE69210_sig, ids, by.x="ProbeName", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE69210_sig=GSE69210_sig[order(GSE69210_sig$adj.P.Val, abs(GSE69210_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE69210_sig=GSE69210_sig[!duplicated(GSE69210_sig$V1),]

GSE69210_sig=GSE69210_sig[!is.na(GSE69210_sig$V1),]

list_ensembl$GSE69210=GSE69210_sigs
list_adjpval$GSE69210=GSE69210_sig$adj.P.Val
list_log2FC$GSE69210=GSE69210_sig$logFC

rm(GSE69210)

```
```{r}
  plotMDS(yave, col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$sex)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$sex)), pch=as.numeric(as.factor(levels(as.factor(phenodata$sex)))))
rm(yave)


rm(x)
rm(y)
rm(y0)
rm(cutoff)
```


## GSE84887

Here we characterize new embryonic stem cell lines with DS (DS-ESCs), and focus on the neural aspects of the disease. Our results show that neural progenitor cells (NPCs) differentiated from five independent DS-ESC lines display increased apoptosis and down-regulation of forehead developmental genes.

```{r}


phenodata=data.frame(sex=c("male", "male", "female", "female", "male","female", "male","female"),
ploidy=c(rep("trisomic", 5), rep("disomic",3)))

data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE84887_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- pData(eset)$ploidy
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(eset, design)

corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)
fit <- lmFit(exprs(eset)[nomi,], design, block=phenodata$sex, cor=corfit$consensus, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE84887=topTable(contr.fit, coef=1, number=Inf)


GSE84887_sig=GSE84887[which(GSE84887$adj.P.Val< pthreshold),]

GSE84887_sig$id=rownames(GSE84887_sig)

GSE84887_sigs=unique(unlist(ids[rownames(GSE84887_sig)]))[!is.na(unique(unlist(ids[rownames(GSE84887_sig)])))]


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE84887_sig=merge(GSE84887_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE84887_sig=GSE84887_sig[order(GSE84887_sig$adj.P.Val, abs(GSE84887_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE84887_sig=GSE84887_sig[!duplicated(GSE84887_sig$V1),]

list_ensembl$GSE84887=GSE84887_sigs
list_adjpval$GSE84887=GSE84887_sig$adj.P.Val
list_log2FC$GSE84887=GSE84887_sig$logFC

rm(GSE84887)
```

```{r}
  plotMDS(eset[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$sex)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$sex)), pch=as.numeric(as.factor(levels(as.factor(phenodata$sex)))))
rm(fit)
rm(contr.fit)
```


## GSE83449

Acute megakaryoblastic leukemia (AMKL) with or without DS.

```{r}

phenodata=data.frame(ploidy=c(rep("disomic",31),rep("trisomic",22)), sex=c("male","female","male","male","male","female","male","male","male","female","female","male","male","male","male","male","female","male","female","female","male","female","male","male","male","male","male","male","male","female","male","female","male","female","female","female","female","male","male","male","female","female","male","female","female","male","female","female","male","female","female","female","male"))
data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE83449_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), hgu133aENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- pData(eset)$ploidy
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(eset, design)

corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)
fit <- lmFit(exprs(eset)[nomi,], design, block=phenodata$sex, cor=corfit$consensus, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE83449=topTable(contr.fit, coef=1, number=Inf)


GSE83449_sig=GSE83449[which(GSE83449$adj.P.Val< pthreshold),]

GSE83449_sigs=unique(unlist(ids[rownames(GSE83449_sig)]))[!is.na(unique(unlist(ids[rownames(GSE83449_sig)])))]

GSE83449_sig$id=rownames(GSE83449_sig)

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE83449_sig=merge(GSE83449_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE83449_sig=GSE83449_sig[order(GSE83449_sig$adj.P.Val, abs(GSE83449_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE83449_sig=GSE83449_sig[!duplicated(GSE83449_sig$V1),]

list_ensembl$GSE83449=GSE83449_sigs
list_adjpval$GSE83449=GSE83449_sig$adj.P.Val
list_log2FC$GSE83449=GSE83449_sig$logFC

rm(GSE83449)
```

```{r}
plotMDS(eset[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$sex)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$sex)), pch=as.numeric(as.factor(levels(as.factor(phenodata$sex)))))
rm(fit)
rm(contr.fit)
```

### GSE59630

DS dorsolateral prefrontal cortex (DFC) and cerebellar cortex (CBC) Hippocampus(HIP), primary visual cortex (V1C), inferior temporal cortex(ITC), posterior superior temporal cortex (STC), posterior inferior pariteal cortex (IPC), primary somatosensory cortex (S1C), medial prefrontal cortex (MFC), ventral prefrontal cortex (VFC), orbital prefrontal cortex (OFC)


```{r, cache.lazy = FALSE}
pd=read.csv("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE59630_RAW/GSE59630.csv",row.names=1, header=T)

area=unlist(lapply(as.character(pd$DS),FUN=function(x)strsplit(x, split="-")[[1]][2]))

phenodata=data.frame(area=rep(area,2),
                     ploidy=c(rep("disomic", length(area)), rep("trisomic", length(area))),
                     age=rep(c(rep("fetal", 4), rep("postnatal", 40), rep("adult", 14)),2),
                     sex=rep(as.character(pd[,3]),2)
                     )

celfiles= oligoClasses::list.celfiles("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE59630_RAW/", full.names=TRUE,listGzipped=T)

data <- oligo::read.celfiles(filenames=celfiles)

eset <- oligo::rma(data, target="core")


require("huex10sttranscriptcluster.db")

ids <-mget(featureNames(eset), huex10sttranscriptclusterENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(paste(paste(phenodata$age, phenodata$area, sep="."), phenodata$ploidy, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-limma::makeContrasts(
  "cbc.adult"=adult.CBC.trisomic-adult.CBC.disomic,
  "dfc.adult"=adult.DFC.trisomic-adult.DFC.disomic,
  "hip.adult"=adult.HIP.trisomic-adult.HIP.disomic,
  "itc.adult"=adult.ITC.trisomic-adult.ITC.disomic,
  "ofc.adult"=adult.OFC.trisomic-adult.OFC.disomic,
  "v1c.adult"=adult.V1C.trisomic-adult.V1C.disomic,
  "vfc.adult"=adult.VFC.trisomic-adult.VFC.disomic,
  "dfc.fetal"=fetal.DFC.trisomic-fetal.DFC.disomic,
  "v1c.fetal"=fetal.V1C.trisomic-fetal.V1C.disomic,
  
   "cbc.postnatal"=postnatal.CBC.trisomic-postnatal.CBC.disomic,
  "dfc.postnatal"=postnatal.DFC.trisomic-postnatal.DFC.disomic,
  "hip.postnatal"=postnatal.HIP.trisomic-postnatal.HIP.disomic,
  "IPC.postnatal"=postnatal.IPC.trisomic-postnatal.IPC.disomic,
  "itc.postnatal"=postnatal.ITC.trisomic-postnatal.ITC.disomic,
  "mfc.postnatal"=postnatal.MFC.trisomic-postnatal.MFC.disomic,
  "ofc.postnatal"=postnatal.OFC.trisomic-postnatal.OFC.disomic,
  "s1c.postnatal"=postnatal.S1C.trisomic-postnatal.S1C.disomic,
  "stc.postnatal"=postnatal.STC.trisomic-postnatal.STC.disomic,
  "v1c.postnatal"=postnatal.V1C.trisomic-postnatal.V1C.disomic,
  "vfc.postnatal"=postnatal.VFC.trisomic-postnatal.VFC.disomic
  ,levels=design)

aw=arrayWeights(eset[nomi,], design)


corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)
fit <- lmFit(exprs(eset)[nomi,], design, block=phenodata$sex, cor=corfit$consensus, weights=aw)



contr.fit <-eBayes(contrasts.fit(fit, contrasts))


cbc.adult=topTable(contr.fit, coef="cbc.adult", number=Inf)
dfc.adult=topTable(contr.fit, coef="dfc.adult", number=Inf)
hip.adult=topTable(contr.fit, coef="hip.adult", number=Inf)
itc.adult=topTable(contr.fit, coef="itc.adult", number=Inf)
ofc.adult=topTable(contr.fit, coef="ofc.adult", number=Inf)
v1c.adult=topTable(contr.fit, coef="v1c.adult", number=Inf)
vfc.adult=topTable(contr.fit, coef="vfc.adult", number=Inf)
dfc.fetal=topTable(contr.fit, coef="dfc.fetal", number=Inf)
v1c.fetal=topTable(contr.fit, coef="v1c.fetal", number=Inf)
cbc.postnatal=topTable(contr.fit, coef="cbc.postnatal", number=Inf)
dfc.postnatal=topTable(contr.fit, coef="dfc.postnatal", number=Inf)
hip.postnatal=topTable(contr.fit, coef="hip.postnatal", number=Inf)
IPC.postnatal=topTable(contr.fit, coef="IPC.postnatal", number=Inf)
itc.postnatal=topTable(contr.fit, coef="itc.postnatal", number=Inf)
mfc.postnatal=topTable(contr.fit, coef="mfc.postnatal", number=Inf)
ofc.postnatal=topTable(contr.fit, coef="ofc.postnatal", number=Inf)
s1c.postnatal=topTable(contr.fit, coef="s1c.postnatal", number=Inf)
stc.postnatal=topTable(contr.fit, coef="stc.postnatal", number=Inf)
v1c.postnatal=topTable(contr.fit, coef="v1c.postnatal", number=Inf)
vfc.postnatal=topTable(contr.fit, coef="vfc.postnatal", number=Inf)


GSE59630cbc.adult_sig=cbc.adult[which(cbc.adult$adj.P.Val< pthreshold),]
GSE59630dfc.adult_sig=dfc.adult[which(dfc.adult$adj.P.Val< pthreshold),]
GSE59630hip.adult_sig=hip.adult[which(hip.adult$adj.P.Val< pthreshold),]
GSE59630ofc.adult_sig=ofc.adult[which(ofc.adult$adj.P.Val< pthreshold),]
GSE59630itc.adult_sig=itc.adult[which(itc.adult$adj.P.Val< pthreshold),]
GSE59630v1c.adult_sig=v1c.adult[which(v1c.adult$adj.P.Val< pthreshold),]
GSE59630vfc.adult_sig=vfc.adult[which(vfc.adult$adj.P.Val< pthreshold),]
GSE59630dfc.fetal_sig=dfc.fetal[which(dfc.fetal$adj.P.Val< pthreshold),]
GSE59630v1c.fetal_sig=v1c.fetal[which(v1c.fetal$adj.P.Val< pthreshold),]
GSE59630cbc.postnatal_sig=cbc.postnatal[which(cbc.postnatal$adj.P.Val< pthreshold),]
GSE59630dfc.postnatal_sig=dfc.postnatal[which(dfc.postnatal$adj.P.Val< pthreshold),]
GSE59630hip.postnatal_sig=hip.postnatal[which(hip.postnatal$adj.P.Val< pthreshold),]
GSE59630IPC.postnatal_sig=IPC.postnatal[which(IPC.postnatal$adj.P.Val< pthreshold),]
GSE59630itc.postnatal_sig=itc.postnatal[which(itc.postnatal$adj.P.Val< pthreshold),]
GSE59630mfc.postnatal_sig=mfc.postnatal[which(mfc.postnatal$adj.P.Val< pthreshold),]
GSE59630ofc.postnatal_sig=ofc.postnatal[which(ofc.postnatal$adj.P.Val< pthreshold),]
GSE59630s1c.postnatal_sig=s1c.postnatal[which(s1c.postnatal$adj.P.Val< pthreshold),]
GSE59630stc.postnatal_sig=stc.postnatal[which(stc.postnatal$adj.P.Val< pthreshold),]
GSE59630v1c.postnatal_sig=v1c.postnatal[which(v1c.postnatal$adj.P.Val< pthreshold),]
GSE59630vfc.postnatal_sig=vfc.postnatal[which(vfc.postnatal$adj.P.Val< pthreshold),]


GSE59630cbc.adult_sig$id=rownames(GSE59630cbc.adult_sig)
GSE59630dfc.adult_sig$id=rownames(GSE59630dfc.adult_sig)
GSE59630hip.adult_sig$id=rownames(GSE59630hip.adult_sig)
GSE59630ofc.adult_sig$id=rownames(GSE59630ofc.adult_sig)
GSE59630itc.adult_sig$id=rownames(GSE59630itc.adult_sig)
GSE59630v1c.adult_sig$id=rownames(GSE59630v1c.adult_sig)
GSE59630vfc.adult_sig$id=rownames(GSE59630vfc.adult_sig)
GSE59630dfc.fetal_sig$id=rownames(GSE59630dfc.fetal_sig)
GSE59630v1c.fetal_sig$id=rownames(GSE59630v1c.fetal_sig)
GSE59630cbc.postnatal_sig$id=rownames(GSE59630cbc.postnatal_sig)
GSE59630dfc.postnatal_sig$id=rownames(GSE59630dfc.postnatal_sig)
GSE59630hip.postnatal_sig$id=rownames(GSE59630hip.postnatal_sig)
GSE59630IPC.postnatal_sig$id=rownames(GSE59630IPC.postnatal_sig)
GSE59630itc.postnatal_sig$id=rownames(GSE59630itc.postnatal_sig)
GSE59630mfc.postnatal_sig$id=rownames(GSE59630mfc.postnatal_sig)
GSE59630ofc.postnatal_sig$id=rownames(GSE59630ofc.postnatal_sig)
GSE59630s1c.postnatal_sig$id=rownames(GSE59630s1c.postnatal_sig)
GSE59630stc.postnatal_sig$id=rownames(GSE59630stc.postnatal_sig)
GSE59630v1c.postnatal_sig$id=rownames(GSE59630v1c.postnatal_sig)
GSE59630vfc.postnatal_sig$id=rownames(GSE59630vfc.postnatal_sig)

GSE59630cbc.adult_sigs=unique(unlist(ids[rownames(GSE59630cbc.adult_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630cbc.adult_sig)])))]
GSE59630dfc.adult_sigs=unique(unlist(ids[rownames(GSE59630dfc.adult_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630dfc.adult_sig)])))]
GSE59630hip.adult_sigs=unique(unlist(ids[rownames(GSE59630hip.adult_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630hip.adult_sig)])))]
GSE59630ofc.adult_sigs=unique(unlist(ids[rownames(GSE59630ofc.adult_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630ofc.adult_sig)])))]
GSE59630itc.adult_sigs=unique(unlist(ids[rownames(GSE59630itc.adult_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630itc.adult_sig)])))]
GSE59630v1c.adult_sigs=unique(unlist(ids[rownames(GSE59630v1c.adult_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630v1c.adult_sig)])))]
GSE59630vfc.adult_sigs=unique(unlist(ids[rownames(GSE59630vfc.adult_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630vfc.adult_sig)])))]
GSE59630dfc.fetal_sigs=unique(unlist(ids[rownames(GSE59630dfc.fetal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630dfc.fetal_sig)])))]
GSE59630v1c.fetal_sigs=unique(unlist(ids[rownames(GSE59630v1c.fetal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630v1c.fetal_sig)])))]
GSE59630cbc.postnatal_sigs=unique(unlist(ids[rownames(GSE59630cbc.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630cbc.postnatal_sig)])))]
GSE59630dfc.postnatal_sigs=unique(unlist(ids[rownames(GSE59630dfc.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630dfc.postnatal_sig)])))]
GSE59630hip.postnatal_sigs=unique(unlist(ids[rownames(GSE59630hip.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630hip.postnatal_sig)])))]
GSE59630IPC.postnatal_sigs=unique(unlist(ids[rownames(GSE59630IPC.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630IPC.postnatal_sig)])))]
GSE59630itc.postnatal_sigs=unique(unlist(ids[rownames(GSE59630itc.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630itc.postnatal_sig)])))]
GSE59630mfc.postnatal_sigs=unique(unlist(ids[rownames(GSE59630mfc.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630mfc.postnatal_sig)])))]
GSE59630ofc.postnatal_sigs=unique(unlist(ids[rownames(GSE59630ofc.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630ofc.postnatal_sig)])))]
GSE59630s1c.postnatal_sigs=unique(unlist(ids[rownames(GSE59630s1c.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630s1c.postnatal_sig)])))]
GSE59630stc.postnatal_sigs=unique(unlist(ids[rownames(GSE59630stc.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630stc.postnatal_sig)])))]
GSE59630v1c.postnatal_sigs=unique(unlist(ids[rownames(GSE59630v1c.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630v1c.postnatal_sig)])))]
GSE59630vfc.postnatal_sigs=unique(unlist(ids[rownames(GSE59630vfc.postnatal_sig)]))[!is.na(unique(unlist(ids[rownames(GSE59630vfc.postnatal_sig)])))]

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]


list_AveExpr=list()


GSE59630dfc.fetal_sig=merge(GSE59630dfc.fetal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630dfc.fetal_sig=GSE59630dfc.fetal_sig[order(GSE59630dfc.fetal_sig$adj.P.Val, abs(GSE59630dfc.fetal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630dfc.fetal_sig=GSE59630dfc.fetal_sig[!duplicated(GSE59630dfc.fetal_sig$V1),]
list_ensembl$GSE59630dfc.fetal=GSE59630dfc.fetal_sigs
list_adjpval$GSE59630dfc.fetal=GSE59630dfc.fetal_sig$adj.P.Val
list_log2FC$GSE59630dfc.fetal=GSE59630dfc.fetal_sig$logFC
list_AveExpr$GSE59630dfc.fetal=GSE59630dfc.fetal_sig$AveExpr
  
  
  
GSE59630dfc.postnatal_sig=merge(GSE59630dfc.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630dfc.postnatal_sig=GSE59630dfc.postnatal_sig[order(GSE59630dfc.postnatal_sig$adj.P.Val, abs(GSE59630dfc.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630dfc.postnatal_sig=GSE59630dfc.postnatal_sig[!duplicated(GSE59630dfc.postnatal_sig$V1),]
list_ensembl$GSE59630dfc.postnatal=GSE59630dfc.postnatal_sigs
list_adjpval$GSE59630dfc.postnatal=GSE59630dfc.postnatal_sig$adj.P.Val
list_log2FC$GSE59630dfc.postnatal=GSE59630dfc.postnatal_sig$logFC
list_AveExpr$GSE59630dfc.postnatal=GSE59630dfc.postnatal_sig$AveExpr


GSE59630dfc.adult_sig=merge(GSE59630dfc.adult_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630dfc.adult_sig=GSE59630dfc.adult_sig[order(GSE59630dfc.adult_sig$adj.P.Val, abs(GSE59630dfc.adult_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630dfc.adult_sig=GSE59630dfc.adult_sig[!duplicated(GSE59630dfc.adult_sig$V1),]
list_ensembl$GSE59630dfc.adult=GSE59630dfc.adult_sigs
list_adjpval$GSE59630dfc.adult=GSE59630dfc.adult_sig$adj.P.Val
list_log2FC$GSE59630dfc.adult=GSE59630dfc.adult_sig$logFC
list_AveExpr$GSE59630dfc.adult=GSE59630dfc.adult_sig$AveExpr

GSE59630cbc.postnatal_sig=merge(GSE59630cbc.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630cbc.postnatal_sig=GSE59630cbc.postnatal_sig[order(GSE59630cbc.postnatal_sig$adj.P.Val, abs(GSE59630cbc.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630cbc.postnatal_sig=GSE59630cbc.postnatal_sig[!duplicated(GSE59630cbc.postnatal_sig$V1),]
list_ensembl$GSE59630cbc.postnatal=GSE59630cbc.postnatal_sigs
list_adjpval$GSE59630cbc.postnatal=GSE59630cbc.postnatal_sig$adj.P.Val
list_log2FC$GSE59630cbc.postnatal=GSE59630cbc.postnatal_sig$logFC
list_AveExpr$GSE59630cbc.postnatal=GSE59630cbc.postnatal_sig$AveExpr


GSE59630cbc.adult_sig=merge(GSE59630cbc.adult_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630cbc.adult_sig=GSE59630cbc.adult_sig[order(GSE59630cbc.adult_sig$adj.P.Val, abs(GSE59630cbc.adult_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630cbc.adult_sig=GSE59630cbc.adult_sig[!duplicated(GSE59630cbc.adult_sig$V1),]
list_ensembl$GSE59630cbc.adult=GSE59630cbc.adult_sigs
list_adjpval$GSE59630cbc.adult=GSE59630cbc.adult_sig$adj.P.Val
list_log2FC$GSE59630cbc.adult=GSE59630cbc.adult_sig$logFC
list_AveExpr$GSE59630cbc.adult=GSE59630cbc.adult_sig$AveExpr


GSE59630hip.postnatal_sig=merge(GSE59630hip.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630hip.postnatal_sig=GSE59630hip.postnatal_sig[order(GSE59630hip.postnatal_sig$adj.P.Val, abs(GSE59630hip.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630hip.postnatal_sig=GSE59630hip.postnatal_sig[!duplicated(GSE59630hip.postnatal_sig$V1),]


if(is.null(GSE59630hip.postnatal_sigs)){
list_ensembl$GSE59630hip.postnatal=character(0)}else{list_ensembl$GSE59630hip.postnatal=GSE59630hip.postnatal_sigs}


list_adjpval$GSE59630hip.postnatal=GSE59630hip.postnatal_sig$adj.P.Val
list_log2FC$GSE59630hip.postnatal=GSE59630hip.postnatal_sig$logFC
list_AveExpr$GSE59630hip.postnatal=GSE59630hip.postnatal_sig$AveExpr

GSE59630hip.adult_sig=merge(GSE59630hip.adult_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630hip.adult_sig=GSE59630hip.adult_sig[order(GSE59630hip.adult_sig$adj.P.Val, abs(GSE59630hip.adult_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630hip.adult_sig=GSE59630hip.adult_sig[!duplicated(GSE59630hip.adult_sig$V1),]
list_ensembl$GSE59630hip.adult=GSE59630hip.adult_sigs
list_adjpval$GSE59630hip.adult=GSE59630hip.adult_sig$adj.P.Val
list_log2FC$GSE59630hip.adult=GSE59630hip.adult_sig$logFC
list_AveExpr$GSE59630hip.adult=GSE59630hip.adult_sig$AveExpr


GSE59630v1c.fetal_sig=merge(GSE59630v1c.fetal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630v1c.fetal_sig=GSE59630v1c.fetal_sig[order(GSE59630v1c.fetal_sig$adj.P.Val, abs(GSE59630v1c.fetal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630v1c.fetal_sig=GSE59630v1c.fetal_sig[!duplicated(GSE59630v1c.fetal_sig$V1),]
list_ensembl$GSE59630v1c.fetal=GSE59630v1c.fetal_sigs
list_adjpval$GSE59630v1c.fetal=GSE59630v1c.fetal_sig$adj.P.Val
list_log2FC$GSE59630v1c.fetal=GSE59630v1c.fetal_sig$logFC
list_AveExpr$GSE59630v1c.fetal=GSE59630v1c.fetal_sig$AveExpr


GSE59630v1c.postnatal_sig=merge(GSE59630v1c.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630v1c.postnatal_sig=GSE59630v1c.postnatal_sig[order(GSE59630v1c.postnatal_sig$adj.P.Val, abs(GSE59630v1c.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630v1c.postnatal_sig=GSE59630v1c.postnatal_sig[!duplicated(GSE59630v1c.postnatal_sig$V1),]
list_ensembl$GSE59630v1c.postnatal=GSE59630v1c.postnatal_sigs
list_adjpval$GSE59630v1c.postnatal=GSE59630v1c.postnatal_sig$adj.P.Val
list_log2FC$GSE59630v1c.postnatal=GSE59630v1c.postnatal_sig$logFC
list_AveExpr$GSE59630v1c.postnatal=GSE59630v1c.postnatal_sig$AveExpr

GSE59630v1c.adult_sig=merge(GSE59630v1c.adult_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630v1c.adult_sig=GSE59630v1c.adult_sig[order(GSE59630v1c.adult_sig$adj.P.Val, abs(GSE59630v1c.adult_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630v1c.adult_sig=GSE59630v1c.adult_sig[!duplicated(GSE59630v1c.adult_sig$V1),]
list_ensembl$GSE59630v1c.adult=GSE59630v1c.adult_sigs
list_adjpval$GSE59630v1c.adult=GSE59630v1c.adult_sig$adj.P.Val
list_log2FC$GSE59630v1c.adult=GSE59630v1c.adult_sig$logFC
list_AveExpr$GSE59630v1c.adult=GSE59630v1c.adult_sig$AveExpr

GSE59630itc.postnatal_sig=merge(GSE59630itc.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630itc.postnatal_sig=GSE59630itc.postnatal_sig[order(GSE59630itc.postnatal_sig$adj.P.Val, abs(GSE59630itc.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630itc.postnatal_sig=GSE59630itc.postnatal_sig[!duplicated(GSE59630itc.postnatal_sig$V1),]
list_ensembl$GSE59630itc.postnatal=GSE59630itc.postnatal_sigs
list_adjpval$GSE59630itc.postnatal=GSE59630itc.postnatal_sig$adj.P.Val
list_log2FC$GSE59630itc.postnatal=GSE59630itc.postnatal_sig$logFC
list_AveExpr$GSE59630itc.postnatal=GSE59630itc.postnatal_sig$AveExpr


GSE59630itc.adult_sig=merge(GSE59630itc.adult_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630itc.adult_sig=GSE59630itc.adult_sig[order(GSE59630itc.adult_sig$adj.P.Val, abs(GSE59630itc.adult_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630itc.adult_sig=GSE59630itc.adult_sig[!duplicated(GSE59630itc.adult_sig$V1),]
list_ensembl$GSE59630itc.adult=GSE59630itc.adult_sigs
list_adjpval$GSE59630itc.adult=GSE59630itc.adult_sig$adj.P.Val
list_log2FC$GSE59630itc.adult=GSE59630itc.adult_sig$logFC
list_AveExpr$GSE59630itc.adult=GSE59630itc.adult_sig$AveExpr


GSE59630stc.postnatal_sig=merge(GSE59630stc.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630stc.postnatal_sig=GSE59630stc.postnatal_sig[order(GSE59630stc.postnatal_sig$adj.P.Val, abs(GSE59630stc.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630stc.postnatal_sig=GSE59630stc.postnatal_sig[!duplicated(GSE59630stc.postnatal_sig$V1),]


if(is.null(GSE59630stc.postnatal_sigs)){
list_ensembl$GSE59630stc.postnatal=character(0)}else{list_ensembl$GSE59630stc.postnatal=GSE59630stc.postnatal_sigs}

list_adjpval$GSE59630stc.postnatal=GSE59630stc.postnatal_sig$adj.P.Val
list_log2FC$GSE59630stc.postnatal=GSE59630stc.postnatal_sig$logFC
list_AveExpr$GSE59630stc.postnatal=GSE59630stc.postnatal_sig$AveExpr

GSE59630IPC.postnatal_sig=merge(GSE59630IPC.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630IPC.postnatal_sig=GSE59630IPC.postnatal_sig[order(GSE59630IPC.postnatal_sig$adj.P.Val, abs(GSE59630IPC.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630IPC.postnatal_sig=GSE59630IPC.postnatal_sig[!duplicated(GSE59630IPC.postnatal_sig$V1),]

if(is.null(GSE59630IPC.postnatal_sigs)){
list_ensembl$GSE59630IPC.postnatal=character(0)}else{list_ensembl$GSE59630IPC.postnatal=GSE59630IPC.postnatal_sigs}

list_adjpval$GSE59630IPC.postnatal=GSE59630IPC.postnatal_sig$adj.P.Val
list_log2FC$GSE59630IPC.postnatal=GSE59630IPC.postnatal_sig$logFC
list_AveExpr$GSE59630IPC.postnatal=GSE59630IPC.postnatal_sig$AveExpr

GSE59630s1c.postnatal_sig=merge(GSE59630s1c.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630s1c.postnatal_sig=GSE59630s1c.postnatal_sig[order(GSE59630s1c.postnatal_sig$adj.P.Val, abs(GSE59630s1c.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630s1c.postnatal_sig=GSE59630s1c.postnatal_sig[!duplicated(GSE59630s1c.postnatal_sig$V1),]


if(is.null(GSE59630s1c.postnatal_sigs)){
list_ensembl$GSE59630s1c.postnatal=character(0)}else{list_ensembl$GSE59630s1c.postnatal=GSE59630s1c.postnatal_sigs}

list_adjpval$GSE59630s1c.postnatal=GSE59630s1c.postnatal_sig$adj.P.Val
list_log2FC$GSE59630s1c.postnatal=GSE59630s1c.postnatal_sig$logFC
list_AveExpr$GSE59630s1c.postnatal=GSE59630s1c.postnatal_sig$AveExpr

GSE59630mfc.postnatal_sig=merge(GSE59630mfc.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630mfc.postnatal_sig=GSE59630mfc.postnatal_sig[order(GSE59630mfc.postnatal_sig$adj.P.Val, abs(GSE59630mfc.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630mfc.postnatal_sig=GSE59630mfc.postnatal_sig[!duplicated(GSE59630mfc.postnatal_sig$V1),]

if(is.null(GSE59630mfc.postnatal_sigs)){
list_ensembl$GSE59630mfc.postnatal=character(0)}else{list_ensembl$GSE59630mfc.postnatal=GSE59630mfc.postnatal_sigs}


list_adjpval$GSE59630mfc.postnatal=GSE59630mfc.postnatal_sig$adj.P.Val
list_log2FC$GSE59630mfc.postnatal=GSE59630mfc.postnatal_sig$logFC
list_AveExpr$GSE59630mfc.postnatal=GSE59630mfc.postnatal_sig$AveExpr

GSE59630vfc.postnatal_sig=merge(GSE59630vfc.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630vfc.postnatal_sig=GSE59630vfc.postnatal_sig[order(GSE59630vfc.postnatal_sig$adj.P.Val, abs(GSE59630vfc.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630vfc.postnatal_sig=GSE59630vfc.postnatal_sig[!duplicated(GSE59630vfc.postnatal_sig$V1),]
list_ensembl$GSE59630vfc.postnatal=GSE59630vfc.postnatal_sigs
list_adjpval$GSE59630vfc.postnatal=GSE59630vfc.postnatal_sig$adj.P.Val
list_log2FC$GSE59630vfc.postnatal=GSE59630vfc.postnatal_sig$logFC
list_AveExpr$GSE59630vfc.postnatal=GSE59630vfc.postnatal_sig$AveExpr


GSE59630vfc.adult_sig=merge(GSE59630vfc.adult_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630vfc.adult_sig=GSE59630vfc.adult_sig[order(GSE59630vfc.adult_sig$adj.P.Val, abs(GSE59630vfc.adult_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630vfc.adult_sig=GSE59630vfc.adult_sig[!duplicated(GSE59630vfc.adult_sig$V1),]


if(is.null(GSE59630vfc.adult_sigs)){
list_ensembl$GSE59630vfc.adult=character(0)}else{list_ensembl$GSE59630vfc.adult=GSE59630vfc.adult_sigs}
list_adjpval$GSE59630vfc.adult=GSE59630vfc.adult_sig$adj.P.Val
list_log2FC$GSE59630vfc.adult=GSE59630vfc.adult_sig$logFC
list_AveExpr$GSE59630vfc.adult=GSE59630vfc.adult_sig$AveExpr

GSE59630ofc.postnatal_sig=merge(GSE59630ofc.postnatal_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630ofc.postnatal_sig=GSE59630ofc.postnatal_sig[order(GSE59630ofc.postnatal_sig$adj.P.Val, abs(GSE59630ofc.postnatal_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630ofc.postnatal_sig=GSE59630ofc.postnatal_sig[!duplicated(GSE59630ofc.postnatal_sig$V1),]
list_ensembl$GSE59630ofc.postnatal=GSE59630ofc.postnatal_sigs
list_adjpval$GSE59630ofc.postnatal=GSE59630ofc.postnatal_sig$adj.P.Val
list_log2FC$GSE59630ofc.postnatal=GSE59630ofc.postnatal_sig$logFC
list_AveExpr$GSE59630ofc.postnatal=GSE59630ofc.postnatal_sig$AveExpr

GSE59630ofc.adult_sig=merge(GSE59630ofc.adult_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE59630ofc.adult_sig=GSE59630ofc.adult_sig[order(GSE59630ofc.adult_sig$adj.P.Val, abs(GSE59630ofc.adult_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE59630ofc.adult_sig=GSE59630ofc.adult_sig[!duplicated(GSE59630ofc.adult_sig$V1),]
list_ensembl$GSE59630ofc.adult=GSE59630ofc.adult_sigs
list_adjpval$GSE59630ofc.adult=GSE59630ofc.adult_sig$adj.P.Val
list_log2FC$GSE59630ofc.adult=GSE59630ofc.adult_sig$logFC
list_AveExpr$GSE59630ofc.adult=GSE59630ofc.adult_sig$AveExpr

```

```{r}
  plotMDS(eset[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), labels=paste(phenodata$age, phenodata$area, sep="."))
legend("bottomleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
rm(fit)
rm(contr.fit)
```



### GSE58463

We created two new mouse models carrying either tandem duplication (Ts2Yah) of the Stch-App interval homologous and syntenic with 21q11.2-q21.3 in humans. Total RNA isolated from gastrocnemius muscles of mice that are trisomic (Ts2Yah) for the Stch-App region on Mmu16 and diploid control mice

```{r}
require(beadarray)
require(illuminaMousev2.db)


BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE58463_non-normalized_data.txt", 
                    sep="\t", ProbeID="ID_REF", skip=0,
                    columns = list(exprs = "AVG_Signal", Detection="Detection Pval"),
                    illuminaAnnotation="Mousev2")

pData(BSData)$ploidy = as.factor(c(rep("trisomic", 6), rep("disomic", 5)))
BSData=normaliseIllumina(BSData, method="quantile", transform="log2")

BSData=BSData[which(fData(BSData)$Status=="regular"),]

expressed=apply(Detection(BSData)<0.05,1,any)
BSData=BSData[expressed,]




ids <-as.character(featureNames(BSData))
qual <- unlist(mget(ids, illuminaMousev2PROBEQUALITY, ifnotfound=NA))
ens <- mget(ids, illuminaMousev2ENSEMBL, ifnotfound=NA)
rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

BSData <-BSData[!rem,]


TS <- pData(BSData)$ploidy
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(exprs(BSData), design)

fit <- lmFit(exprs(BSData), design,  weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


Ts2Yah=topTable(contr.fit, coef=1, number=Inf)


GSE58463Ts2Yah_sig=Ts2Yah[which(Ts2Yah$adj.P.Val< pthreshold),]

GSE58463Ts2Yah_sig$id=rownames(GSE58463Ts2Yah_sig)

GSE58463Ts2Yah_sigs=unique(unlist(ens[rownames(GSE58463Ts2Yah_sig)]))[!is.na(unique(unlist(ens[rownames(GSE58463Ts2Yah_sig)])))]


ids_df=list_to_df(ens)
ids_df=ids_df[complete.cases(ids_df),]

GSE58463Ts2Yah_sig=merge(GSE58463Ts2Yah_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE58463Ts2Yah_sig=GSE58463Ts2Yah_sig[order(GSE58463Ts2Yah_sig$adj.P.Val, abs(GSE58463Ts2Yah_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE58463Ts2Yah_sig=GSE58463Ts2Yah_sig[!duplicated(GSE58463Ts2Yah_sig$V1),]



mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE58463Ts2Yah_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

GSE58463Ts2Yah_sigs_hum=unique(mouse_human$human)

mouse_human=merge(GSE58463Ts2Yah_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE58463Ts2Yah_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE58463Ts2Yah_sigs_hum=unique(GSE58463Ts2Yah_sig_hum$human)

list_ensembl$GSE58463Ts2Yah=GSE58463Ts2Yah_sigs_hum
list_adjpval$GSE58463Ts2Yah=GSE58463Ts2Yah_sig_hum$adj.P.Val
list_log2FC$GSE58463Ts2Yah=GSE58463Ts2Yah_sig_hum$logFC

```


```{r}
  plotMDS(exprs(BSData), col=as.numeric(as.factor(pData(BSData)$ploidy)))
legend("bottomleft", legend=levels(pData(BSData)$ploidy), fill=as.numeric(as.factor(levels(pData(BSData)$ploidy))))

```



### GSE65055

Five trisomy 21 and five controls from chorionic villi (fibroblasts). Warning: all disomic are male, and all trisomic are female!

```{r}
phenodata=data.frame(ploidy=c("trisomic", "disomic", "trisomic",rep("disomic", 3),rep("trisomic", 2), "disomic", "trisomic"),
sex=c("female", "male", "female", "male", "male", "male", "female", "female", "male", "female"))




celfiles=list.celfiles("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE65055_RAW/", full.names=TRUE,listGzipped=T)

data <- oligo::read.celfiles(filenames=celfiles)

eset <- oligo::rma(data, target="core")


require("hugene20sttranscriptcluster.db")

ids <-mget(featureNames(eset),hugene20sttranscriptclusterENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor( phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(exprs(eset)[nomi,], design)

fit <- lmFit(exprs(eset)[nomi,], design,  weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


topT=topTable(contr.fit, coef=1, number=dim(eset)[1])

GSE65055_sig=topT[which(topT$adj.P.Val< pthreshold),]

GSE65055_sigs=unique(unlist(ids[rownames(GSE65055_sig)]))

GSE65055_sig$id=rownames(GSE65055_sig)

GSE65055_sig=merge(GSE65055_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE65055_sig=GSE65055_sig[order(GSE65055_sig$adj.P.Val, abs(GSE65055_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE65055_sig=GSE65055_sig[!duplicated(GSE65055_sig$V1),]

if(is.null(GSE65055_sigs)){
list_ensembl$GSE65055=character(0)}else{list_ensembl$GSE65055=GSE65055_sigs}

list_adjpval$GSE65055=GSE65055_sig$adj.P.Val
list_log2FC$GSE65055=GSE65055_sig$logFC



```


```{r}
plotMDS(eset[nomi,], col=as.numeric(as.factor(phenodata$ploidy)))
legend("bottomleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
rm(fit)
rm(contr.fit)
```



### GSE70102

We profiled gene expression at the maternal-fetal interface during the second trimester of pregnancy (13-22 wks) in trisomy 21 (T21; Down syndrome, n = 8), and in euploid pregnancies (n = 4). FISH confirmed the ploidy of the samples.


```{r}
require(affy)


phenodata=data.frame(ploidy=c(rep("trisomic", 8), rep("disomic",4)))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE70102_RAW/", phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(exprs(eset)[nomi,], design)

fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE70102=topTable(contr.fit, coef=1, number=Inf)


GSE70102_sig=GSE70102[which(GSE70102$adj.P.Val< pthreshold),]

GSE70102_sigs=unique(unlist(ids[rownames(GSE70102_sig)]))

GSE70102_sig$id=rownames(GSE70102_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE70102_sig=merge(GSE70102_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE70102_sig=GSE70102_sig[order(GSE70102_sig$adj.P.Val, abs(GSE70102_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE70102_sig=GSE70102_sig[!duplicated(GSE70102_sig$V1),]

list_ensembl$GSE70102=GSE70102_sigs
list_adjpval$GSE70102=GSE70102_sig$adj.P.Val
list_log2FC$GSE70102=GSE70102_sig$logFC

```

```{r}
plotMDS(eset[nomi,], col=as.numeric(as.factor(phenodata$ploidy)))
legend("bottomleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))

```


### GSE68074

Ts65Dn mouse hippocampi

```{r}
phenodata=data.frame(ploidy=c("trisomic", "disomic", "disomic", "disomic", "trisomic", "trisomic", "disomic"))

setwd("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE68074_RAW/")
x <- read.maimages(list.files(),source="agilent",green.only=TRUE, path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE68074_RAW/")
#background correction and normalization
y <- limma::backgroundCorrect(x,method="normexp",normexp.method="rma")
y <- normalizeBetweenArrays(y,method="quantile")



##filtering out low expressed probes
#compute the 95% percentile of the negative control probes on each array
neg95 <- apply(y$E[y$genes$ControlType==-1,],2,function(x) quantile(x,p=0.95))
cutoff <- matrix(1.1*neg95,nrow(y),ncol(y),byrow=TRUE)
#keep probes that are at least 10% brighter than the negative controls on at least four arrays (because there are four replicates)
isexpr <- rowSums(y$E > cutoff) >= 4

#Regular probes are code as ??0?? in the ControlType column
y0 <- y[y$genes$ControlType==0 & isexpr,]
yave <- avereps(y0,ID=y0$genes[,"SystematicName"])


anno_agilent_mouse=getBM(attributes=c("ensembl_gene_id","agilent_sureprint_g3_ge_8x60k"),filters="agilent_sureprint_g3_ge_8x60k", values =yave$genes$ProbeName, mart = mouse)

isanno=which(yave$genes$ProbeName %in% anno_agilent_mouse$agilent_sureprint_g3_ge_8x60k)
yave=yave[isanno,]
TS <- as.factor(phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


#array weights
aw=arrayWeights(yave, design)
#fit the linear model blocking for the mouse
fit <- lmFit(yave,design,weights=aw)

contrasts <-makeContrasts(trisomic-disomic,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE68074=topTable(contr.fit, coef=1, number=Inf)

GSE68074_sig=GSE68074[which(GSE68074$adj.P.Val< pthreshold),]

GSE68074_sigs=anno_agilent_mouse[which(anno_agilent_mouse$agilent_sureprint_g3_ge_8x60k %in% rownames(GSE68074_sig)),1]


# mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE68074_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
# 
# colnames(mouse_human)=c("mouse", "human")
# 
# GSE68074_sigs_hum=unique(mouse_human$human)
# length(GSE68074_sigs)

list_ensembl$GSE68074=character(0)
list_adjpval$GSE68074=numeric(0)
list_log2FC$GSE68074=numeric(0)
```

```{r}
plotMDS(yave, col=as.numeric(as.factor(phenodata$ploidy)))
legend("bottomleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))

```


### GSE36787


We analyzed the transcriptome of (i) T21 iPSC-derived heamtopoietic progenitors expressing wtGATA1 (6 replicates) as well as of (ii) euploid iPSC-derived progenitors expressing wtGATA1 (2 replicates). CD43+/41+/235+ progenitors.

```{r}
phenodata=data.frame(ploidy=c(rep("disomic", 2), rep("trisomic", 6)))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE36787_RAW/")

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("hugene10sttranscriptcluster.db")
ids <-mget(featureNames(eset), hugene10sttranscriptclusterENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]
# ids=as.character(unlist(ids))


TS <- as.factor( phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)
aw=arrayWeights(exprs(eset)[as.character(nomi),], design)


fit <- lmFit(exprs(eset)[as.character(nomi),], design,weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE36787=topTable(contr.fit, coef=1, number=Inf)

GSE36787_sig=GSE36787[which(GSE36787$adj.P.Val< pthreshold),]

GSE36787_sigs=unique(unlist(ids[rownames(GSE36787_sig)]))

GSE36787_sig$id=rownames(GSE36787_sig)

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE36787_sig=merge(GSE36787_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE36787_sig=GSE36787_sig[order(GSE36787_sig$adj.P.Val, abs(GSE36787_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE36787_sig=GSE36787_sig[!duplicated(GSE36787_sig$V1),]

list_ensembl$GSE36787=GSE36787_sigs
list_adjpval$GSE36787=GSE36787_sig$adj.P.Val
list_log2FC$GSE36787=GSE36787_sig$logFC



```



```{r}
plotMDS(exprs(eset)[as.character(nomi),], col=as.numeric(as.factor(phenodata$ploidy)))
legend("bottomleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))

```



### GSE62538

We analyzed the forebrain whole transcriptome from embryonic day 15.5 (E15.5) Ts1Cje (n=5) and wild-type (n=5) 

```{r}
phenodata=data.frame(ploidy=c(rep("disomic", 5), rep("trisomic", 5)),
sex=c("male", "female", "female", "female", "male", "male", "female", "male", "male", "female"))

data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE62538_RAW/",phenoData=phenodata, verbose=T)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("mogene10sttranscriptcluster.db")
ids <-mget(featureNames(eset), mogene10sttranscriptclusterENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]

TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(exprs(eset)[as.character(nomi),], design)

fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE62538=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])

GSE62538_sig=GSE62538[which(GSE62538$adj.P.Val< pthreshold),]

GSE62538_sigs=unique(unlist(ids[rownames(GSE62538_sig)]))

GSE62538_sig$id=rownames(GSE62538_sig)

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE62538_sig=merge(GSE62538_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE62538_sig=GSE62538_sig[order(GSE62538_sig$adj.P.Val, abs(GSE62538_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE62538_sig=GSE62538_sig[!duplicated(GSE62538_sig$V1),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE62538_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE62538_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE62538_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE62538_sigs_hum=unique(GSE62538_sig_hum$human)


list_ensembl$GSE62538=GSE62538_sigs_hum
list_adjpval$GSE62538=GSE62538_sig_hum$adj.P.Val
list_log2FC$GSE62538=GSE62538_sig_hum$logFC
```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$sex)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$sex)), pch=as.numeric(as.factor(levels(as.factor(phenodata$sex)))))
```

### GSE49050

The Ts1Cje mouse model of Down syndrome (DS) has partial triplication of mouse chromosome 16 (MMU16), which is partially homologous to human chromosome 21. The mouse model develops various neuropathological features identified in DS individuals. We analysed the effect of partial triplication of the MMU16 segment on global gene expression in the cerebral cortex, cerebellum and hippocampus of Ts1Cje mice at 4 time-points; postnatal day (P)1, P15, P30 and P84. Just female.


```{r}
df=read.csv("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE49050_RAW/Workbook1.csv", header=F, row.names=1)

phenodata=data.frame(
ploidy=unlist(lapply(df$V2, FUN=function(x) (strsplit(as.character(x), split=" ")[[1]][1]))),
timepoint=unlist(lapply(df$V2, FUN=function(x) (strsplit(as.character(x), split=" ")[[1]][2]))),
area=unlist(lapply(df$V2, FUN=function(x) (strsplit(as.character(x), split=" ")[[1]][3])))
)
data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE49050_RAW/")

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("mouse4302.db")
ids <-mget(featureNames(eset), mouse4302ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]

TS <- as.factor( paste(phenodata$ploidy, phenodata$area, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts("HPC"=Ts1Cje.hippocampus-Wildtype.hippocampus,
                          "CTX"=Ts1Cje.cerebral-Wildtype.cerebral,
                          "CER"=Ts1Cje.cerebellum-Wildtype.cerebellum,
                          levels=design)

aw=arrayWeights(exprs(eset)[as.character(nomi),], design)

corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$timepoint)

fit <- lmFit(exprs(eset)[nomi,], design, weights = aw, block=phenodata$timepoint, cor=corfit$consensus)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE49050.hpc=topTable(contr.fit, coef="HPC", number=Inf)
GSE49050.ctx=topTable(contr.fit, coef="CTX", number=Inf)
GSE49050.cer=topTable(contr.fit, coef="CER", number=Inf)


GSE49050.hpc_sig=GSE49050.hpc[which(GSE49050.hpc$adj.P.Val< pthreshold),]
GSE49050.ctx_sig=GSE49050.ctx[which(GSE49050.ctx$adj.P.Val< pthreshold),]
GSE49050.cer_sig=GSE49050.cer[which(GSE49050.cer$adj.P.Val< pthreshold),]


GSE49050.hpc_sig.Ts1Cje=unique(unlist(ids[rownames(GSE49050.hpc_sig)]))
GSE49050.ctx_sig.Ts1Cje=unique(unlist(ids[rownames(GSE49050.ctx_sig)]))
GSE49050.cer_sig.Ts1Cje=unique(unlist(ids[rownames(GSE49050.cer_sig)]))

GSE49050.hpc_sig$id=rownames(GSE49050.hpc_sig)
GSE49050.cer_sig$id=rownames(GSE49050.cer_sig)
GSE49050.ctx_sig$id=rownames(GSE49050.ctx_sig)


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE49050.hpc_sig=merge(GSE49050.hpc_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE49050.hpc_sig=GSE49050.hpc_sig[order(GSE49050.hpc_sig$adj.P.Val, abs(GSE49050.hpc_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE49050.hpc_sig=GSE49050.hpc_sig[!duplicated(GSE49050.hpc_sig$V1),]

GSE49050.cer_sig=merge(GSE49050.cer_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE49050.cer_sig=GSE49050.cer_sig[order(GSE49050.cer_sig$adj.P.Val, abs(GSE49050.cer_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE49050.cer_sig=GSE49050.cer_sig[!duplicated(GSE49050.cer_sig$V1),]

GSE49050.ctx_sig=merge(GSE49050.ctx_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE49050.ctx_sig=GSE49050.ctx_sig[order(GSE49050.ctx_sig$adj.P.Val, abs(GSE49050.ctx_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE49050.ctx_sig=GSE49050.ctx_sig[!duplicated(GSE49050.ctx_sig$V1),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE49050.cer_sig.Ts1Cje, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=merge(GSE49050.cer_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE49050.cer_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE49050.cer_sigs_hum=unique(GSE49050.cer_sig_hum$human)


list_ensembl$GSE49050.cer=GSE49050.cer_sigs_hum
list_adjpval$GSE49050.cer=GSE49050.cer_sig_hum$adj.P.Val
list_log2FC$GSE49050.cer=GSE49050.cer_sig_hum$logFC

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE49050.ctx_sig.Ts1Cje, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=merge(GSE49050.ctx_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE49050.ctx_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE49050.ctx_sigs_hum=unique(GSE49050.ctx_sig_hum$human)


list_ensembl$GSE49050.ctx=GSE49050.ctx_sigs_hum
list_adjpval$GSE49050.ctx=GSE49050.ctx_sig_hum$adj.P.Val
list_log2FC$GSE49050.ctx=GSE49050.ctx_sig_hum$logFC

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE49050.hpc_sig.Ts1Cje, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=merge(GSE49050.hpc_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE49050.hpc_sig_hum=mouse_human[!duplicated(mouse_human$human),]

GSE49050.hpc_sigs_hum=unique(GSE49050.hpc_sig_hum$human)


list_ensembl$GSE49050.hpc=GSE49050.hpc_sigs_hum
list_adjpval$GSE49050.hpc=GSE49050.hpc_sig_hum$adj.P.Val
list_log2FC$GSE49050.hpc=GSE49050.hpc_sig_hum$logFC

```


```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$timepoint)), pch=as.numeric(as.factor(phenodata$area)))
legend("topleft", legend=levels(phenodata$timepoint), fill=as.numeric(as.factor(levels(phenodata$timepoint))))
legend("left", legend=levels(as.factor(phenodata$area)), pch=as.numeric(as.factor(levels(as.factor(phenodata$area)))))

plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$area)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("left", legend=levels(as.factor(phenodata$area)), pch=as.numeric(as.factor(levels(as.factor(phenodata$area)))))
```


### GSE49635

We analyzed the cerebral cortex and hippocampus whole transcriptome from 8-10 weeks old Ts1Cje (n=6) and wild-type (n=5)

```{r}
phenodata=data.frame(ploidy=rep(c(rep("disomic", 5), rep("trisomic", 6)),2),
area=c(rep("cortex", 11), rep("hippocampus", 11)))

data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE49635_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("mogene10sttranscriptcluster.db")
ids <-mget(featureNames(eset), mogene10sttranscriptclusterENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]

TS <- as.factor( paste(pData(eset)$area,pData(eset)$ploidy,sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts("ctx"=cortex.trisomic-cortex.disomic, "hpc"=hippocampus.trisomic - hippocampus.disomic,levels=design)

aw=arrayWeights(exprs(eset)[as.character(nomi),], design)

fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


cortex=topTable(contr.fit, coef="ctx", number=Inf)
hippocampus=topTable(contr.fit, coef="hpc", number=Inf)

cortex_sig=cortex[which(cortex$adj.P.Val< pthreshold),]
GSE49635cortex_sigs=unique(unlist(ids[rownames(cortex_sig)]))


GSE49635cortex_sig=cortex_sig
GSE49635cortex_sig$id=rownames(GSE49635cortex_sig)


hippocampus_sig=hippocampus[which(hippocampus$adj.P.Val< pthreshold),]
GSE49635hippocampus_sigs=unique(unlist(ids[rownames(hippocampus_sig)]))

GSE49635hippocampus_sig=hippocampus_sig
GSE49635hippocampus_sig$id=rownames(GSE49635hippocampus_sig)


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE49635cortex_sig=merge(GSE49635cortex_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE49635cortex_sig=GSE49635cortex_sig[order(GSE49635cortex_sig$adj.P.Val, abs(GSE49635cortex_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE49635cortex_sig=GSE49635cortex_sig[!duplicated(GSE49635cortex_sig$V1),]

GSE49635hippocampus_sig=merge(GSE49635hippocampus_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE49635hippocampus_sig=GSE49635hippocampus_sig[order(GSE49635hippocampus_sig$adj.P.Val, abs(GSE49635hippocampus_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE49635hippocampus_sig=GSE49635hippocampus_sig[!duplicated(GSE49635hippocampus_sig$V1),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values =GSE49635cortex_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=merge(GSE49635cortex_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE49635cortex_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE49635cortex_sigs_hum=unique(GSE49635cortex_sig_hum$human)


list_ensembl$GSE49635cortex=GSE49635cortex_sigs_hum
list_adjpval$GSE49635cortex=GSE49635cortex_sig_hum$adj.P.Val
list_log2FC$GSE49635cortex=GSE49635cortex_sig_hum$logFC

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values =GSE49635hippocampus_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE49635hippocampus_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE49635hippocampus_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE49635hippocampus_sigs_hum=unique(GSE49635hippocampus_sig_hum$human)


list_ensembl$GSE49635hippocampus=GSE49635hippocampus_sigs_hum
list_adjpval$GSE49635hippocampus=GSE49635hippocampus_sig_hum$adj.P.Val
list_log2FC$GSE49635hippocampus=GSE49635hippocampus_sig_hum$logFC
```


```{r}

plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$area)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("left", legend=levels(as.factor(phenodata$area)), pch=as.numeric(as.factor(levels(as.factor(phenodata$area)))))
```

### GSE48611

Three independent RNA samples were collected from Down syndrome (DS) and control iPSCs between passages 24 and 48.
Three independent RNA samples were collected from 30 day old neurons differentiated from Down syndrome (DS) and control iPSCs.

```{r}
phenodata=data.frame(ploidy=rep(c(rep("trisomic", 6), rep("disomic", 3)),2),
                     cellline=rep(c(rep("ds1",3), rep("ds4",3), rep("ds2u",3)),2),
                     type=c(rep("iPSC", 9), rep("neurons", 9)))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE48611_RAW/",phenoData=phenodata)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]
# ids=as.character(unlist(ids))

TS <- as.factor(paste(pData(eset)$ploidy, pData(eset)$type, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts("ipsc"=trisomic.iPSC-disomic.iPSC, "neurons"=trisomic.neurons- disomic.neurons,levels=design)

aw=arrayWeights(exprs(eset)[as.character(nomi),], design)

corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$cellline)

fit <- lmFit(exprs(eset)[nomi,], design, weights = aw, block=phenodata$cellline, cor=corfit$consensus)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


ipsc=topTable(contr.fit, coef="ipsc", number=Inf)
neurons=topTable(contr.fit, coef="neurons", number=Inf)

ipsc_sig=ipsc[which(ipsc$adj.P.Val< pthreshold),]
neurons_sig=neurons[which(neurons$adj.P.Val< pthreshold),]

ipsc_sig$id=rownames(ipsc_sig)
neurons_sig$id=rownames(neurons_sig)
GSE48611ipsc_sig=ipsc_sig
GSE48611neurons_sig=neurons_sig


GSE48611ipsc_sigs=unique(unlist(ids[rownames(ipsc_sig)]))
GSE48611neurons_sigs=unique(unlist(ids[rownames(neurons_sig)]))


ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE48611ipsc_sig=merge(GSE48611ipsc_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE48611ipsc_sig=GSE48611ipsc_sig[order(GSE48611ipsc_sig$adj.P.Val, abs(GSE48611ipsc_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE48611ipsc_sig=GSE48611ipsc_sig[!duplicated(GSE48611ipsc_sig$V1),]
list_ensembl$GSE48611ipsc=GSE48611ipsc_sigs
list_adjpval$GSE48611ipsc=GSE48611ipsc_sig$adj.P.Val
list_log2FC$GSE48611ipsc=GSE48611ipsc_sig$logFC


GSE48611neurons_sig=merge(GSE48611neurons_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE48611neurons_sig=GSE48611neurons_sig[order(GSE48611neurons_sig$adj.P.Val, abs(GSE48611neurons_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE48611neurons_sig=GSE48611neurons_sig[!duplicated(GSE48611neurons_sig$V1),]
list_ensembl$GSE48611neurons=GSE48611neurons_sigs
list_adjpval$GSE48611neurons=GSE48611neurons_sig$adj.P.Val
list_log2FC$GSE48611neurons=GSE48611neurons_sig$logFC
```


```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$type)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$type)), pch=as.numeric(as.factor(levels(as.factor(phenodata$type)))))

```




### GSE48051

Transcriptomic profiles of 10 cultivated amniocyte samples with Ts21 and 9 with normal euploid constitution


```{r}
phenodata=data.frame(
ploidy=c(rep("disomic", 9), rep("trisomic", 10)))


require(limma)
#reading into the files
x <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE48051_RAW/"),source="genepix",green.only=TRUE, path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE48051_RAW/")
#background correction and normalization
y <- limma::backgroundCorrect(x,method="normexp",normexp.method="rma")
y <- normalizeBetweenArrays(y,method="quantile")


TS <- as.factor( phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


#array weights
aw=arrayWeights(y, design)
#fit the linear model blocking for the mouse
fit <- lmFit(y,design, weights=aw)

contrasts <-makeContrasts(trisomic-disomic,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE48051=topTable(contr.fit, coef=1, number=Inf)

GSE48051_sig=GSE48051[which(GSE48051$adj.P.Val< pthreshold),]

GSE48051_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="agilent_wholegenome_4x44k_v2", values =GSE48051_sig$ID, mart = human)$ensembl_gene_id)

ids_df=getBM(attributes=c("ensembl_gene_id","agilent_wholegenome_4x44k_v2"),filters="agilent_wholegenome_4x44k_v2", values =GSE48051_sig$ID, mart = human)

ids_df=ids_df[complete.cases(ids_df),]

GSE48051_sig=merge(GSE48051_sig, ids_df, by.x="ID", by.y="agilent_wholegenome_4x44k_v2", all.x=TRUE, all.y=FALSE)
GSE48051_sig=GSE48051_sig[order(GSE48051_sig$adj.P.Val, abs(GSE48051_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE48051_sig=GSE48051_sig[!duplicated(GSE48051_sig$ensembl_gene_id),]

list_ensembl$GSE48051=GSE48051_sigs
list_adjpval$GSE48051=GSE48051_sig$adj.P.Val
list_log2FC$GSE48051=GSE48051_sig$logFC

```

```{r}
plotMDS((eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
```


### GSE42956

Control (WT) and Down Syndrome iPSC lines were generated via episomal reprogramming of human donor fibroblasts. Two iPSC clones conforming to iPS criteria (determined by immunocytochemistry detection of pluripotency markers) were developed from each fibroblast donor. Two control (WT) and DS lines each were further characterized and underwent neural differentiation. Multiple biological replicates of donor fibroblast and iPSC from both control (WT) and DS lines, including 3 euploid DS samples and 3 MEL1 hESC controls (total 54 samples) were hybridized to Illumina HT-12 v4 microarray for gene expression analysis.

```{r}
phenodata=data.frame(ploidy=c(rep("disomic",12), rep(c(rep("trisomic", 3), rep("disomic",3)),3), rep("disomic", 3), rep("trisomic", 3),rep("disomic", 6), rep("trisomic",6)),

type=c(rep("ipcs", 6), rep("fibroblasts", 12), rep("ipcs", 9), rep("fibroblasts", 9), rep("ipcs", 12)),
sex=c(rep("male",3), rep("male", 3), rep("female", 3), rep("male",21), rep("female",3), rep("male",6), rep("female",3), rep("male",6)))


phenodata$donorline=c(rep("CRL-2429",3),rep("CRL-1502",3),rep("CRL-1502",3),rep("CRL-2429",3),rep("CCL54",3),rep("CRL-2429",3),rep("CCL54",3),rep("CRL-2429",3),
rep("CCL54",3),rep("CRL-2429",3),rep("CRL-1502",3),rep("CCL54",3),rep("CRL-2429",3),rep("CRL-1502",3),rep("CCL54",3),rep("CCL54",3))


phenodata$passage=c(rep("7",3),rep("11",3),rep("0",3),rep("9",3),rep("7",3),rep("0",3),rep("12",3),rep("7",3),
            rep("11",3),rep("8",3),rep("9",3),rep("8",3),rep("CRL-2429",3),rep("55",3),rep("CCL54",3),rep("55",3))

phenodata$clone=c(rep("c16",3),rep("c29",3),rep("0",3),rep("0",3),rep("0",3),rep("0",3),rep("c11",3),rep("c11",3),
            rep("c18",3),rep("0",3),rep("0",3),rep("0",3),rep("c11",3),rep("c32",3),rep("c11",3),rep("c18",3))

require(illuminaHumanv4.db)


BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE42956_non-normalized_data.txt", 
                    sep="\t", ProbeID="PROBE_ID", skip=0,
                    columns = list(exprs = "AVG_Signal", Detection="Detection Pval"),
                    illuminaAnnotation="Humanv4")

BSData=normaliseIllumina(BSData, method="quantile", transform="log2")
pData(BSData) = phenodata

BSData=BSData[which(fData(BSData)$Status=="regular"),]

expressed=apply(Detection(BSData)<0.05,1,any)
BSData=BSData[expressed,]



ids <-as.character(featureNames(BSData))
qual <- unlist(mget(ids, illuminaHumanv4PROBEQUALITY, ifnotfound=NA))
ens <- mget(ids, illuminaHumanv4ENSEMBL, ifnotfound=NA)
rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

BSData <-BSData[!rem,]


TS <- as.factor(paste(pData(BSData)$type, pData(BSData)$ploidy, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

corfit<-duplicateCorrelation(exprs(BSData),design, block=pData(BSData)$sex)

contrasts <-makeContrasts("ipcs"=ipcs.trisomic - ipcs.disomic, "fibroblasts"=fibroblasts.trisomic - fibroblasts.disomic,levels=design)

aw=arrayWeights(exprs(BSData), design)

fit <- lmFit(exprs(BSData), design,  weights=aw, block=pData(BSData)$sex, cor=corfit$consensus)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


ipcs=topTable(contr.fit, coef="ipcs", number=Inf)
fibroblasts=topTable(contr.fit, coef="fibroblasts", number=Inf)


ipcs_sig=ipcs[which(ipcs$adj.P.Val< pthreshold),]
fibroblasts_sig=fibroblasts[which(fibroblasts$adj.P.Val< pthreshold),]


ipcs_sig$id=rownames(ipcs_sig)
fibroblasts_sig$id=rownames(fibroblasts_sig)
GSE42956ipcs_sig=ipcs_sig
GSE42956fibroblasts_sig=fibroblasts_sig

GSE42956ipcs_sigs=unique(unlist(ens[rownames(ipcs_sig)]))
GSE42956fibroblasts_sigs=unique(unlist(ens[rownames(fibroblasts_sig)]))


ids_df=list_to_df(ens)
ids_df=ids_df[complete.cases(ids_df),]

GSE42956ipcs_sig=merge(GSE42956ipcs_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE42956ipcs_sig=GSE42956ipcs_sig[order(GSE42956ipcs_sig$adj.P.Val, abs(GSE42956ipcs_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE42956ipcs_sig=GSE42956ipcs_sig[!duplicated(GSE42956ipcs_sig$V1),]
list_ensembl$GSE42956ipcs=GSE42956ipcs_sigs
list_adjpval$GSE42956ipcs=GSE42956ipcs_sig$adj.P.Val
list_log2FC$GSE42956ipcs=GSE42956ipcs_sig$logFC

GSE42956fibroblasts_sig=merge(GSE42956fibroblasts_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE42956fibroblasts_sig=GSE42956fibroblasts_sig[order(GSE42956fibroblasts_sig$adj.P.Val, abs(GSE42956fibroblasts_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE42956fibroblasts_sig=GSE42956fibroblasts_sig[!duplicated(GSE42956fibroblasts_sig$V1),]
list_ensembl$GSE42956fibroblasts=GSE42956fibroblasts_sigs
list_adjpval$GSE42956fibroblasts=GSE42956fibroblasts_sig$adj.P.Val
list_log2FC$GSE42956fibroblasts=GSE42956fibroblasts_sig$logFC

```


```{r}
plotMDS(exprs(BSData), col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$type)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$type)), pch=as.numeric(as.factor(levels(as.factor(phenodata$type)))))

```



### GSE42772

We performed a broad-spectrum microarray analysis using mRNA from 3 NL and 5 DS astrocyte cultures.
To explore the relation between gene expression and potential oxidative stress in DS samples, we also evaluated the transcription profile in NL astrocytes subjected to mild oxidative stress (50µM H2O2 for 24 hr).

```{r}
#NADA DE SIGNIFICATIVO
phenodata=data.frame(ploidy=rep(c(rep("trisomic",5), rep("disomic", 3)),2),

treatment=c(rep("H2O2", 8),rep("ctrl", 8)))

require(illuminaHumanv2.db)


BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE42772_non-normalized.txt",
                    sep="\t", ProbeID="ID_REF", skip=4,
                    columns = list(exprs = "AVG_Signal", Detection="Detection Pval"),
                    illuminaAnnotation="Humanv2")

BSData=normaliseIllumina(BSData, method="quantile", transform="log2")
pData(BSData) = phenodata

BSData=BSData[which(fData(BSData)$Status=="regular"),]

 # expressed=apply(Detection(BSData)<0.05,1,any)
 # BSData=BSData[expressed,]

#
 ids <-as.character(featureNames(BSData))
 qual <- unlist(mget(ids, illuminaHumanv2PROBEQUALITY, ifnotfound=NA))
 ens <- mget(ids, illuminaHumanv2ENSEMBL, ifnotfound=NA)
 rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

 # BSData <-BSData[!rem,]


TS <- as.factor(paste(phenodata$treatment, phenodata$ploidy, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts("ctrl"=ctrl.trisomic - ctrl.disomic, "h202"=H2O2.trisomic - H2O2.disomic,levels=design)

# aw=arrayWeights(exprs(BSData), design)

fit <- lmFit(exprs(BSData), design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


ctrl=topTable(contr.fit, coef="ctrl", number=Inf)
h2o2=topTable(contr.fit, coef="h202", number=Inf)


ctrl_sig=ctrl[which(ctrl$adj.P.Val< 0.1),]
h2o2_sig=h2o2[which(h2o2$adj.P.Val< 0.1),]


list_ensembl$GSE42772=character(0)
list_adjpval$GSE42772=numeric(0)
list_log2FC$GSE42772=numeric(0)

list_ensembl$GSE42772h2o2=character(0)
list_adjpval$GSE42772h2o2=numeric(0)
list_log2FC$GSE42772h2o2=numeric(0)
```

```{r}
plotMDS(exprs(BSData), col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$treatment)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$treatment)), pch=as.numeric(as.factor(levels(as.factor(phenodata$treatment)))))

```

	
### GSE38931

 Here we have generated induced pluripotent stem cells (iPSCs) from DS fibroblasts + 2 euploid ones.

```{r}

phenodata=data.frame(ploidy=rep(c("trisomic", rep("disomic", 2)),2))
# 
# treatment=c(rep("H20", 8),rep("ctrl", 8)))
# 
# require(illuminaHumanv2.db)
# 
# 
BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE38931_non-normalized.txt",
                    sep="\t", ProbeID="PROBE_ID", skip=0,
                    columns = list(exprs = "AVG_Signal", Detection="Detection Pval"),
                    illuminaAnnotation="Humanv4")
# 
BSData=normaliseIllumina(BSData, method="quantile", transform="log2")
pData(BSData) = phenodata

BSData=BSData[which(fData(BSData)$Status=="regular"),]

expressed=apply(Detection(BSData)<0.05,1,any)
BSData=BSData[expressed,]

#
ids <-as.character(featureNames(BSData))
qual <- unlist(mget(ids, illuminaHumanv4PROBEQUALITY, ifnotfound=NA))
ens <- mget(ids, illuminaHumanv4ENSEMBL, ifnotfound=NA)
rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

BSData <-BSData[!rem,]

# 
TS <- as.factor( pData(BSData)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

# 
contrasts <-makeContrasts(trisomic-disomic,levels=design)
# 
aw=arrayWeights(exprs(BSData), design)
# 
fit <- lmFit(exprs(BSData), design,  weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE38931=topTable(contr.fit, 1, number=Inf)


GSE38931_sig=GSE38931[which(GSE38931$adj.P.Val< pthreshold),]
# h2o2_sig=h2o2[which(h2o2$adj.P.Val< 0.1),]
# 
GSE38931_sigs=unique(unlist(ens[rownames(GSE38931_sig)]))
# GSE42956fibroblasts_sig=unique(unlist(ens[rownames(fibroblasts_sig)]))

GSE38931_sig$id=rownames(GSE38931_sig)
ids_df=list_to_df(ens)
ids_df=ids_df[complete.cases(ids_df),]

GSE38931_sig=merge(GSE38931_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE38931_sig=GSE38931_sig[order(GSE38931_sig$adj.P.Val, abs(GSE38931_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE38931_sig=GSE38931_sig[!duplicated(GSE38931_sig$V1),]


if(is.null(GSE38931_sigs)){
list_ensembl$GSE38931=character(0)}else{list_ensembl$GSE38931=GSE38931_sigs}

list_adjpval$GSE38931=GSE38931_sig$adj.P.Val
list_log2FC$GSE38931=GSE38931_sig$logFC
```

```{r}
plotMDS(exprs(BSData), col=as.numeric(as.factor(phenodata$ploidy)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))

```




### GSE34458

Molecular consequences of trisomy in lymphoblastoid cell lines from patients with Down syndrome. This project analyses differentially expressed genes between humans with trisomy 21 and humans without trisomy 21.


```{r}
BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE34458_CHD_Illumina_HT12_Raw_data.txt", 
                    sep="\t", ProbeID="PROBE_ID", skip=0,
                    columns = list(exprs = "AVG_Signal", Detection="Detection Pval"),
                    illuminaAnnotation="Humanv3")

BSData=normaliseIllumina(BSData, method="quantile", transform="log2")
pData(BSData) = phenodata

BSData=BSData[which(fData(BSData)$Status=="regular"),]

expressed=apply(Detection(BSData)<0.05,1,any)
BSData=BSData[expressed,]

dim(BSData)



ids <-as.character(featureNames(BSData))
qual <- unlist(mget(ids, illuminaHumanv4PROBEQUALITY, ifnotfound=NA))
ens <- mget(ids, illuminaHumanv4ENSEMBL, ifnotfound=NA)
rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

BSData <-BSData[!rem,]



ploidy=c(rep(c("trisomic", "euploid"), 10), "euploid", "trisomic", "euploid")


sex=c("female","female", "male", "female", "male", "male","male","male", "male", "female", "male", "male", "male", "female", "female", "male",
      "female", "female", "male", "male", "male", "female", "female")


require("illuminaHumanv3.db")
ids <-mget(featureNames(BSData), illuminaHumanv3ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-euploid,levels=design)

aw=arrayWeights(exprs(BSData)[as.character(nomi),], design)

corfit<-duplicateCorrelation(exprs(BSData)[nomi,],design, block=sex)

fit <- lmFit(exprs(BSData)[nomi,], design, weights = aw, block=sex, cor=corfit$consensus)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE34458_topT=topTable(contr.fit, 1, number=Inf)

GSE34458_sig=GSE34458_topT[which(GSE34458_topT$adj.P.Val< pthreshold),]


GSE34458_sigs=unique(unlist(ens[rownames(GSE34458_sig)]))

GSE34458_sig$id=rownames(GSE34458_sig)
ids_df=list_to_df(ens)
ids_df=ids_df[complete.cases(ids_df),]

GSE34458_sig=merge(GSE34458_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE34458_sig=GSE34458_sig[order(GSE34458_sig$adj.P.Val, abs(GSE34458_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE34458_sig=GSE34458_sig[!duplicated(GSE34458_sig$V1),]

list_ensembl$GSE34458=GSE34458_sigs
list_adjpval$GSE34458=GSE34458_sig$adj.P.Val
list_log2FC$GSE34458=GSE34458_sig$logFC



```

```{r}
plotMDS(exprs(BSData), col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(sex)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(sex)), pch=as.numeric(as.factor(levels(as.factor(sex)))))

```


### GSE35665

All 37 samples, including 15 DS patients and 22 control individuals were analysized and divided into two age-matched groups: N group (5 DS versus 7 control individuals, age: 3 days to 38 days) and C group (10 DS versus 15 control individuals, age: 1 year to 13 years). N group includes DS (D11-D15) and Control (C16-C22), and C group includes DS (D1-D10) and Control (C1-C15).

```{r}
phenodata=data.frame("ploidy"=c(rep("trisomic",15),rep("disomic",22)),
"age"=c(rep("child",10), rep("neonate", 5),rep("child",15), rep("neonate", 7)),


"sex"=c("female", rep("male", 4), rep("female",3), rep("male",2),
      "male", rep("female",4),
      "female", rep("male",5), rep(c("female", "male"),2), "male", rep("female",2), rep("male",2),
      rep("male",2), "female", rep("male",2), rep("female",2)))


celfiles=oligoClasses::list.celfiles("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE35665_RAW/", full.names=TRUE,listGzipped=T)
data <- oligo::read.celfiles(filenames=celfiles)

eset <- oligo::rma(data, target="core")


ids <-mget(featureNames(eset), huex10sttranscriptclusterENSEMBL, ifnotfound=NA)

nomi=names(ids)[!is.na(ids)]


TS <- as.factor( paste(phenodata$age,phenodata$ploidy,sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts("child"=child.trisomic-child.disomic, "neonate"=neonate.trisomic-neonate.disomic,levels=design)

aw=arrayWeights(exprs(eset)[as.character(nomi),], design)


corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)


fit <- lmFit(exprs(eset)[nomi,], design, weights = aw, block=phenodata$sex, cor=corfit$consensus)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


child=topTable(contr.fit, coef="child", number=Inf)
neonate=topTable(contr.fit, coef="neonate", number=Inf)

child_sig=child[which(child$adj.P.Val< pthreshold),]
GSE35665child_sigs=unique(unlist(ids[rownames(child_sig)]))


neonate_sig=neonate[which(neonate$adj.P.Val< pthreshold),]
GSE35665neonate_sigs=unique(unlist(ids[rownames(neonate_sig)]))

child_sig$id=rownames(child_sig)
neonate_sig$id=rownames(neonate_sig)
GSE35665child_sig=child_sig
GSE35665neonate_sig=neonate_sig

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE35665child_sig=merge(GSE35665child_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE35665child_sig=GSE35665child_sig[order(GSE35665child_sig$adj.P.Val, abs(GSE35665child_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE35665child_sig=GSE35665child_sig[!duplicated(GSE35665child_sig$V1),]
list_ensembl$GSE35665child=GSE35665child_sigs
list_adjpval$GSE35665child=GSE35665child_sig$adj.P.Val
list_log2FC$GSE35665child=GSE35665child_sig$logFC
list_AveExpr$GSE35665child=GSE35665child_sig$AveExpr

GSE35665neonate_sig=merge(GSE35665neonate_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE35665neonate_sig=GSE35665neonate_sig[order(GSE35665neonate_sig$adj.P.Val, abs(GSE35665neonate_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE35665neonate_sig=GSE35665neonate_sig[!duplicated(GSE35665neonate_sig$V1),]
list_ensembl$GSE35665neonate=GSE35665neonate_sigs
list_adjpval$GSE35665neonate=GSE35665neonate_sig$adj.P.Val
list_log2FC$GSE35665neonate=GSE35665neonate_sig$logFC
list_AveExpr$GSE35665neonate=GSE35665neonate_sig$AveExpr

```

```{r}
plotMDS(exprs(eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$age)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$age)), pch=as.numeric(as.factor(levels(as.factor(phenodata$age)))))

```

### GSE33911

To identify the genes, which led to the augmented hematopoiesis in DS-iPS cells, we performed gene expression profiling of D12-DS-iPS cells (DS-iPS-T32, DS-iPS-T33 and DS-iPS-T31) and control hiPS cells (253G1, 253G4 and A31) generated by 3 factors.
Gene expression in blood cells differentiated from human induced pluripotent stem cells derived from patients with Down syndrome and healthy donor was measured in 12 days after induction of differentiation. Eight cell lines were used.

```{r}
phenodata=data.frame(ploidy=c(rep("disomic",3), rep("trisomic",3)))

setwd("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE33911_RAW/")
x <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE33911_RAW/"),source="agilent",green.only=TRUE, path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE33911_RAW/")
#background correction and normalization
y <- limma::backgroundCorrect(x,method="normexp",normexp.method="rma")
y <- normalizeBetweenArrays(y,method="quantile")



##filtering out low expressed probes
#compute the 95% percentile of the negative control probes on each array
neg95 <- apply(y$E[y$genes$ControlType==-1,],2,function(x) quantile(x,p=0.95))
cutoff <- matrix(1.1*neg95,nrow(y),ncol(y),byrow=TRUE)
#keep probes that are at least 10% brighter than the negative controls on at least four arrays (because there are four replicates)
isexpr <- rowSums(y$E > cutoff) >= 3

#Regular probes are code as ??0?? in the ControlType column
y0 <- y[y$genes$ControlType==0 & isexpr,]
yave <- avereps(y0,ID=y0$genes[,"SystematicName"])

anno_agilent_human=getBM(attributes=c("ensembl_gene_id","agilent_sureprint_g3_ge_8x60k"),filters="agilent_sureprint_g3_ge_8x60k", values =yave$genes$ProbeName, mart = human)


isanno=which(yave$genes$ProbeName %in% anno_agilent_human$agilent_sureprint_g3_ge_8x60k)
yave=yave[isanno,]
TS <- as.factor(phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


#array weights
aw=arrayWeights(yave, design)
#fit the linear model blocking for the mouse
fit <- lmFit(yave,design,weights=aw)

contrasts <-makeContrasts(trisomic-disomic,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE33911=topTable(contr.fit, coef=1, number=Inf)

GSE33911_sig=GSE33911[which(GSE33911$adj.P.Val< pthreshold),]

GSE33911_sigs=unique(anno_agilent_human[which(anno_agilent_human$agilent_sureprint_g3_ge_8x60k %in% GSE33911_sig$ProbeName),1])



GSE33911_sig=merge(GSE33911_sig, anno_agilent_human, by.x="ProbeName", by.y="agilent_sureprint_g3_ge_8x60k", all.x=TRUE, all.y=FALSE)
GSE33911_sig=GSE33911_sig[order(GSE33911_sig$adj.P.Val, abs(GSE33911_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE33911_sig=GSE33911_sig[!duplicated(GSE33911_sig$ensembl_gene_id),]

list_ensembl$GSE33911=GSE33911_sigs
list_adjpval$GSE33911=GSE33911_sig$adj.P.Val
list_log2FC$GSE33911=GSE33911_sig$logFC

```

```{r}
plotMDS(yave, col=as.numeric(as.factor(phenodata$ploidy)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))

```


### GSE23910

Among the 14 DS infants and 42 age-matched individuals with cardiac defect, only 4 from each group provided enough biological material (thymus fragments) for isolation of high-quality RNA to perform array analysis. Eight thymic samples were analyzed: C1, C2, C3 and C4 are controls, and D1, D2, D3 and D4 are from patients with Down syndrome.

Lima FA, Moreira-Filho CA, Ramos PL, Brentani H et al. Decreased AIRE expression and global thymic hypofunction in Down syndrome. J Immunol 2011 Sep 15;187(6):3422-30. PMID: 21856934



```{r}
phenodata=data.frame(sex=c("male", "male", "male", "female", "male", "female", "female", "female"),
ploidy=c(rep("disomic", 3), rep("trisomic", 4), "disomic")
)


require(limma)
#reading into the files
x <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE23910_RAW/"),source="agilent",green.only=TRUE, path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE23910_RAW/")
#background correction and normalization
y <- limma::backgroundCorrect(x,method="normexp",normexp.method="rma")
y <- normalizeBetweenArrays(y,method="quantile")



##filtering out low expressed probes
#compute the 95% percentile of the negative control probes on each array
neg95 <- apply(y$E[y$genes$ControlType==-1,],2,function(x) quantile(x,p=0.95))
cutoff <- matrix(1.1*neg95,nrow(y),ncol(y),byrow=TRUE)
#keep probes that are at least 10% brighter than the negative controls on at least four arrays (because there are four replicates)
isexpr <- rowSums(y$E > cutoff) >= 4

#Regular probes are code as ??0?? in the ControlType column
y0 <- y[y$genes$ControlType==0 & isexpr,]
yave <- avereps(y0,ID=y0$genes[,"SystematicName"])

TS <- as.factor( phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

corfit <- duplicateCorrelation(yave,design,block=phenodata$sex)#it represents the correlation between measurements made on the same mouse

#array weights
aw=arrayWeights(yave, design)
#fit the linear model blocking for the mouse
# fit <- lmFit(yave,design,block=phenodata$sex,correlation=corfit$consensus, weights=aw)
###you cannot block for sex
fit <- lmFit(yave,design, weights=aw)

contrasts <-makeContrasts(trisomic-disomic,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE23910=topTable(contr.fit, coef=1, number=Inf)

GSE23910_sig=GSE23910[which(GSE23910$adj.P.Val< pthreshold),]

GSE23910_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="agilent_wholegenome_4x44k_v1", values =GSE23910_sig$ProbeName, mart = human)$ensembl_gene_id)

ids_df=getBM(attributes=c("ensembl_gene_id","agilent_wholegenome_4x44k_v1"),filters="agilent_wholegenome_4x44k_v1", values =GSE23910_sig$ProbeName, mart = human)



ids_df=ids_df[complete.cases(ids_df),]

GSE23910_sig=merge(GSE23910_sig, ids_df, by.x="ProbeName", by.y="agilent_wholegenome_4x44k_v1", all.x=TRUE, all.y=FALSE)
GSE23910_sig=GSE23910_sig[order(GSE23910_sig$adj.P.Val, abs(GSE23910_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE23910_sig=GSE23910_sig[!duplicated(GSE23910_sig$ensembl_gene_id),]

list_ensembl$GSE23910=GSE23910_sigs
list_adjpval$GSE23910=GSE23910_sig$adj.P.Val[!is.na(GSE23910_sig$ensembl_gene_id)]
list_log2FC$GSE23910=GSE23910_sig$logFC[!is.na(GSE23910_sig$ensembl_gene_id)]

```

```{r}
plotMDS(yave, col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$sex)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$sex)), pch=as.numeric(as.factor(levels(as.factor(phenodata$sex)))))

```


### GSE21094

Gene expression profiling (GEP) can reveal characteristic signatures associated with distinct biologic subtypes of acute lymphoblastic leukemia (ALL).
We performed GEP on Down syndrome (DS) and comparison non-Down syndrome (NDS) ALL cases to identify biologic differences between these groups. 	

Ficoll-enriched, cryopreserved diagnostic bone marrow samples were obtained from patients with newly diagnosed B-precursor acute lymphoblastic leukemia.

Loudin MG, Wang J, Leung HC, Gurusiddappa S et al. Genomic profiling in Down syndrome acute lymphoblastic leukemia identifies histone gene deletions associated with altered methylation profiles. Leukemia 2011 Oct;25(10):1555-63. PMID: 21647151


```{r}


pheno=read.csv("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE21094.csv", header=F)

ploidy=unlist(lapply(as.character(pheno$V1),FUN=function(x)strsplit(x, split=", ")[[1]][1]))

data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE21094_RAW/")

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


aw=arrayWeights(eset, design)

contrasts <-makeContrasts(ALL_DS-ALL_NDS,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE21094=topTable(contr.fit, coef=1, number=Inf)

GSE21094_sig=GSE21094[which(GSE21094$adj.P.Val< pthreshold),]

GSE21094_sigs=unique(unlist(ids[rownames(GSE21094_sig)]))

GSE21094_sig$id=rownames(GSE21094_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE21094_sig=merge(GSE21094_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE21094_sig=GSE21094_sig[order(GSE21094_sig$adj.P.Val, abs(GSE21094_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE21094_sig=GSE21094_sig[!duplicated(GSE21094_sig$V1),]
list_ensembl$GSE21094=GSE21094_sigs
list_adjpval$GSE21094=GSE21094_sig$adj.P.Val
list_log2FC$GSE21094=GSE21094_sig$logFC

```

```{r}
plotMDS(exprs(eset)[nomi,], col=as.numeric(as.factor(ploidy)))
legend("topleft", legend=levels(as.factor(ploidy)), fill=as.numeric(as.factor(levels(as.factor(ploidy)))))

```


### GSE24272

```{r}

GSE24272 <- getGEO('GSE24272',GSEMatrix=TRUE)

eset=log2(exprs(GSE24272$GSE24272_series_matrix.txt.gz))

# starts=featureData(GSE24272$GSE24272_series_matrix.txt.gz)$START
# stops=featureData(GSE24272$GSE24272_series_matrix.txt.gz)$STOP
# chrs=featureData(GSE24272$GSE24272_series_matrix.txt.gz)$CHROMOSOME
ids=featureData(GSE24272$GSE24272_series_matrix.txt.gz)$ID
require(org.Mm.eg.db)
# df <- data.frame(chr=chrs, start=starts, end=stops, id=ids)
# df=df[complete.cases(df),]
# df$chr=as.character(df$chr)
# 
# require(GenomicRanges)
# df_range=makeGRangesFromDataFrame(df, keep.extra.columns = T)
# require("TxDb.Mmusculus.UCSC.mm9.knownGene")
# txdb=genes(TxDb.Mmusculus.UCSC.mm9.knownGene)
# require("ChIPpeakAnno")
# anno <- annotatePeakInBatch(df_range, AnnotationData=txdb, output="inside")
#     macs.anno <- addGeneIDs(annotatedPeak=anno, 
#                             orgAnn="org.Mm.eg.db", 
#                             IDs2Add="ensembl",
#                             feature_id_type="entrez_id"
#                            )
# toens=cbind(as.character(macs.anno$id), macs.anno$ensembl)
xx <- as.list(org.Mm.egACCNUM2EG)
xx.ann=xx[ids]


TS <- as.factor(paste(gsub(" ", "", pData(GSE24272$GSE24272_series_matrix.txt.gz)$genotype),gsub(" ", "",pData(GSE24272$GSE24272_series_matrix.txt.gz)$tissue) , sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts("fetalliver"=Ts1Cje.fetalliver-wildtype.fetalliver, "placenta"=Ts1Cje.placenta-wildtype.placenta,levels=design)

aw=arrayWeights(eset, design)


fit <- lmFit(eset, design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


fetalliver=topTable(contr.fit, coef="fetalliver", number=Inf)
placenta=topTable(contr.fit, coef="placenta", number=Inf)

fetalliver_sig=fetalliver[which(fetalliver$adj.P.Val< pthreshold),]
placenta_sig=placenta[which(placenta$adj.P.Val< pthreshold),]


GSE24272fetalliver_sigs=unique(getBM(attributes="ensembl_gene_id",filters="entrezgene", values =unique(unlist(xx.ann[rownames(fetalliver_sig)])), mart = mouse)$ensembl_gene_id)

fetalliver_sig$id=rownames(fetalliver_sig)
ids_df=list_to_df(xx.ann)
ids_df=ids_df[complete.cases(ids_df),]
fetalliver_sig$id=rownames(fetalliver_sig)
fetalliver_sig=merge(fetalliver_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
fetalliver_sig=fetalliver_sig[order(fetalliver_sig$adj.P.Val, abs(fetalliver_sig$logFC), decreasing=c(FALSE, TRUE)),]
fetalliver_sig=fetalliver_sig[!duplicated(fetalliver_sig$V1),]

ids_df=getBM(attributes=c("ensembl_gene_id","entrezgene"),filters="entrezgene", values =fetalliver_sig$V1, mart = mouse)
ids_df=ids_df[complete.cases(ids_df),]

fetalliver_sig=merge(fetalliver_sig, ids_df, by.x="V1", by.y="entrezgene", all.x=TRUE, all.y=FALSE)
fetalliver_sig=fetalliver_sig[order(fetalliver_sig$adj.P.Val, abs(fetalliver_sig$logFC), decreasing=c(FALSE, TRUE)),]
fetalliver_sig=fetalliver_sig[!duplicated(fetalliver_sig$ensembl_gene_id),]
fetalliver_sig=fetalliver_sig[!is.na(fetalliver_sig$ensembl_gene_id),]
GSE24272fetalliver_sig=fetalliver_sig

GSE24272placenta_sigs=unique(getBM(attributes="ensembl_gene_id",filters="entrezgene", values =unique(unlist(xx.ann[rownames(placenta_sig)])), mart = mouse)$ensembl_gene_id)

ids_df=list_to_df(xx.ann)
ids_df=ids_df[complete.cases(ids_df),]
placenta_sig$id=rownames(placenta_sig)
placenta_sig=merge(placenta_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
placenta_sig=placenta_sig[order(placenta_sig$adj.P.Val, abs(placenta_sig$logFC), decreasing=c(FALSE, TRUE)),]
placenta_sig=placenta_sig[!duplicated(placenta_sig$V1),]

ids_df=getBM(attributes=c("ensembl_gene_id","entrezgene"),filters="entrezgene", values =placenta_sig$V1, mart = mouse)
ids_df=ids_df[complete.cases(ids_df),]

placenta_sig=merge(placenta_sig, ids_df, by.x="V1", by.y="entrezgene", all.x=TRUE, all.y=FALSE)
placenta_sig=placenta_sig[order(placenta_sig$adj.P.Val, abs(placenta_sig$logFC), decreasing=c(FALSE, TRUE)),]
placenta_sig=placenta_sig[!duplicated(placenta_sig$ensembl_gene_id),]
placenta_sig=placenta_sig[!is.na(placenta_sig$ensembl_gene_id),]
GSE24272placenta_sig=placenta_sig

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE24272fetalliver_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
mouse_human=merge(GSE24272fetalliver_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE24272fetalliver_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE24272fetalliver_sigs_hum=unique(GSE24272fetalliver_sig_hum$human)


list_ensembl$GSE24272fetalliver=GSE24272fetalliver_sigs_hum
list_adjpval$GSE24272fetalliver=GSE24272fetalliver_sig_hum$adj.P.Val
list_log2FC$GSE24272fetalliver=GSE24272fetalliver_sig_hum$logFC



mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE24272placenta_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=merge(GSE24272placenta_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE24272placenta_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE24272placenta_sigs_hum=unique(GSE24272placenta_sig_hum$human)


list_ensembl$GSE24272placenta=GSE24272placenta_sigs_hum
list_adjpval$GSE24272placenta=GSE24272placenta_sig_hum$adj.P.Val
list_log2FC$GSE24272placenta=GSE24272placenta_sig_hum$logFC

```

```{r}
plotMDS(eset, col=as.numeric(as.factor(pData(GSE24272$GSE24272_series_matrix.txt.gz)$genotype)), pch=as.numeric(as.factor(pData(GSE24272$GSE24272_series_matrix.txt.gz)$tissue)))

legend("topleft", legend=levels(as.factor(pData(GSE24272$GSE24272_series_matrix.txt.gz)$genotype)), fill=as.numeric(as.factor(levels(as.factor(pData(GSE24272$GSE24272_series_matrix.txt.gz)$genotype)))))

legend("bottomleft", legend=levels(as.factor(pData(GSE24272$GSE24272_series_matrix.txt.gz)$tissue)), pch=as.numeric(as.factor(levels(as.factor(pData(GSE24272$GSE24272_series_matrix.txt.gz)$tissue)))))

```

### GSE24554

Gene expression of mandibular precursor from embryonic day 13.5 trisomic and euploid embryos from the Ts65Dn Down syndrome mouse model. Results provide insight into importance of non-trisomic genes in organogenesis.

```{r}

require(illuminaMousev2.db)
BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE24554_non-normalized_probe.txt", 
                    sep="\t", ProbeID="ID_REF", skip=0,
                    columns = list(exprs = "AVG_Signal", Detection="Detection Pval"),
                    illuminaAnnotation="Mousev2")

pData(BSData)$ploidy = as.factor(c(rep("disomic", 11), rep("trisomic", 13)))
BSData=normaliseIllumina(BSData, method="quantile", transform="log2")

BSData=BSData[which(fData(BSData)$Status=="regular"),]

expressed=apply(Detection(BSData)<0.05,1,any)
BSData=BSData[expressed,]




ids <-as.character(featureNames(BSData))
qual <- unlist(mget(ids, illuminaMousev2PROBEQUALITY, ifnotfound=NA))
ens <- mget(ids, illuminaMousev2ENSEMBL, ifnotfound=NA)
rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

BSData <-BSData[!rem,]



TS <- pData(BSData)$ploidy
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(exprs(BSData), design)

fit <- lmFit(exprs(BSData), design,  weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE24554=topTable(contr.fit, coef=1, number=Inf)


GSE24554_sig=GSE24554[which(GSE24554$adj.P.Val< pthreshold),]

GSE24554_sigs=unique(unlist(ens[rownames(GSE24554_sig)]))


list_ensembl$GSE24554=character(0)
list_adjpval$GSE24554=numeric(0)
list_log2FC$GSE24554=numeric(0)
```
```{r}
plotMDS(exprs(BSData), col=as.numeric(as.factor(pData(BSData)$ploidy)))
legend("topleft", legend=levels(as.factor(pData(BSData)$ploidy)), fill=as.numeric(as.factor(levels(as.factor(pData(BSData)$ploidy)))))

```


### GSE16677

The goal of this study is to define a gene expression signature unique to DS-AMKL (acute megakaryoblastic leukemia or FAB M7 leukemia).
A similar study was done previously, but using unfractionated patient leukemic samples. In this study, we sorted megakaryocytic leukemia blasts from patients and then compared their gene expression signatures to those from similarly sorted blasts from patients with non-DS AMKL. This allowed us to identify a gene expression signature more unique to DS-AMKL samples.

The leukemic blasts were sorted based on CD41, CD7, CD117, CD33, and CD34 antibodies as previously described (Klin. Padiatr. 217, 126-134).
PMID: 20679399

```{r}
phenodata=data.frame(type=c(rep("blasts",11), rep("celllines",6)),
ploidy=c(rep("trisomic", 6), rep("disomic", 5), "disomic", "trisomic", "disomic", "trisomic", "disomic", "disomic")
)

data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE16677_RAW/")

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")


ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(paste(phenodata$ploidy, phenodata$type, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


aw=arrayWeights(eset, design)

contrasts <-makeContrasts("blasts"=trisomic.blasts-disomic.blasts, "cell"=trisomic.celllines-disomic.celllines,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


blasts=topTable(contr.fit, coef="blasts", number=Inf)
cells=topTable(contr.fit, coef="cell", number=Inf)

GSE16677blasts_sig=blasts[which(blasts$adj.P.Val< pthreshold),]
GSE16677cells_sig=cells[which(cells$adj.P.Val< pthreshold),]

GSE16677blasts_sigs=unique(unlist(ids[rownames(GSE16677blasts_sig)]))
GSE16677cells_sigs=unique(unlist(ids[rownames(GSE16677cells_sig)]))


GSE16677blasts_sig$id=rownames(GSE16677blasts_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE16677blasts_sig=merge(GSE16677blasts_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE16677blasts_sig=GSE16677blasts_sig[order(GSE16677blasts_sig$adj.P.Val, abs(GSE16677blasts_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE16677blasts_sig=GSE16677blasts_sig[!duplicated(GSE16677blasts_sig$V1),]
list_ensembl$GSE16677blasts=GSE16677blasts_sigs
list_adjpval$GSE16677blasts=GSE16677blasts_sig$adj.P.Val
list_log2FC$GSE16677blasts=GSE16677blasts_sig$logFC


GSE16677cells_sig$id=rownames(GSE16677cells_sig)
GSE16677cells_sig=merge(GSE16677cells_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE16677cells_sig=GSE16677cells_sig[order(GSE16677cells_sig$adj.P.Val, abs(GSE16677cells_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE16677cells_sig=GSE16677cells_sig[!duplicated(GSE16677cells_sig$V1),]
list_ensembl$GSE16677cells=GSE16677cells_sigs
list_adjpval$GSE16677cells=GSE16677cells_sig$adj.P.Val
list_log2FC$GSE16677cells=GSE16677cells_sig$logFC
```

```{r}
plotMDS(exprs(eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$type)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$type)), pch=as.numeric(as.factor(levels(as.factor(phenodata$type)))))

```


### GSE17760

Down syndrome is the most common form of genetic mental retardation. How Trisomy 21 causes mental retardation remains unclear and its effects on adult neurogenesis have not been addressed. To gain insight into the mechanisms causing mental retardation we used microarrays to investigate gene expression differences between Ts1Cje (a mouse model of Down syndrome) and C57BL/6 littermate control neurospheres. The neurospheres were generated from neural stem cells and progenitors isolated from the lateral walls of the lateral ventricles from adult mice.
 
```{r}

data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE17760_RAW/")

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

ids <-mget(featureNames(eset), mouse4302ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]

ploidy=rep(c("disomic", "trisomic"),3)
TS <- as.factor( ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts(trisomic-disomic,
                          levels=design)

aw=arrayWeights(exprs(eset)[as.character(nomi),], design)


fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE17760=topTable(contr.fit, coef=1, number=Inf)


GSE17760_sig=GSE17760[which(GSE17760$adj.P.Val< pthreshold),]

GSE17760_sigs=unique(unlist(ids[rownames(GSE17760_sig)]))

GSE17760_sig$id=rownames(GSE17760_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE17760_sig=merge(GSE17760_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE17760_sig=GSE17760_sig[order(GSE17760_sig$adj.P.Val, abs(GSE17760_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE17760_sig=GSE17760_sig[!duplicated(GSE17760_sig$V1),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE17760_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=merge(GSE17760_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE17760_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE17760_sigs_hum=unique(GSE17760_sig_hum$human)


list_ensembl$GSE17760=GSE17760_sigs_hum
list_adjpval$GSE17760=GSE17760_sig_hum$adj.P.Val
list_log2FC$GSE17760=GSE17760_sig_hum$logFC
```

```{r}
plotMDS(exprs(eset)[nomi,], col=as.numeric(as.factor(ploidy)))
legend("topleft", legend=levels(as.factor(ploidy)), fill=as.numeric(as.factor(levels(as.factor(ploidy)))))

```


### GSE17459

DS-ALL is a highly heterogeneous disease with predominance of an aberrant exp. of CRLF2 cooperating with mutated JAK2

Acute lymphoblastic pediatric leukemia specimens of Down's syndrome are examined for gene expression profiles and specific genetic aberrations. Gene expression profiling and specific genetic variation analysis identify novel pathways involved in DS-ALL pathogenesis.


```{r}
data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE17459_RAW/")

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

phenodata=data.frame(ploidy=c(rep("trisomic", 25), rep("disomic", 72),rep("trisomic", 6),rep("disomic", 9)),
                     study=c(rep("AIEOP", 25+72), rep("ICH",15)))

ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(paste(phenodata$ploidy, phenodata$study, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


aw=arrayWeights(eset, design)

contrasts <-makeContrasts("AIEOP"=trisomic.AIEOP-disomic.AIEOP, "ICH"=trisomic.ICH-disomic.ICH,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)
contr.fit <-eBayes(contrasts.fit(fit, contrasts))


aieop=topTable(contr.fit, coef="AIEOP", number=Inf)
ich=topTable(contr.fit, coef="ICH", number=Inf)

GSE17459aieop_sig=aieop[which(aieop$adj.P.Val< pthreshold),]
GSE17459ich_sig=ich[which(ich$adj.P.Val< pthreshold),]

GSE17459aieop_sigs=unique(unlist(ids[rownames(GSE17459aieop_sig)]))
GSE17459ich_sigs=unique(unlist(ids[rownames(GSE17459ich_sig)]))

GSE17459aieop_sig$id=rownames(GSE17459aieop_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE17459aieop_sig=merge(GSE17459aieop_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE17459aieop_sig=GSE17459aieop_sig[order(GSE17459aieop_sig$adj.P.Val, abs(GSE17459aieop_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE17459aieop_sig=GSE17459aieop_sig[!duplicated(GSE17459aieop_sig$V1),]
list_ensembl$GSE17459aieop=GSE17459aieop_sigs
list_adjpval$GSE17459aieop=GSE17459aieop_sig$adj.P.Val
list_log2FC$GSE17459aieop=GSE17459aieop_sig$logFC


GSE17459ich_sig$id=rownames(GSE17459ich_sig)
GSE17459ich_sig=merge(GSE17459ich_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE17459ich_sig=GSE17459ich_sig[order(GSE17459ich_sig$adj.P.Val, abs(GSE17459ich_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE17459ich_sig=GSE17459ich_sig[!duplicated(GSE17459ich_sig$V1),]
list_ensembl$GSE17459ich=GSE17459ich_sigs
list_adjpval$GSE17459ich=GSE17459ich_sig$adj.P.Val
list_log2FC$GSE17459ich=GSE17459ich_sig$logFC

```


```{r}
plotMDS(exprs(eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)), pch=as.numeric(as.factor(phenodata$study)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))
legend("bottomleft", legend=levels(as.factor(phenodata$study)), pch=as.numeric(as.factor(levels(as.factor(phenodata$study)))))

```

### GSE19681

The goal of this study is to define the global gene expression profile of primary leukemic blasts from patients.
Here we report the global gene expression profile of 2 patients with AML FAB M7, and 3 patients with Down syndrome AML FAB M7.

```{r}
data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE19681_RAW/")

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

phenodata=data.frame(ploidy=c(rep("disomic", 2), rep("trisomic",3)))

ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor(phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic-disomic,levels=design)

#convergence problem at gene 1621: array weights not estimable
fit <- lmFit(exprs(eset)[nomi,], design)
contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE19681=topTable(contr.fit, coef=1, number=Inf)

GSE19681_sig=GSE19681[which(GSE19681$adj.P.Val< pthreshold),]

GSE19681_sigs=unique(unlist(ids[rownames(GSE19681_sig)]))

GSE19681_sig$id=rownames(GSE19681_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE19681_sig=merge(GSE19681_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE19681_sig=GSE19681_sig[order(GSE19681_sig$adj.P.Val, abs(GSE19681_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE19681_sig=GSE19681_sig[!duplicated(GSE19681_sig$V1),]

if(is.null(GSE19681_sigs)){
list_ensembl$GSE19681=character(0)}else{list_ensembl$GSE19681=GSE19681_sigs}

list_adjpval$GSE19681=GSE19681_sig$adj.P.Val
list_log2FC$GSE19681=GSE19681_sig$logFC

```


```{r}
plotMDS(exprs(eset)[nomi,], col=as.numeric(as.factor(phenodata$ploidy)))
legend("topleft", legend=levels(phenodata$ploidy), fill=as.numeric(as.factor(levels(phenodata$ploidy))))

```

### GSE9805

Neural precursor cells (NPCs) were isolated from mouse E14 neocortex of Down syndrome Ts1Cje mice and euploid littermates. We compared gene expression profiles from trisomic and wild-type cells using pangenomic microarrays.

```{r}

setwd("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE9805_RAW/")
RG <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE9805_RAW/"), path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE9805_RAW/",columns=list(R = "F635 Median", 
                G = "F532 Median", Rb = "B635 Median", Gb = "B532 Median"),annotation=list(ID="ID", Name="Name"))
#background correction and normalization
y <- limma::backgroundCorrect(RG,method="normexp",normexp.method="rma")
y <- normalizeWithinArrays(y, method="loess")
y <- normalizeBetweenArrays(y,method="Aquantile")

targets=data.frame(Name=c("GSM247454.txt",
"GSM247455.txt",
"GSM247456.txt",
"GSM247457.txt",
"GSM247458.txt",
"GSM247459.txt"),  Cy5=c("Ts1Cje",
"Ts1Cje",
"Ts1Cje","WT",
"WT",
"WT"),Cy3=c("WT",
"WT",
"WT",
"Ts1Cje",
"Ts1Cje",
"Ts1Cje"))
design <- cbind(DyeEffect=1,MUvsWT=c(1,1,1,-1,-1,-1))

aw=arrayWeights(y, design)

fit <- lmFit(y, design, weights = aw)
fit <- eBayes(fit)


GSE9805=topTable(fit, coef="MUvsWT", number=Inf)

GSE9805_sig=GSE9805[which(GSE9805$adj.P.Val< pthreshold),]

if(dim(GSE9805_sig)[1]>0){
GSE9805_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="mgi_symbol", values =GSE9805_sig$Name, mart = mouse)$ensembl_gene_id)
list_ensembl$GSE9805=GSE9805_sigs
list_adjpval$GSE9805=GSE9805_sig$adj.P.Val
list_log2FC$GSE9805=GSE9805_sig$logFC
}



list_ensembl$GSE9805=character(0)
list_adjpval$GSE9805=numeric(0)
list_log2FC$GSE9805=numeric(0)
```



```{r}
plotMDS(y, col=as.numeric(as.factor(c("Ts1Cje",
"Ts1Cje",
"Ts1Cje","WT",
"WT",
"WT"))))
legend("topleft", legend=levels(as.factor(c("Ts1Cje",
"Ts1Cje",
"Ts1Cje","WT",
"WT",
"WT"))), fill=as.numeric(as.factor(levels(as.factor(c("Ts1Cje",
"Ts1Cje",
"Ts1Cje","WT",
"WT",
"WT"))))))





```


### GSE11448


We analyzed gene expression in the cerebellum of Ts1Cje and euploid mice at P0, P3, P7 and P10 using pangenomic two colors microarrays containing 25344 probes representing approximately 15574 mouse genes. 56 samples from individual cerebellum were hybridized on 28 microarrays. On each microarray we hybridized a Ts1Cje sample versus an euploid sample and always a male versus a female. In addition on the same microarray we always compared samples from mice of the same age or with a maximum difference of 4 days (P0 versus P3, P3 versus P7 or P7 versus P10).


```{r}
targets=read.csv("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE11448.csv", header=T)

RG <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE11448_RAW/"), path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE11448_RAW/", source="genepix")


y <- limma::backgroundCorrect(RG,method="normexp",normexp.method="rma")
y <- normalizeWithinArrays(y, method="loess")
y <- normalizeBetweenArrays(y,method="Aquantile")


targets2 <- targetsA2C(targets)

u <- unique(targets2$Target)
 f <- factor(targets2$Target, levels=u)
design <- model.matrix(~0+f)
colnames(design) <- u

corfit <- intraspotCorrelation(y, design)


fit <- lmscFit(y, design, correlation=corfit$consensus)


cont.matrix <- makeContrasts("P0"=Ts1Cje.P0-Euploid.P0,"P3"=Ts1Cje.P3-Euploid.P3, "P7"=Ts1Cje.P7-Euploid.P7, "P10"=Ts1Cje.P10-Euploid.P10,levels=design)

fit2 <- contrasts.fit(fit, cont.matrix)
fit2 <- eBayes(fit2)

GSE11448_p0=topTable(fit2, adjust="BH", coef="P0", number=Inf)
GSE11448_p3=topTable(fit2, adjust="BH", coef="P3", number=Inf)
GSE11448_p7=topTable(fit2, adjust="BH", coef="P7", number=Inf)
GSE11448_p10=topTable(fit2, adjust="BH", coef="P10", number=Inf)


GSE11448_p0_sig=GSE11448_p0[which(GSE11448_p0$adj.P.Val< pthreshold),]
GSE11448_p3_sig=GSE11448_p3[which(GSE11448_p3$adj.P.Val< pthreshold),]
GSE11448_p7_sig=GSE11448_p7[which(GSE11448_p7$adj.P.Val< pthreshold),]
GSE11448_p10_sig=GSE11448_p10[which(GSE11448_p10$adj.P.Val< pthreshold),]


if(dim(GSE11448_p0_sig)[1]>0){
GSE11448_p0_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="mgi_symbol", values =GSE11448_p0_sig$Name, mart = mouse)$ensembl_gene_id)
ids_df=getBM(attributes=c("ensembl_gene_id","mgi_symbol"),filters="mgi_symbol", values =GSE11448_p0_sig$Name, mart = mouse)

GSE11448_p0_sig=merge(GSE11448_p0_sig, ids_df, by.x="Name", by.y="mgi_symbol", all.x=TRUE, all.y=FALSE)
GSE11448_p0_sig=GSE11448_p0_sig[order(GSE11448_p0_sig$adj.P.Val, abs(GSE11448_p0_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE11448_p0_sig=GSE11448_p0_sig[!duplicated(GSE11448_p0_sig$ensembl_gene_id),]
GSE11448_p0_sig=GSE11448_p0_sig[!is.na(GSE11448_p0_sig$ensembl_gene_id),]

if(length(GSE11448_p0_sigs)>0){
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE11448_p0_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE11448_p0_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE11448_p0_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE11448_p0_sigs_hum=unique(GSE11448_p0_sig_hum$human)


list_ensembl$GSE11448_p0=GSE11448_p0_sigs_hum
list_adjpval$GSE11448_p0=GSE11448_p0_sig_hum$adj.P.Val
list_log2FC$GSE11448_p0=GSE11448_p0_sig_hum$logFC

}}

if(dim(GSE11448_p3_sig)[1]>0){
GSE11448_p3_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="mgi_symbol", values =GSE11448_p3_sig$Name, mart = mouse)$ensembl_gene_id)
if(length(GSE11448_p3_sigs)>0){
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE11448_p3_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
GSE11448_p3_sigs_hum=unique(mouse_human$human)
length(GSE11448_p3_sigs_hum)
} else{
  list_ensembl$GSE11448_p3=character(0)
list_adjpval$GSE11448_p3=numeric(0)
list_log2FC$GSE11448_p3=numeric(0)
}

} else{
  list_ensembl$GSE11448_p3=character(0)
list_adjpval$GSE11448_p3=numeric(0)
list_log2FC$GSE11448_p3=numeric(0)
}



if(dim(GSE11448_p7_sig)[1]>0){
GSE11448_p7_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="mgi_symbol", values =GSE11448_p7_sig$Name, mart = mouse)$ensembl_gene_id)
ids_df=getBM(attributes=c("ensembl_gene_id","mgi_symbol"),filters="mgi_symbol", values =GSE11448_p7_sig$Name, mart = mouse)

GSE11448_p7_sig=merge(GSE11448_p7_sig, ids_df, by.x="Name", by.y="mgi_symbol", all.x=TRUE, all.y=FALSE)
GSE11448_p7_sig=GSE11448_p7_sig[order(GSE11448_p7_sig$adj.P.Val, abs(GSE11448_p7_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE11448_p7_sig=GSE11448_p7_sig[!duplicated(GSE11448_p7_sig$ensembl_gene_id),]
GSE11448_p7_sig=GSE11448_p7_sig[!is.na(GSE11448_p7_sig$ensembl_gene_id),]

if(length(GSE11448_p7_sigs)>0){
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE11448_p7_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE11448_p7_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE11448_p7_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE11448_p7_sigs_hum=unique(GSE11448_p7_sig_hum$human)


list_ensembl$GSE11448_p7=GSE11448_p7_sigs_hum
list_adjpval$GSE11448_p7=GSE11448_p7_sig_hum$adj.P.Val
list_log2FC$GSE11448_p7=GSE11448_p7_sig_hum$logFC

}}

if(dim(GSE11448_p10_sig)[1]>0){
GSE11448_p10_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="mgi_symbol", values =GSE11448_p10_sig$Name, mart = mouse)$ensembl_gene_id)
ids_df=getBM(attributes=c("ensembl_gene_id","mgi_symbol"),filters="mgi_symbol", values =GSE11448_p10_sig$Name, mart = mouse)

GSE11448_p10_sig=merge(GSE11448_p10_sig, ids_df, by.x="Name", by.y="mgi_symbol", all.x=TRUE, all.y=FALSE)
GSE11448_p10_sig=GSE11448_p10_sig[order(GSE11448_p10_sig$adj.P.Val, abs(GSE11448_p10_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE11448_p10_sig=GSE11448_p10_sig[!duplicated(GSE11448_p10_sig$ensembl_gene_id),]
GSE11448_p10_sig=GSE11448_p10_sig[!is.na(GSE11448_p10_sig$ensembl_gene_id),]

if(length(GSE11448_p10_sigs)>0){
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE11448_p10_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE11448_p10_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE11448_p10_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE11448_p10_sigs_hum=unique(GSE11448_p10_sig_hum$human)


list_ensembl$GSE11448_p10=GSE11448_p10_sigs_hum
list_adjpval$GSE11448_p10=GSE11448_p10_sig_hum$adj.P.Val
list_log2FC$GSE11448_p10=GSE11448_p10_sig_hum$logFC

}}
```


```{r}

genotypes=as.factor(unlist(lapply(c(as.character(targets$Cy3), as.character(targets$Cy5)),FUN=function(x)strsplit(x, split=".", fixed=T)[[1]][1])))
times=as.factor(unlist(lapply(c(as.character(targets$Cy3), as.character(targets$Cy5)),FUN=function(x)strsplit(x, split=".", fixed=T)[[1]][2])))

plotMDS(cbind(RG$G, RG$R), col=as.numeric(times), pch=as.numeric(genotypes))

legend("left", legend=levels(times), fill=as.numeric(times))
legend("topright", legend=levels(genotypes), pch=as.numeric(genotypes))




```


### GSE14021


The molecular mechanisms that lead to the cognitive defects characteristic of Down syndrome (DS), the most frequent cause of mental retardation, have remained elusive. Here we use a transgenic DS mouse model to show that DYRK1A gene dosage imbalance deregulates chromosomal clusters of genes located near neuron-restrictive silencer factor (REST/NRSF) binding sites. We found that DYRK1A binds the SWI/SNF-complex known to interact with REST/NRSF. Mutation of a REST/NRSF binding site in the promoter of the REST/NRSF target gene L1cam modifies the transcriptional effect of Dyrk1A-dosage imbalance on L1cam. DyrkA dosage imbalance perturbs Rest/Nrsf levels with decreased Rest/Nrsf expression in embryonic neurons and increased expression in adult neurons. We identified a coordinated deregulation of multiple genes that are responsible for the cellular phenotypic traits present in DS such as dendritic growth impairment and microcephaly during prenatal cortex development. Dyrk1a overexpression in primary mouse cortical neurons reduced the neuritic complexity. In the postnatal hippocampus, DYRK1A overexpression suppresses a form of synaptic plasticity that may be sufficient to cause DS cognitive defects. We propose that DYRK1A overexpression-related neuronal gene deregulation generates the brain phenotypic changes that characterize DS, with an accessory role for the gene dosage imbalance of other chromosome 21 genes.

Transcriptional analysis of E12.5 telencephalon from 152F7 transgenic mouse (Dyrk1A transgenic)


```{r}
setwd("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14021_RAW/")

anno=read.csv(file="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14021_RAW/GPL2872_old_annotations.csv",header=T, na.strings = "")

RG <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14021_RAW/")[2:5], path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14021_RAW/", source="genepix")

#background correction and normalization
y <- limma::backgroundCorrect(RG,method="normexp",normexp.method="rma")
y <- normalizeWithinArrays(y, method="loess")
y <- normalizeBetweenArrays(y,method="Aquantile")


targets=data.frame(Name=c("GSM247454.txt",
"GSM352175.gpr.gz",
"GSM352176.gpr.gz",
"GSM352177.gpr.gz"),
Cy5=c("WT", "WT","152F7", "152F7"
),
Cy3=c("152F7", "152F7","WT", "WT")

)

design <- modelMatrix(targets, ref="WT")

biolrep <- c(1, 1, 2, 2)
corfit <- duplicateCorrelation(y, design, ndups = 1, block = biolrep)
aw=arrayWeights(y, design)

fit <- lmFit(y, design, block = biolrep, cor = corfit$consensus, weights = aw)
fit <- eBayes(fit)


 GSE14021=topTable(fit,  number=Inf)


GSE14021_sig=GSE14021[which(GSE14021$adj.P.Val< pthreshold),]


GSE14021_sigs= as.character(anno$ENSEMBL_ID[which(anno$SPOT_ID %in% GSE14021_sig$ID)])[!is.na(as.character(anno$ENSEMBL_ID[which(anno$SPOT_ID %in% GSE14021_sig$ID)]))]

ids_df=cbind(as.character(anno$ENSEMBL_ID), as.character(anno$SPOT_ID))
colnames(ids_df)=c("V1", "V2")
ids_df=ids_df[complete.cases(ids_df),]

GSE14021_sig=merge(GSE14021_sig, ids_df, by.x="ID", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE14021_sig=GSE14021_sig[order(GSE14021_sig$adj.P.Val, abs(GSE14021_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE14021_sig=GSE14021_sig[!duplicated(GSE14021_sig$V1),]
GSE14021_sig=GSE14021_sig[!is.na(GSE14021_sig$V1),]


GSE14021_sigs=getBM(attributes=c("ensembl_gene_id"),filters="ensembl_transcript_id", values =GSE14021_sigs, mart = mouse)$ensembl_gene_id

ids_df=getBM(attributes=c("ensembl_gene_id", "ensembl_transcript_id"),filters="ensembl_transcript_id", values =as.character(GSE14021_sig$V1), mart = mouse)


GSE14021_sig=merge(GSE14021_sig, ids_df, by.x="V1", by.y="ensembl_transcript_id", all.x=TRUE, all.y=FALSE)
GSE14021_sig=GSE14021_sig[order(GSE14021_sig$adj.P.Val, abs(GSE14021_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE14021_sig=GSE14021_sig[!duplicated(GSE14021_sig$ensembl_gene_id),]
GSE14021_sig=GSE14021_sig[!is.na(GSE14021_sig$ensembl_gene_id),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE14021_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE14021_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE14021_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE14021_sigs_hum=unique(GSE14021_sig_hum$human)


list_ensembl$GSE14021=GSE14021_sigs_hum
list_adjpval$GSE14021=GSE14021_sig_hum$adj.P.Val
list_log2FC$GSE14021=GSE14021_sig_hum$logFC

```


```{r}
plotMDS(y, col=as.numeric(as.factor(c("152F7", "152F7","WT", "WT")
  
  
)))
legend("topleft", legend=levels(as.factor(c("152F7", "152F7","WT", "WT")
  
)), fill=as.numeric(as.factor(levels(as.factor(c("152F7", "152F7","WT", "WT")
  
)))))
```


### GSE14030

Transcriptional analysis of murine neurobastoma N18 cell line transfected with a pAd-Dyrk1a vector
 	
```{r}
setwd("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14030_RAW/")
RG <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14030_RAW/"), path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14030_RAW/", source="genepix")
#background correction and normalization
y <- limma::backgroundCorrect(RG,method="normexp",normexp.method="rma")
y <- normalizeWithinArrays(y, method="loess")
y <- normalizeBetweenArrays(y,method="Aquantile")

targets=data.frame(Name=c(
"GSM352270.gpr.gz",
"GSM352271.gpr.gz",
"GSM352272.gpr.gz",
"GSM352277.gpr.gz",
"GSM352278.gpr.gz"),  Cy5=c("Dyrk1A",
"Dyrk1A",
"Dyrk1A","WT",
"WT"
),Cy3=c("WT",
"WT",
"WT",
"Dyrk1A", "Dyr1A"))


design <- cbind(DyeEffect=1,MUvsWT=c(1,1,1,-1,-1))

aw=arrayWeights(y, design)

fit <- lmFit(y, design, weights = aw)
fit <- eBayes(fit)


GSE14030=topTable(fit, coef="MUvsWT", number=Inf)

GSE14030_sig=GSE14030[which(GSE14030$adj.P.Val< pthreshold),]

anno=read.csv(file="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE14021_RAW/GPL2872_old_annotations.csv",header=T, na.strings = "")

GSE14030_sigs= as.character(anno$ENSEMBL_ID[which(anno$SPOT_ID %in% GSE14030_sig$ID)])[!is.na(as.character(anno$ENSEMBL_ID[which(anno$SPOT_ID %in% GSE14030_sig$ID)]))]


ids_df=cbind(as.character(anno$ENSEMBL_ID), as.character(anno$SPOT_ID))
colnames(ids_df)=c("V1", "V2")
ids_df=ids_df[complete.cases(ids_df),]

GSE14030_sig=merge(GSE14030_sig, ids_df, by.x="ID", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE14030_sig=GSE14030_sig[order(GSE14030_sig$adj.P.Val, abs(GSE14030_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE14030_sig=GSE14030_sig[!duplicated(GSE14030_sig$V1),]
GSE14030_sig=GSE14030_sig[!is.na(GSE14030_sig$V1),]


ids_df=getBM(attributes=c("ensembl_gene_id", "ensembl_transcript_id"),filters="ensembl_transcript_id", values =GSE14030_sigs, mart = mouse)


GSE14030_sig=merge(GSE14030_sig, ids_df, by.x="V1", by.y="ensembl_transcript_id", all.x=TRUE, all.y=FALSE)
GSE14030_sig=GSE14030_sig[order(GSE14030_sig$adj.P.Val, abs(GSE14030_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE14030_sig=GSE14030_sig[!duplicated(GSE14030_sig$ensembl_gene_id),]
GSE14030_sig=GSE14030_sig[!is.na(GSE14030_sig$ensembl_gene_id),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE14030_sig$ensembl_gene_id, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE14030_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE14030_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE14030_sigs_hum=unique(GSE14030_sig_hum$human)


list_ensembl$GSE14030=GSE14030_sigs_hum
list_adjpval$GSE14030=GSE14030_sig_hum$adj.P.Val
list_log2FC$GSE14030=GSE14030_sig_hum$logFC

```


```{r}
plotMDS(cbind(RG$G, RG$R), labels=c(as.character(targets$Cy3),as.character(targets$Cy5)))

```


### GSE13123

Male germ cells. Three biological replicate RNA samples from enriched male germ cells from two mouse Down Syndrome mouse models, Ts1Cje and Ts65Dn, and diploid wild-type littermate controls, WT, from two time points, 16-17 and 19-20 days post partum(dpp), were assayed for gene expression using Illumina MouseWG-6 v1.1 BeadArrays.

```{r}

BSData=read.ilmn(files=list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE13123_RAW/"), 
                           path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE13123_RAW/",
                 probeid = "ProbeID",
                    sep="\t")

y <- neqc(BSData)

expressed <- rowSums(y$other$Detection < 0.05) >= 3

y <- y[expressed,]


phenodata=data.frame(strain=c(rep("WT",6), rep("Ts1Cje",6), rep("Ts65Dn", 6)), age=rep(c(rep("16_17", 3), rep("19_20", 3)),3),
                     ploidy=c(rep("disomic",6), rep("trisomic",12)))


TS <- as.factor(paste(phenodata$ploidy, phenodata$age, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


anno=read.table("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE13123_anno.txt", sep="\t", header=T)



corfit <- duplicateCorrelation(y,design,block=phenodata$strain)#it represents the correlation between measurements made on the same mouse

#array weights
aw=arrayWeights(y, design)
#fit the linear model blocking for the mouse
fit <- lmFit(y,design,block=phenodata$strain,correlation=corfit$consensus, weights=aw)

contrasts <-makeContrasts("16-17"=trisomic.16_17-disomic.16_17, "19-20"=trisomic.19_20-disomic.19_20,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE13123.16_17=topTable(contr.fit, coef="16-17", number=Inf)
GSE13123.19_20=topTable(contr.fit, coef="19-20", number=Inf)

GSE13123.16_17_sig=GSE13123.16_17[which(GSE13123.16_17$adj.P.Val< pthreshold),]

GSE13123.19_20_sig=GSE13123.19_20[which(GSE13123.19_20$adj.P.Val< pthreshold),]

if(dim(GSE13123.16_17_sig)[1]>0){
GSE13123.16_17_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="mgi_symbol", values =as.character(anno$SYMBOL[which(anno$TARGET_ID %in% GSE13123.16_17_sig$TargetID)]), mart = mouse)$ensembl_gene_id)

ids_df=getBM(attributes=c("ensembl_gene_id","mgi_symbol"),filters="mgi_symbol", values =as.character(anno$SYMBOL[which(anno$TARGET_ID %in% GSE13123.16_17_sig$TargetID)]), mart = mouse)

ids_df=merge(ids_df, anno, by.x="mgi_symbol", by.y="SYMBOL", all.x=TRUE, all.y=FALSE)


GSE13123.16_17_sig=merge(GSE13123.16_17_sig, ids_df, by.x="TargetID", by.y="TARGET_ID", all.x=TRUE, all.y=FALSE)
GSE13123.16_17_sig=GSE13123.16_17_sig[order(GSE13123.16_17_sig$adj.P.Val, abs(GSE13123.16_17_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE13123.16_17_sig=GSE13123.16_17_sig[!duplicated(GSE13123.16_17_sig$ensembl_gene_id),]


if(length(GSE13123.16_17_sigs)>0){
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE13123.16_17_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE13123.16_17_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE13123.16_17_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE13123.16_17_sigs_hum=unique(GSE13123.16_17_sig_hum$human)


list_ensembl$GSE13123.16_17=GSE13123.16_17_sigs_hum
list_adjpval$GSE13123.16_17=GSE13123.16_17_sig_hum$adj.P.Val
list_log2FC$GSE13123.16_17=GSE13123.16_17_sig_hum$logFC
}}

if(dim(GSE13123.19_20_sig)[1]>0){
  
ids_df=getBM(attributes=c("ensembl_gene_id","mgi_symbol"),filters="mgi_symbol", values =as.character(anno$SYMBOL[which(anno$TARGET_ID %in% GSE13123.19_20_sig$TargetID)]), mart = mouse)

ids_df=merge(ids_df, anno, by.x="mgi_symbol", by.y="SYMBOL", all.x=TRUE, all.y=FALSE)

  
GSE13123.19_20_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="mgi_symbol", values =as.character(anno$SYMBOL[which(anno$TARGET_ID %in% GSE13123.19_20_sig$TargetID)]), mart = mouse)$ensembl_gene_id)

GSE13123.19_20_sig=merge(GSE13123.19_20_sig, ids_df, by.x="TargetID", by.y="TARGET_ID", all.x=TRUE, all.y=FALSE)
GSE13123.19_20_sig=GSE13123.19_20_sig[order(GSE13123.19_20_sig$adj.P.Val, abs(GSE13123.19_20_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE13123.19_20_sig=GSE13123.19_20_sig[!duplicated(GSE13123.19_20_sig$ensembl_gene_id),]
GSE13123.19_20_sig=GSE13123.19_20_sig[!is.na(GSE13123.19_20_sig$ensembl_gene_id),]


if(length(GSE13123.19_20_sig)>0){
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE13123.19_20_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
mouse_human=merge(GSE13123.19_20_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE13123.19_20_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE13123.19_20_sigs_hum=unique(GSE13123.19_20_sig_hum$human)


list_ensembl$GSE13123.19_20=GSE13123.19_20_sigs_hum
list_adjpval$GSE13123.19_20=GSE13123.19_20_sig_hum$adj.P.Val
list_log2FC$GSE13123.19_20=GSE13123.19_20_sig_hum$logFC
}}

```


```{r}
plotMDS(y, col=as.numeric(as.factor(phenodata$strain)), pch=as.numeric(as.factor(phenodata$age)))
        
legend("topleft", legend=levels(as.factor(phenodata$strain)), fill=as.numeric(as.factor(levels(as.factor(phenodata$strain)))))
legend("bottomleft", legend=levels(as.factor(phenodata$age)), pch=as.numeric(as.factor(levels(as.factor(phenodata$age)))))
```

###  GSE11472


We analyzed gene expression in the EGL of Ts1Cje and euploid mice at P0 using pangenomic Illumina mouse-6 v1.1 expression beadchips containing 46 632 probes representing approximately 19 000 mouse genes. 18 samples from individual cerebellar EGL were hybridized on 18 microarrays (6 by slide). On each slide, we hybridized 6 samples frome the same litter.
This experiment included 9 Ts1Cje and 9 euploid mice at P0 from three litters 

```{r}


GSE11472 <- getGEO('GSE11472',GSEMatrix=TRUE)

eset=exprs(GSE11472$GSE11472_series_matrix.txt.gz)

TS <- as.factor(c("Ts1Cje", "Ts1Cje", "euploid","euploid", "Ts1Cje", "euploid", "Ts1Cje", "euploid", "Ts1Cje", "euploid", "euploid", "Ts1Cje", "euploid", "euploid", "Ts1Cje", "euploid", "Ts1Cje", "Ts1Cje"))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

aw=arrayWeights(eset, design)
#fit the linear model blocking for the mouse
fit <- lmFit(eset,design, weights=aw)

contrasts <-makeContrasts(Ts1Cje-euploid,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE11472=topTable(contr.fit,  number=Inf)

GSE11472_sig=GSE11472[which(GSE11472$adj.P.Val< pthreshold),]


GSE11472_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="illumina_mousewg_6_v1", values =rownames(GSE11472_sig), mart = mouse)$ensembl_gene_id)

ids_df=getBM(attributes=c("ensembl_gene_id","illumina_mousewg_6_v1"),filters="illumina_mousewg_6_v1", values =rownames(GSE11472_sig), mart = mouse)
ids_df=ids_df[complete.cases(ids_df),]

GSE11472_sig$id=rownames(GSE11472_sig)
GSE11472_sig=merge(GSE11472_sig, ids_df, by.x="id", by.y="illumina_mousewg_6_v1", all.x=TRUE, all.y=FALSE)
GSE11472_sig=GSE11472_sig[order(GSE11472_sig$adj.P.Val, abs(GSE11472_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE11472_sig=GSE11472_sig[!duplicated(GSE11472_sig$ensembl_gene_id),]
GSE11472_sig=GSE11472_sig[!is.na(GSE11472_sig$ensembl_gene_id),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = GSE11472_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(GSE11472_sig,mouse_human, by.x="ensembl_gene_id", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
GSE11472_sig_hum=mouse_human[!duplicated(mouse_human$human),]
GSE11472_sigs_hum=unique(GSE11472_sig_hum$human)


list_ensembl$GSE11472=GSE11472_sigs_hum
list_adjpval$GSE11472=GSE11472_sig_hum$adj.P.Val
list_log2FC$GSE11472=GSE11472_sig_hum$logFC

```

```{r}
plotMDS(eset, col=as.numeric(TS))
        
legend("topleft", legend=levels(TS), fill=as.numeric(as.factor(levels(TS))))
```

### GSE10758

Congenital development disorders with variable severity occur in trisomy 21. However, how these phenotypic abnormalities develop with variations remains elusive. We hypothesize that the difference in euploidy gene expression variation among trisomy 21 tissues are perturbed by the presence of an extra copy of chromosome 21 and this may contribute to the phenotypic variations in Down syndrome.
We used DNA microarray to measure the differences in gene expression variance (representing variation) between four human trisomy 21 pthresholdtes and six human euploid pthresholdtes. 

```{r}
ploidy=c(rep("trisomic", 4), rep("disomic", 6))


setwd("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE10758_RAW/")
x <- read.maimages(list.files("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE10758_RAW/"),source="agilent",green.only=TRUE, path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/GSE10758_RAW/")
#background correction and normalization
y <- limma::backgroundCorrect(x,method="normexp",normexp.method="rma")
y <- normalizeBetweenArrays(y,method="quantile")



##filtering out low expressed probes
#compute the 95% percentile of the negative control probes on each array
neg95 <- apply(y$E[y$genes$ControlType==-1,],2,function(x) quantile(x,p=0.95))
cutoff <- matrix(1.1*neg95,nrow(y),ncol(y),byrow=TRUE)
#keep probes that are at least 10% brighter than the negative controls on at least four arrays (because there are four replicates)
isexpr <- rowSums(y$E > cutoff) >= 4

#Regular probes are code as ??0?? in the ControlType column
y0 <- y[y$genes$ControlType==0 & isexpr,]
yave <- avereps(y0,ID=y0$genes[,"SystematicName"])




TS <- as.factor(ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


#array weights
aw=arrayWeights(yave, design)
#fit the linear model blocking for the mouse
fit <- lmFit(yave,design,weights=aw)

contrasts <-makeContrasts(trisomic-disomic,levels=design)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


GSE10758=topTable(contr.fit, coef=1, number=Inf)


anno=read.csv(file="~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/annotations_GPL887.csv", header=T)

GSE10758_sig=GSE10758[which(GSE10758$adj.P.Val< pthreshold),]



GSE10758_sigs=as.character(anno$ENSEMBL_ID[which(anno$SPOT_ID %in% GSE10758_sig$ProbeName)][!is.na(anno$ENSEMBL_ID[which(anno$SPOT_ID %in% GSE10758_sig$ProbeName)])])

ids_df=cbind(as.character(anno$ENSEMBL_ID), as.character(anno$SPOT_ID))
colnames(ids_df)=c("V1", "V2")
ids_df=ids_df[complete.cases(ids_df),]


GSE10758_sig=merge(GSE10758_sig, ids_df, by.x="ProbeName", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE10758_sig=GSE10758_sig[order(GSE10758_sig$adj.P.Val, abs(GSE10758_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE10758_sig=GSE10758_sig[!duplicated(GSE10758_sig$V1),]
GSE10758_sig=GSE10758_sig[!is.na(GSE10758_sig$V1),]


if(length(GSE10758_sigs)>0){
GSE10758_sigs=unique(getBM(attributes=c("ensembl_gene_id"),filters="ensembl_transcript_id", values =GSE10758_sigs, mart = human)$ensembl_gene_id)

ids_df=getBM(attributes=c("ensembl_gene_id", "ensembl_transcript_id"),filters="ensembl_gene_id", values =GSE10758_sigs, mart = human)

GSE10758_sig=merge(GSE10758_sig, ids_df, by.x="V1", by.y="ensembl_transcript_id", all.x=TRUE, all.y=FALSE)
GSE10758_sig=GSE10758_sig[order(GSE10758_sig$adj.P.Val, abs(GSE10758_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE10758_sig=GSE10758_sig[!duplicated(GSE10758_sig$ensembl_gene_id),]
GSE10758_sig=GSE10758_sig[!is.na(GSE10758_sig$ensembl_gene_id),]

list_ensembl$GSE10758=GSE10758_sigs
list_adjpval$GSE10758=GSE10758_sig$adj.P.Val
list_log2FC$GSE10758=GSE10758_sig$logFC

}




```

```{r}
plotMDS(y, col=as.numeric(TS))
        
legend("topleft", legend=levels(TS), fill=as.numeric(as.factor(levels(TS))))
```




### GSE1789

Expression profiles in normal and Down syndrome fetal heart explants.


```{r}
require("GEOquery")
GSE1789 <- getGEO('GSE1789',GSEMatrix=TRUE)


GSE1789$GSE1789_series_matrix.txt.gz=GSE1789$GSE1789_series_matrix.txt.gz[,6:13]

ploidy=c(rep("trisomic",5), rep("disomic",3))
# ploidy=c(rep("trisomic",10), rep("disomic",3))

heartdefect=c(rep("trisomic",5), rep("disomic",3))
eset=log2(exprs(GSE1789$GSE1789_series_matrix.txt.gz))

ids <-mget(featureNames(GSE1789[[1]]), hgu133aENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]

TS <- as.factor(ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

aw=arrayWeights(eset[nomi,], design)


contrasts <-makeContrasts(trisomic-disomic,levels=design)

fit <- lmFit(eset[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))

GSE1789=topTable(contr.fit, coef=1, number=Inf)

GSE1789_sig=GSE1789[which(GSE1789$adj.P.Val< pthreshold),]



GSE1789_sigs=unique(unlist(ids[rownames(GSE1789_sig)]))


list_ensembl$GSE1789=character(0)
list_adjpval$GSE1789=numeric(0)
list_log2FC$GSE1789=numeric(0)
```


```{r}
plotMDS(eset, col=as.numeric(TS))
        
legend("topleft", legend=levels(TS), fill=as.numeric(as.factor(levels(TS))))
```

### GSE4119

Expression profiling of acute megakaryoblastic leukemia: mononuclear cells were isolated on a density gradient. This series includes samples of acute megakaryoblastic leukemia from 24 patients with Down Syndrome, 28 patients without Down Syndrome, 9 samples with acute myelomonocytic leukemia of patients without Down Syndrome, 8 samples from neonates with Down Syndrome and with transient myeloproliferative disease


```{r}
require("GEOquery")
GSE4119 <- getGEO('GSE4119',GSEMatrix=TRUE)


require(affyPLM)

eset=normalize.ExpressionSet.quantiles(GSE4119[[1]])

eset=log2(exprs(eset))


ploidy=c(rep("disomic", 24), rep("trisomic",3), rep("disomic",3), "trisomic",rep("disomic", 20), rep("trisomic", 21))

type=c(rep("AMKL", 25), "AML", "AML", rep("AMKL", 15),rep("AML", 9), rep("AMKL", 21))



ids <-mget(featureNames(GSE4119[[1]]), hgu133aENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]



TS <- as.factor( paste(type, ploidy, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts("AMKL"=AMKL.trisomic-AMKL.disomic, "AML"=AML.trisomic-AML.disomic,levels=design)



aw=arrayWeights(eset[nomi,1:72], design)

fit <- lmFit(eset[nomi,1:72], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))

GSE4119.amkl=topTable(contr.fit, coef="AMKL", number=Inf)
GSE4119.aml=topTable(contr.fit, coef="AML", number=Inf)

GSE4119.amkl_sig=GSE4119.amkl[which(GSE4119.amkl$adj.P.Val< pthreshold),]
GSE4119.aml_sig=GSE4119.aml[which(GSE4119.aml$adj.P.Val< pthreshold),]



GSE4119.amkl_sigs=unique(unlist(ids[rownames(GSE4119.amkl_sig)]))
GSE4119.aml_sigs=unique(unlist(ids[rownames(GSE4119.aml_sig)]))

GSE4119.amkl_sig$id=rownames(GSE4119.amkl_sig)

ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

GSE4119.amkl_sig=merge(GSE4119.amkl_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE4119.amkl_sig=GSE4119.amkl_sig[order(GSE4119.amkl_sig$adj.P.Val, abs(GSE4119.amkl_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE4119.amkl_sig=GSE4119.amkl_sig[!duplicated(GSE4119.amkl_sig$V1),]

list_ensembl$GSE4119.amkl=GSE4119.amkl_sigs
list_adjpval$GSE4119.amkl=GSE4119.amkl_sig$adj.P.Val
list_log2FC$GSE4119.amkl=GSE4119.amkl_sig$logFC

GSE4119.aml_sig$id=rownames(GSE4119.aml_sig)


GSE4119.aml_sig=merge(GSE4119.aml_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
GSE4119.aml_sig=GSE4119.aml_sig[order(GSE4119.aml_sig$adj.P.Val, abs(GSE4119.aml_sig$logFC), decreasing=c(FALSE, TRUE)),]
GSE4119.aml_sig=GSE4119.aml_sig[!duplicated(GSE4119.aml_sig$V1),]

list_ensembl$GSE4119.aml=GSE4119.aml_sigs
list_adjpval$GSE4119.aml=GSE4119.aml_sig$adj.P.Val
list_log2FC$GSE4119.aml=GSE4119.aml_sig$logFC


```


```{r}
plotMDS(eset, col=as.numeric(as.factor(ploidy)), pch=as.numeric(as.factor(type)))
        
legend("topleft", legend=levels(as.factor(ploidy)), fill=as.numeric(as.factor(levels(as.factor(ploidy)))))
legend("bottomleft", legend=levels(as.factor(type)), pch=as.numeric(as.factor(levels(as.factor(type)))))
```



### E-TABM-473

Transcription profiling of liver tissue from humans, wild type mice and a Down syndrome mouse model containing human chromosome 21 to determine whether genetic sequence or nuclear enviornment directs tissue-specific transcription factor binding

```{r}
require(ArrayExpress)
rawset = ArrayExpress("E-TABM-473")

BLData = readIllumina(dir="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-TABM-473.raw.1/",useImages=FALSE, illuminaAnnotation = "Mousev1")

group_data=read.table("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-TABM-473.sdrf.txt", header=T, sep="\t")

BSData=summarize(BLData, useSampleFac=T, sampleFac = group_data$Source.Name)


det = calculateDetection(BSData)
Detection(BSData) <- det


BSData=BSData[,c(1,2,4,5,6,8,9,10,12)]  #onlymouse



BSData=normaliseIllumina(BSData, method="quantile")


ids <-as.character(featureNames(BSData))
qual <- unlist(mget(ids, illuminaMousev1PROBEQUALITY, ifnotfound=NA))
ens <- mget(ids, illuminaMousev1ENSEMBL, ifnotfound=NA)
rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

BSData <-BSData[!rem,]

pData(BSData)$ploidy=c("trisomic", "disomic", "trisomic", "disomic", "trisomic", "disomic", "disomic", "disomic", "trisomic")
pData(BSData)$age=c("2", "2", "2", "2", "11", "2", "2", "2", "2")

TS <- as.factor(paste(pData(BSData)$ploidy, pData(BSData)$age, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts(trisomic.2-disomic.2,levels=design)

aw=arrayWeights(exprs(BSData), design)


fit <- lmFit(exprs(BSData), design,  weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


ETABM473=topTable(contr.fit, coef=1, number=Inf)


ETABM473_sig=ETABM473[which(ETABM473$adj.P.Val< pthreshold),]


ETABM473_sigs=unique(unlist(ens[rownames(ETABM473_sig)]))[!is.na(unique(unlist(ens[rownames(ETABM473_sig)])))]

ETABM473_sig$id=rownames(ETABM473_sig)
ids_df=list_to_df(ens)
ids_df=ids_df[complete.cases(ids_df),]

ETABM473_sig=merge(ETABM473_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
ETABM473_sig=ETABM473_sig[order(ETABM473_sig$adj.P.Val, abs(ETABM473_sig$logFC), decreasing=c(FALSE, TRUE)),]
ETABM473_sig=ETABM473_sig[!duplicated(ETABM473_sig$V1),]


if(length(ETABM473_sigs)>0){
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = ETABM473_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(ETABM473_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
ETABM473_sig_hum=mouse_human[!duplicated(mouse_human$human),]
ETABM473_sigs_hum=unique(ETABM473_sig_hum$human)


list_ensembl$ETABM473=ETABM473_sigs_hum
list_adjpval$ETABM473=ETABM473_sig_hum$adj.P.Val
list_log2FC$ETABM473=ETABM473_sig_hum$logFC
}

```


```{r}
plotMDS(exprs(BSData), col=as.numeric(as.factor(as.factor(paste(pData(BSData)$ploidy)))), pch=as.numeric(as.factor(as.factor(paste(pData(BSData)$age)))))
        
legend("topleft", legend=levels(as.factor(paste(pData(BSData)$ploidy))), fill=as.numeric(as.factor(levels(as.factor(pData(BSData)$ploidy)))))
legend("bottomleft", legend=levels(as.factor(pData(BSData)$age)), pch=as.numeric(as.factor(levels(as.factor(pData(BSData)$age)))))
```



### E-MTAB-312

```{r}
BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/Down_syndrome_array_data.txt", 
                    sep="\t", ProbeID="Probe Set ID", skip=0,
                    columns = list(exprs = "Signal", Detection="Detection"),
                    illuminaAnnotation="Mousev1",annoCols=c("Ensembl"))

fData(BSData)$Ensembl=gsub("---", NA, fData(BSData)$Ensembl)

BSData=normaliseIllumina(BSData, method="quantile", transform="log2")

index1=which(Detection(BSData)[,1]=="P")
index2=which(Detection(BSData)[,2]=="P")
index3=which(Detection(BSData)[,3]=="P")
index4=which(Detection(BSData)[,4]=="P")
index5=which(Detection(BSData)[,5]=="P")
index6=which(Detection(BSData)[,6]=="P")

uno=intersect(intersect(index1, index2), index3)
due=intersect(intersect(index4, index5), index6)

BSData=BSData[intersect(uno, due),]

BSData=BSData[!is.na(fData(BSData)$Ensembl),]

TS <- as.factor(c(rep("disomic",4), "trisomic", "trisomic"))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts(trisomic-disomic,levels=design)

aw=arrayWeights(BSData, design)
corfit<-duplicateCorrelation(BSData,design, block=c(rep(c("nt", "t"),3)))

fit <- lmFit(BSData, design, block=c(rep(c("nt", "t"),3)), cor=corfit$consensus, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


EMTAB312=topTable(contr.fit, coef=1, number=Inf)

EMTAB312_sig=EMTAB312[which(EMTAB312$adj.P.Val< pthreshold),]


EMTAB312_sig=EMTAB312_sig[order(EMTAB312_sig$adj.P.Val, abs(EMTAB312_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMTAB312_sig=EMTAB312_sig[!duplicated(EMTAB312_sig$Ensembl),]
EMTAB312_sig=EMTAB312_sig[!is.na(EMTAB312_sig$Ensembl),]

list_ensembl$EMTAB312=EMTAB312_sig$Ensembl
list_adjpval$EMTAB312=EMTAB312_sig$adj.P.Val
list_log2FC$EMTAB312=EMTAB312_sig$logFC
```


```{r}
plotMDS(exprs(BSData), col=as.numeric(TS), pch=as.numeric(as.factor(as.factor(c(rep(c("nt", "t"),3))))))
        
legend("topright", legend=levels(TS), fill=as.numeric(as.factor(levels(TS))))
legend("bottomright", legend=levels(as.factor(c(rep(c("untreated", "treated"),3)))), pch=as.numeric(as.factor(levels(as.factor(c(rep(c("nt", "t"),3)))))))
```

### E-MEXP-3355

Transcription profiling by array of mouse heart from Ts65Dn

```{r}
phenodata=data.frame(ploidy=c("wildtype",
"wildtype",
"Ts65Dn",
"wildtype",
"Ts65Dn",
"wildtype",
"Ts65Dn",
"wildtype",
"Ts65Dn",
"Ts65Dn"))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-MEXP-3355.raw.1/",phenoData=phenodata, verbose=T)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

require("mogene10sttranscriptcluster.db")
ids <-mget(featureNames(eset), mogene10sttranscriptclusterENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]

TS <- as.factor( phenodata$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)


contrasts <-makeContrasts(Ts65Dn-wildtype,levels=design)

aw=arrayWeights(exprs(eset)[as.character(nomi),], design)

fit <- lmFit(exprs(eset)[nomi,], design, weights = aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


EMEXP3355=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])

EMEXP3355_sig=EMEXP3355[which(EMEXP3355$adj.P.Val< pthreshold),]

EMEXP3355_sigs=unique(unlist(ids[rownames(EMEXP3355_sig)]))

EMEXP3355_sig$id=rownames(EMEXP3355_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

EMEXP3355_sig=merge(EMEXP3355_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
EMEXP3355_sig=EMEXP3355_sig[order(EMEXP3355_sig$adj.P.Val, abs(EMEXP3355_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMEXP3355_sig=EMEXP3355_sig[!duplicated(EMEXP3355_sig$V1),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = EMEXP3355_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(EMEXP3355_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
EMEXP3355_sig_hum=mouse_human[!duplicated(mouse_human$human),]
EMEXP3355_sigs_hum=unique(EMEXP3355_sig_hum$human)


list_ensembl$EMEXP3355=EMEXP3355_sigs_hum
list_adjpval$EMEXP3355=EMEXP3355_sig_hum$adj.P.Val
list_log2FC$EMEXP3355=EMEXP3355_sig_hum$logFC
```

```{r}
plotMDS(eset, col=as.numeric(TS))
        
legend("topright", legend=levels(TS), fill=as.numeric(as.factor(levels(TS))))
```

### E-MATB-1238

```{r}
phenodata=data.frame(sex=c("female",
"male",
"female"
,"male"
,"female"
,"male"
,"female"
,"male"
,"female"
,"male"
,"male"
,"female"), ploidy=c(rep("trisomic",6), rep("disomic", 6)))


data=ReadAffy(celfile.path="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-MTAB-1238.raw.1/",phenoData=phenodata, verbose=T)

eset <- expresso(data,bgcorrect.method="rma",
normalize.method="quantiles",
pmcorrect.method="pmonly",
summary.method="medianpolish")

ids <-mget(featureNames(eset), hgu133plus2ENSEMBL, ifnotfound=NA)
nomi=names(ids)[!is.na(ids)]
# ids=as.character(unlist(ids))

TS <- as.factor( pData(eset)$ploidy)
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)

aw=arrayWeights((eset)[nomi,], design)

corfit<-duplicateCorrelation(exprs(eset)[nomi,],design, block=phenodata$sex)

contrasts <-makeContrasts(trisomic-disomic,levels=design)


fit <- lmFit(exprs(eset)[nomi,], design, block=phenodata$sex, cor=corfit$consensus, weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


EMATB1238=topTable(contr.fit, coef=1, number=dim(exprs(eset)[nomi,])[1])

EMATB1238_sig=EMATB1238[which(EMATB1238$adj.P.Val< pthreshold),]

EMATB1238_sigs=unique(unlist(ids[rownames(EMATB1238_sig)]))[!is.na(unique(unlist(ids[rownames(EMATB1238_sig)])))]

EMATB1238_sig$id=rownames(EMATB1238_sig)
ids_df=list_to_df(ids)
ids_df=ids_df[complete.cases(ids_df),]

EMATB1238_sig=merge(EMATB1238_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
EMATB1238_sig=EMATB1238_sig[order(EMATB1238_sig$adj.P.Val, abs(EMATB1238_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMATB1238_sig=EMATB1238_sig[!duplicated(EMATB1238_sig$V1),]

list_ensembl$EMATB1238=EMATB1238_sigs
list_adjpval$EMATB1238=EMATB1238_sig$adj.P.Val
list_log2FC$EMATB1238=EMATB1238_sig$logFC
```

```{r}
plotMDS(eset[nomi,], col=as.numeric(as.factor(pData(eset)$ploidy)), pch=as.numeric(as.factor(pData(eset)$sex)))
        
legend("bottomleft", legend=levels(as.factor(pData(eset)$ploidy)), fill=as.numeric(as.factor(levels(as.factor(pData(eset)$ploidy)))))
legend("bottomright", legend=levels(as.factor(pData(eset)$sex)), pch=as.numeric(as.factor(levels(as.factor(pData(eset)$sex)))))
```


### E-MTAB-2574

Transcription profiling by array of the Ts(1716)65Dn Down syndrome mouse model bone marrow lineage- c-Kit+ Sca-1+ (LSK), CMP, GMP and MEP progenitor cell populations.

```{r}


BSData=readBeadSummaryData(dataFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-MTAB-2574.raw.1/iltp2346_Sample_Probe_Profile.txt", 
                           qcFile="/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/E-MTAB-2574.raw.1/iltp2346_control probe profile.txt", skip=7,columns = list(exprs = "AVG_Signal", se.exprs="BEAD_STDEV",
                        nObservations = "Avg_NBEADS", Detection="Detection"),
                    illuminaAnnotation="Mousev2",qc.skip=0)

#subset only for trisomic and wt
BSData=BSData[,c(3,5,7,8,11,12,13,16,20,21,24,28,29,33,34,36,37,38,41,44,45,46,47,48)]

pData(BSData)$ploidy =as.factor(c("trisomic", "disomic", "trisomic", "disomic", "disomic", "trisomic", "disomic", "trisomic", "disomic", "disomic", "trisomic", "disomic", "disomic", "trisomic", "trisomic", "disomic", "disomic", "trisomic", "trisomic", "trisomic", "disomic", "trisomic", "disomic", "trisomic"))

pData(BSData)$celltype =as.factor(c("MEP", "GMP", "LSK", "LSK", "CMP", "MEP", "MEP", "MEP", "LSK", "CMP", "GMP", "CMP", "GMP", "GMP", "CMP", "LSK", "MEP", "LSK", "GMP","CMP", "MEP", "LSK", "GMP", "CMP"))

BSData=normaliseIllumina(BSData, method="quantile", transform="log2")

BSData=BSData[which(fData(BSData)$Status=="regular"),]

expressed=apply(Detection(BSData)<0.05,1,any)
BSData=BSData[expressed,]


conversion=read.csv("/Users/ilario/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/probe_id_address_id_wg6mouse.csv",row.names=1, header=T)

ids <-as.character(featureNames(BSData))

conversion=conversion[ids,]

featureNames(BSData)=conversion

ids=featureNames(BSData)
qual <- unlist(mget(ids, illuminaMousev2PROBEQUALITY, ifnotfound=NA))
ens <- mget(ids, illuminaMousev2ENSEMBL, ifnotfound=NA)
rem <- qual=="No match"|qual=="Bad"|is.na(qual)|is.na(ens)

BSData <-BSData[!rem,]



TS <- as.factor(paste(pData(BSData)$ploidy, pData(BSData)$celltype, sep="."))
design <- model.matrix(~0+TS)
colnames(design) <- levels(TS)



contrasts <-makeContrasts("CMP"=trisomic.CMP - disomic.CMP, "LSK"= trisomic.LSK - disomic.LSK, "GMP"=trisomic.GMP - disomic.GMP, "MEP"= trisomic.MEP - disomic.MEP,levels=design)

aw=arrayWeights(exprs(BSData), design)

fit <- lmFit(exprs(BSData), design,  weights=aw)

contr.fit <-eBayes(contrasts.fit(fit, contrasts))


EMTAB2574_CMP=topTable(contr.fit, coef="CMP", number=Inf)
EMTAB2574_GMP=topTable(contr.fit, coef="GMP", number=Inf)
EMTAB2574_LSK=topTable(contr.fit, coef="LSK", number=Inf)
EMTAB2574_MEP=topTable(contr.fit, coef="MEP", number=Inf)


EMTAB2574_CMP_sig=EMTAB2574_CMP[which(EMTAB2574_CMP$adj.P.Val< pthreshold),]
if(length(rownames(EMTAB2574_CMP_sig))>0){
EMTAB2574_CMP_sigs=unique(unlist(ens[rownames(EMTAB2574_CMP_sig)]))[!is.na(unique(unlist(ens[rownames(EMTAB2574_CMP_sig)])))]



mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = EMTAB2574_CMP_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
EMTAB2574_CMP_sigs_hum=unique(mouse_human$human)
length(EMTAB2574_CMP_sigs_hum)
}

EMTAB2574_GMP_sig=EMTAB2574_GMP[which(EMTAB2574_GMP$adj.P.Val< pthreshold),]
if(length(rownames(EMTAB2574_GMP_sig))>0){
EMTAB2574_GMP_sigs=unique(unlist(ens[rownames(EMTAB2574_GMP_sig)]))[!is.na(unique(unlist(ens[rownames(EMTAB2574_GMP_sig)])))]

EMTAB2574_GMP_sig$id=rownames(EMTAB2574_GMP_sig)

ids_df=list_to_df(ens)
ids_df=ids_df[complete.cases(ids_df),]

EMTAB2574_GMP_sig=merge(EMTAB2574_GMP_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
EMTAB2574_GMP_sig=EMTAB2574_GMP_sig[order(EMTAB2574_GMP_sig$adj.P.Val, abs(EMTAB2574_GMP_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMTAB2574_GMP_sig=EMTAB2574_GMP_sig[!duplicated(EMTAB2574_GMP_sig$V1),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = EMTAB2574_GMP_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")
mouse_human=merge(EMTAB2574_GMP_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
EMTAB2574_GMP_sig_hum=mouse_human[!duplicated(mouse_human$human),]
EMTAB2574_GMP_sigs_hum=unique(EMTAB2574_GMP_sig_hum$human)


list_ensembl$EMTAB2574_GMP=EMTAB2574_GMP_sigs_hum
list_adjpval$EMTAB2574_GMP=EMTAB2574_GMP_sig_hum$adj.P.Val
list_log2FC$EMTAB2574_GMP=EMTAB2574_GMP_sig_hum$logFC
}


EMTAB2574_MEP_sig=EMTAB2574_MEP[which(EMTAB2574_MEP$adj.P.Val< pthreshold),]
if(length(rownames(EMTAB2574_MEP_sig))>0){
EMTAB2574_MEP_sigs=unique(unlist(ens[rownames(EMTAB2574_MEP_sig)]))[!is.na(unique(unlist(ens[rownames(EMTAB2574_MEP_sig)])))]

EMTAB2574_MEP_sig$id=rownames(EMTAB2574_MEP_sig)
EMTAB2574_MEP_sig=merge(EMTAB2574_MEP_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
EMTAB2574_MEP_sig=EMTAB2574_MEP_sig[order(EMTAB2574_MEP_sig$adj.P.Val, abs(EMTAB2574_MEP_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMTAB2574_MEP_sig=EMTAB2574_MEP_sig[!duplicated(EMTAB2574_MEP_sig$V1),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = EMTAB2574_MEP_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(EMTAB2574_MEP_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
EMTAB2574_MEP_sig_hum=mouse_human[!duplicated(mouse_human$human),]
EMTAB2574_MEP_sigs_hum=unique(EMTAB2574_MEP_sig_hum$human)


list_ensembl$EMTAB2574_MEP=EMTAB2574_MEP_sigs_hum
list_adjpval$EMTAB2574_MEP=EMTAB2574_MEP_sig_hum$adj.P.Val
list_log2FC$EMTAB2574_MEP=EMTAB2574_MEP_sig_hum$logFC
}

EMTAB2574_LSK_sig=EMTAB2574_LSK[which(EMTAB2574_LSK$adj.P.Val< pthreshold),]
if(length(rownames(EMTAB2574_LSK_sig))>0){
EMTAB2574_LSK_sigs=unique(unlist(ens[rownames(EMTAB2574_LSK_sig)]))[!is.na(unique(unlist(ens[rownames(EMTAB2574_LSK_sig)])))]

EMTAB2574_LSK_sig$id=rownames(EMTAB2574_LSK_sig)
EMTAB2574_LSK_sig=merge(EMTAB2574_LSK_sig, ids_df, by.x="id", by.y="V2", all.x=TRUE, all.y=FALSE)
EMTAB2574_LSK_sig=EMTAB2574_LSK_sig[order(EMTAB2574_LSK_sig$adj.P.Val, abs(EMTAB2574_LSK_sig$logFC), decreasing=c(FALSE, TRUE)),]
EMTAB2574_LSK_sig=EMTAB2574_LSK_sig[!duplicated(EMTAB2574_LSK_sig$V1),]

mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = EMTAB2574_LSK_sigs, mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")


mouse_human=merge(EMTAB2574_LSK_sig,mouse_human, by.x="V1", by.y="mouse", all.x=FALSE, all.y=TRUE)
mouse_human=mouse_human[order(mouse_human$adj.P.Val, abs(mouse_human$logFC), decreasing=c(FALSE, TRUE)),]
EMTAB2574_LSK_sig_hum=mouse_human[!duplicated(mouse_human$human),]
EMTAB2574_LSK_sigs_hum=unique(EMTAB2574_LSK_sig_hum$human)


list_ensembl$EMTAB2574_LSK=EMTAB2574_LSK_sigs_hum
list_adjpval$EMTAB2574_LSK=EMTAB2574_LSK_sig_hum$adj.P.Val
list_log2FC$EMTAB2574_LSK=EMTAB2574_LSK_sig_hum$logFC
}


list_ensembl$EMTAB2574_CMP=character(0)
list_adjpval$EMTAB2574_CMP=numeric(0)
list_log2FC$EMTAB2574_CMP=numeric(0)
```

```{r}
plotMDS(exprs(BSData), col=as.numeric(as.factor(pData(BSData)$ploidy)), pch=as.numeric(as.factor(pData(BSData)$celltype)))
        
legend("topright", legend=levels(as.factor(pData(BSData)$ploidy)), fill=as.numeric(as.factor(pData(BSData)$ploidy)))
legend("topleft", legend=levels(as.factor(pData(BSData)$celltype)), pch=as.numeric(as.factor(levels(as.factor(pData(BSData)$celltype)))))
```


### SRP026566


Triplication of a 21q22 region contributes to B cell transformation through HMGN1 overexpression and loss of histone H3 Lys27 trimethylation.

Down syndrome (DS) confers a 20-fold increased risk of B cell acute lymphoblastic leukemia (ALL), yet the mechanisms underlying this association are undefined.  We show that triplication of only 31 genes orthologous to the putative DS Critical Region (DSCR) on chr.21q22 is sufficient to promote B cell-autonomous self-renewal, in vivo maturation defects and leukemogenesis in concert with BCR-ABL.  DSCR triplication results in histone H3 lysine 27 (H3K27) hypomethylation at gene promoters and a transcriptional signature characterized by de-repression of genes targeted by polycomb repressor complex 2 (PRC2), which methylates H3K27.  The same signature is highly enriched among human DS-associated ALLs.  Pharmacologic inhibition of PRC2 function is sufficient to confer self-renewal in wild-type B cells while inhibition of H3K27me3 demethylases completely blocks self-renewal induced by DSCR triplication.  Finally, overexpression of the DSCR factor HMGN1, a nucleosome remodeling protein that suppresses H3K27me3, is necessary for self-renewal in B cells with DSCR triplication.


```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP026566.RData")
require(DESeq2)
summary(res)
resSRP026566=res
universe_human=rownames(resSRP026566)
resSRP026566=as.data.frame(resSRP026566)
resSRP026566=resSRP026566[order(resSRP026566$padj),]
resSRP026566=resSRP026566[complete.cases(resSRP026566),]
resSRP026566_sig=resSRP026566[which(resSRP026566$padj<pthreshold),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = rownames(resSRP026566_sig), mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

resSRP026566_sig$ens=rownames(resSRP026566_sig)

mouse_human=merge(resSRP026566_sig,mouse_human, by.x="ens", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$padj, abs(mouse_human$log2FoldChange), decreasing=c(FALSE, TRUE)),]
resSRP026566_sig_hum=mouse_human[!duplicated(mouse_human$human),]

resSRP026566_sigs_hum=unique(resSRP026566_sig_hum$human)



list_ensembl$GSE48553=resSRP026566_sigs_hum
list_adjpval$GSE48553=resSRP026566_sig_hum$padj
list_log2FC$GSE48553=resSRP026566_sig_hum$log2FoldChange

#dim(resSRP026566[which(resSRP026566$padj<.01),])
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


### SRP072769

Although it is clear that trisomy 21 causes Down syndrome, the molecular events acting downstream of the trisomy remain ill defined. Using complementary genomics analyses, we identified the interferon pathway as the major signaling cascade consistently activated by trisomy 21 in human cells. Transcriptome analysis revealed that trisomy 21 activates the interferon transcriptional response in fibroblast and lymphoblastoid cell lines, as well as circulating monocytes and T cells. Trisomy 21 cells show increased induction of interferon-stimulated genes and decreased expression of ribosomal proteins and translation factors. An shRNA screen determined that the interferon-activated kinases JAK1 and TYK2 suppress proliferation of trisomy 21 fibroblasts, and this defect is rescued by pharmacological JAK inhibition. Therefore, we propose that interferon activation, likely via increased gene dosage of the four interferon receptors encoded on chromosome 21, contributes to many of the clinical impacts of trisomy 21, and that interferon antagonists could have therapeutic benefits.

#### Fibroblasts

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP072769_fibroblasts.RData")

resSRP072769=res
summary(res)
universe_human=rownames(resSRP072769)
resSRP072769=as.data.frame(resSRP072769)
resSRP072769=resSRP072769[order(resSRP072769$padj),]
resSRP072769=resSRP072769[complete.cases(resSRP072769),]
resSRP072769_sig=resSRP072769[which(resSRP072769$padj<pthreshold),]

resSRP072769_sigs=rownames(resSRP072769_sig)

#dim(resSRP026566[which(resSRP026566$padj<.01),])


list_ensembl$GSE79842fibroblasts=resSRP072769_sigs
list_adjpval$GSE79842fibroblasts=resSRP072769_sig$padj
list_log2FC$GSE79842fibroblasts=resSRP072769_sig$log2FoldChange
```



```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment,shape=colData$sex)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```



#### Lymphoblastoids

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP072769_lymphoblastoids.RData")
summary(res)
resSRP072769_l=res
universe_human=rownames(resSRP072769_l)
resSRP072769_l=as.data.frame(resSRP072769_l)
resSRP072769_l=resSRP072769_l[order(resSRP072769_l$padj),]
resSRP072769_l=resSRP072769_l[complete.cases(resSRP072769_l),]
resSRP072769_l_sig=resSRP072769_l[which(resSRP072769_l$padj<pthreshold),]

resSRP072769_l_sigs=rownames(resSRP072769_l_sig)

#dim(resSRP026566[which(resSRP072769_l$padj<.01),])


list_ensembl$GSE79842lymphoblastoid=resSRP072769_l_sigs
list_adjpval$GSE79842lymphoblastoid=resSRP072769_l_sig$padj
list_log2FC$GSE79842lymphoblastoid=resSRP072769_l_sig$log2FoldChange
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

### SRP078911

RNA was sequenced from Normal and DP 16 mice to identify consistent changes in gene expression across cohorts


```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP078911.RData")
summary(res)
resSRP078911=res
universe_human=rownames(resSRP078911)
resSRP078911=as.data.frame(resSRP078911)
resSRP078911=resSRP078911[order(resSRP078911$padj),]
resSRP078911=resSRP078911[complete.cases(resSRP078911),]
resSRP078911_sig=resSRP078911[which(resSRP078911$padj<pthreshold),]

resSRP078911_sigs=rownames(resSRP078911_sig)

resSRP078911_sig$ens=rownames(resSRP078911_sig)
  
mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = rownames(resSRP078911_sig), mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)
colnames(mouse_human)=c("mouse", "human")

mouse_human=merge(resSRP078911_sig,mouse_human, by.x="ens", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$padj, abs(mouse_human$log2FoldChange), decreasing=c(FALSE, TRUE)),]
resSRP078911_sig_hum=mouse_human[!duplicated(mouse_human$human),]

resSRP078911_sigs_hum=unique(resSRP078911_sig_hum$human)
  
length(resSRP078911_sigs_hum)

#dim(resSRP078911[which(resSRP078911$padj<.01),])

list_ensembl$GSE84526=resSRP078911_sigs_hum
list_adjpval$GSE84526=resSRP078911_sig_hum$padj
list_log2FC$GSE84526=resSRP078911_sig_hum$log2FoldChange
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


### GSE84531

#### Monocytes

```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP078912_monocytes.RData")
summary(res)
resSRP078912=res
universe_human=rownames(resSRP078912)
resSRP078912=as.data.frame(resSRP078912)
resSRP078912=resSRP078912[order(resSRP078912$padj),]
resSRP078912=resSRP078912[complete.cases(resSRP078912),]
resSRP078912_sig=resSRP078912[which(resSRP078912$padj<pthreshold),]

resSRP078912_sigs=rownames(resSRP078912_sig)

#dim(resSRP026566[which(resSRP026566$padj<.01),])

list_ensembl$GSE84531monocytes=resSRP078912_sigs
list_adjpval$GSE84531monocytes=resSRP078912_sig$padj
list_log2FC$GSE84531monocytes=resSRP078912_sig$log2FoldChange
```



```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=colData$treatment,shape=colData$sex)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```


### primary mouse proB Ts1Rhr cells expanded in culture

The dataset was taken from this article:

https://www.ncbi.nlm.nih.gov/pubmed/30428356

3 sample euploid vs 3 trisomic

```{r}

load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP164925.RData")

summary(res)
resSRP164925=as.data.frame(res)

summary(resLFC)


resSRP164925=resSRP164925[order(resSRP164925$padj),]
resSRP164925_sig=resSRP164925[complete.cases(resSRP164925),]
resSRP164925_sig=resSRP164925[which(resSRP164925$padj < pthreshold),]


mouse_human=getLDS(attributes = c("ensembl_gene_id"), filters = "ensembl_gene_id", values = rownames(resSRP164925_sig), mart = mouse, attributesL = c("ensembl_gene_id"), martL = human)

colnames(mouse_human)=c("mouse", "human")
resSRP164925_sig$ens=rownames(resSRP164925_sig)

mouse_human=merge(resSRP164925_sig,mouse_human, by.x="ens", by.y="mouse", all.x=FALSE, all.y=TRUE)

mouse_human=mouse_human[order(mouse_human$padj, abs(mouse_human$log2FoldChange), decreasing=c(FALSE, TRUE)),]
resSRP164925_sig_hum=mouse_human[!duplicated(mouse_human$human),]

resSRP164925_sigs_hum=unique(resSRP164925_sig_hum$human)
  
length(resSRP164925_sigs_hum)


list_ensembl$GSE121066=resSRP164925_sigs_hum
list_adjpval$GSE121066=resSRP164925_sig_hum$padj
list_log2FC$GSE121066=resSRP164925_sig_hum$log2FoldChange

```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=treatment)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```




#### GSE128622: White Blood cells

Stranded (SRP188973) and unstranded (SRP188969) data


```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP188973_SRP188969.RData")
summary(res)
resGSE128621=res
universe_human=rownames(resGSE128621)
resGSE128621=as.data.frame(resGSE128621)
resGSE128621=resGSE128621[order(resGSE128621$padj),]
resGSE128621=resGSE128621[complete.cases(resGSE128621),]
resGSE128621_sig=resGSE128621[which(resGSE128621$padj<pthreshold),]

resGSE128621_sigs=rownames(resGSE128621_sig)

#dim(resGSE128621[which(resGSE128621$padj<.01),])

list_ensembl$GSE128621=resGSE128621_sigs
list_adjpval$GSE128621=resGSE128621_sig$padj
list_log2FC$GSE128621=resGSE128621_sig$log2FoldChange
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("karyotype"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=colData$karyotype,shape=as.factor(colData$batch))) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}

ggplot(data, aes(PC1, PC2, color=colData$karyotype,shape=as.factor(colData$sex))) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r saving}
save(list_ensembl,  list_adjpval, list_log2FC, file="201911_de_list_ds.RData")
```

#### T cell


```{r}
load("~/OneDrive - CRG - Centre de Regulacio Genomica/Dropbox (CRG)/metanalysis/SRP078912_tcells.RData")
summary(res)
resSRP078912_l=res
universe_human=rownames(resSRP078912_l)
resSRP078912_l=as.data.frame(resSRP078912_l)
resSRP078912_l=resSRP078912_l[order(resSRP078912_l$padj),]
resSRP078912_l=resSRP078912_l[complete.cases(resSRP078912_l),]
resSRP078912_l_sig=resSRP078912_l[which(resSRP078912_l$padj<pthreshold),]

resSRP078912_l_sigs=rownames(resSRP078912_l_sig)

#dim(resSRP078912_l[which(resSRP078912_l$padj<.01),])

list_ensembl$GSE84531tcells=resSRP078912_l_sigs
list_adjpval$GSE84531tcells=resSRP078912_l_sig$padj
list_log2FC$GSE84531tcells=resSRP078912_l_sig$log2FoldChange
```


```{r}
require(ggplot2)
#transformation of count data (regularized log), that normalize for the library size and ribimizes differences between samples for rows with small counts)
vsd <- vst(dds, blind=FALSE)
notAllZero <- (rowSums(counts(dds))>0)

edata=assay(vsd[notAllZero,])

#select only non zero counts
#cluster analysis
data <- plotPCA(vsd[notAllZero,], intgroup=c("treatment"), returnData=TRUE)
percentVar <- round(100 * attr(data, "percentVar"))
ggplot(data, aes(PC1, PC2, color=colData$treatment,shape=colData$sex)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

```{r}
save(list_ensembl,list_adjpval, list_log2FC, file="de_list_ds.RData")
```



```{r saving2}
save(list_ensembl,  list_adjpval, list_log2FC, file="201911_de_list_ds2.RData")
```
